---
title: "Cohort 2 Function"
author: "Liz Mallott"
date: "1/28/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Overview of analysis
This analysis examines if the composition of COGs and MetaCyc reaction pathways differs based on self-reported Ethnicity and what specific taxa are differentially abundant. 
HUMAnN (v3.0.0.alpha.4) was used to profile the functional composition of metagenomes.

## COGs

### Set up the environment
Load the necessary libraries.
```{r libraries_genes, results=FALSE, error=FALSE}
library(tidyverse)
library(vegan)
library(ggplot2)
library(ggtext)
library(cowplot)
library(car)
library(multcomp)
library(ggpubr)
library(fdrtool)
```

### Importing data
Import metadata
```{r import_meta_gene, error=FALSE}
metadata_gene = read_tsv("tables/year2_sample_mapping_genes.tsv") %>% 
  mutate(Ethnicity = case_when(Ethnicity == "African_American" ~ "Black", 
                               Ethnicity == "Caucasian" ~ "White"))
```

Import normalized, unstratified gene abundance table with outliers (30f5, 37f1, and 33f1) removed. Transpose.
```{r import_gene, error=FALSE}
gene_table = read_tsv("year2_genefamilies_cog_cpm_unstratified_noout.tsv")
genes = gene_table %>% gather(file, count, 2:160) %>% 
  spread(Gene_Family, count)
```

### Filter samples

Filter out samples that have no identified species or are controls (blanks and mock communities). Separate oral samples from fecal samples.

```{r filter_gene}
genes_fecal = genes %>% left_join(metadata_gene, by = "file") %>% 
  filter(source == "fecal") 
genes_fecal_filtered = genes_fecal %>% 
  mutate(total = rowSums(genes_fecal[,2:56005])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0) 
write_csv(genes_fecal_filtered, "gene_family_cog_cpm_fecal.csv")
genes_fecal_filtered = read_csv("tables/gene_family_cog_cpm_fecal.csv")

genes_oral = genes %>% left_join(metadata_gene, by = "file") %>% 
  filter(source == "oral")
genes_oral_filtered = genes_oral %>% 
  mutate(total = rowSums(genes_oral[,2:56005])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0) 
write_csv(genes_oral, "gene_family_cog_cpm_oral.csv")
genes_oral_filtered = read_csv("tables/gene_family_cog_cpm_oral.csv")
```

### Calculating beta diversity metrics
Vegan (2.5-7) is used to calculate the Bray-Curtis dissimilarity and Jaccard similarity metrics.
```{r beta_metrics_gene, error=FALSE}
brayf_gene = vegdist(genes_fecal_filtered[,2:40168], method = "bray")
jaccf_gene = vegdist(genes_fecal_filtered[,2:40168], method = "jaccard", binary = T)
brayo_gene = vegdist(genes_oral_filtered[,2:32391], method = "bray")
jacco_gene = vegdist(genes_oral_filtered[,2:32391], method = "jaccard", binary = T)

genes_fecal_filtered_black = genes_fecal_filtered %>% filter(Ethnicity == "Black")
genes_fecal_filtered_white = genes_fecal_filtered %>% filter(Ethnicity == "White")
brayf_gene_black = vegdist(genes_fecal_filtered_black[,2:40168], method = "bray")
jaccf_gene_black = vegdist(genes_fecal_filtered_black[,2:40168], method = "jaccard", binary = T)
brayf_gene_white = vegdist(genes_fecal_filtered_white[,2:40168], method = "bray")
jaccf_gene_white = vegdist(genes_fecal_filtered_white[,2:40168], method = "jaccard", binary = T)

genes_oral_filtered_black = genes_oral_filtered %>% filter(Ethnicity == "Black")
genes_oral_filtered_white = genes_oral_filtered %>% filter(Ethnicity == "White")
brayo_gene_black = vegdist(genes_oral_filtered_black[,2:32391], method = "bray")
jacco_gene_black = vegdist(genes_oral_filtered_black[,2:32391], method = "jaccard", binary = T)
brayo_gene_white = vegdist(genes_oral_filtered_white[,2:32391], method = "bray")
jacco_gene_white = vegdist(genes_oral_filtered_white[,2:32391], method = "jaccard", binary = T)
```

### PERMANOVAs - All Samples
Differences in the COG profiles of the community associated with self-reported Ethnicity is assessed using permutational multivariate analysis of variance.

#### Bray-Curtis
Fecal
```{r permanova_brayf_gene}
adonis2(brayf_gene ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = genes_fecal_filtered, 
        permutations = 4999)

anova(betadisper(brayf_gene, group = genes_fecal_filtered$Ethnicity))

anova(betadisper(brayf_gene, group = genes_fecal_filtered$`Study period`))

anova(betadisper(brayf_gene, group = genes_fecal_filtered$`Antibiotic use`))

anova(betadisper(brayf_gene, group = genes_fecal_filtered$`Contraceptive use`))

anova(betadisper(brayf_gene, group = genes_fecal_filtered$subject_id))
```

Oral
```{r permanova_brayo_gene}
adonis2(brayo_gene ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = genes_oral_filtered, 
        permutations = 4999)

anova(betadisper(brayo_gene, group = genes_oral_filtered$Ethnicity))

anova(betadisper(brayo_gene, group = genes_oral_filtered$`Study period`))

anova(betadisper(brayo_gene, group = genes_oral_filtered$`Antibiotic use`))

anova(betadisper(brayo_gene, group = genes_oral_filtered$`Contraceptive use`))

anova(betadisper(brayo_gene, group = genes_oral_filtered$subject_id))
```
NMDS visualization of Bray-Curtis dissimilarity with all samples.
```{r nmds_bray_gene, echo=FALSE, error=FALSE}
mds_bray_f_gene<-metaMDS(brayf_gene, k=2, trymax=499)
mds_bray_f_gene_points<-mds_bray_f_gene$points
mds_bray_f_gene_points2<-merge(x=mds_bray_f_gene_points, 
                               y = genes_fecal_filtered, 
                                  by.x = "row.names", by.y = "row.names")
brayfg <- ggplot(mds_bray_f_gene_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_f_gene_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("COGs - Bray-Curtis - Fecal") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.001, r<sup>2</sup> = 4.2% (All)</b><br><i>p</i> = 0.348, r<sup>2</sup> = 6.4% (Before)<br><b><i>p</i> = 0.013, r<sup>2</sup> = 4.3% (During)</b><br><i>p</i> = 0.125, r<sup>2</sup> = 5.3% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.160",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayfg

mds_bray_o_gene<-metaMDS(brayo_gene, k=2, trymax=499)
mds_bray_o_gene_points<-mds_bray_o_gene$points
mds_bray_o_gene_points2<-merge(x=mds_bray_o_gene_points, 
                               y = genes_oral_filtered, 
                                  by.x = "row.names", by.y = "row.names")
brayog <- ggplot(mds_bray_o_gene_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_o_gene_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("COGs - Bray-Curtis - Oral") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.053, r<sup>2</sup> = 6.0% (All)<br><i>p</i> = 0.272, r<sup>2</sup> = 7.3% (Before)<br><i>p</i> = 0.084, r<sup>2</sup> = 10.9% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.111",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayog

brayfgy2s <- ggplot(mds_bray_f_gene_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_f_gene_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.975, r<sup>2</sup> = 1.0% (Both)<br><i>p</i> = 0.998, r<sup>2</sup> = 1.3% (Black)<br><i>p</i> = 0.893, r<sup>2</sup> = 2.4% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.160",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayfgy2s

brayogy2s <- ggplot(mds_bray_o_gene_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","After"="#2171b5")) + 
  scale_fill_manual(values = c("Before"="#238b45","After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_o_gene_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.956, r<sup>2</sup> = 1.3% (Both)<br><i>p</i> = 0.877, r<sup>2</sup> = 2.6% (Black)<br><i>p</i> = 0.428, r<sup>2</sup> = 7.3% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.111",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayogy2s
```

```{r nmds_save_bray_gene}
tiff(file = "figures/nmds_plot_cog_bray_e.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayfg + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfg + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayog + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()

tiff(file = "figures/nmds_plot_cog_bray_s.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayfgs + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfgs + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayogs + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
#### Jaccard
Fecal
```{r permanova_jaccf_gene}
adonis2(jaccf_gene ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = genes_fecal_filtered, 
        permutations = 4999)

anova(betadisper(jaccf_gene, group = genes_fecal_filtered$Ethnicity))

anova(betadisper(jaccf_gene, group = genes_fecal_filtered$`Study period`))

anova(betadisper(jaccf_gene, group = genes_fecal_filtered$`Antibiotic use`))

anova(betadisper(jaccf_gene, group = genes_fecal_filtered$`Contraceptive use`))

anova(betadisper(jaccf_gene, group = genes_fecal_filtered$subject_id))
```

Oral
```{r permanova_jacco_gene}
adonis2(jacco_gene ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = genes_oral_filtered, 
        permutations = 4999)

anova(betadisper(jacco_gene, group = genes_oral_filtered$Ethnicity))

anova(betadisper(jacco_gene, group = genes_oral_filtered$`Study period`))

anova(betadisper(jacco_gene, group = genes_oral_filtered$`Antibiotic use`))

anova(betadisper(jacco_gene, group = genes_oral_filtered$`Contraceptive use`))

anova(betadisper(jacco_gene, group = genes_oral_filtered$subject_id))
```

NMDS visualization of Jaccard similarity with all samples.
```{r nmds_jacc, echo=FALSE, error=FALSE}
mds_jacc_f_gene<-metaMDS(jaccf_gene, k=2, trymax=499, distance = "jaccard")
mds_jacc_f_gene_points<-mds_jacc_f_gene$points
mds_jacc_f_gene_points2<-merge(x=mds_jacc_f_gene_points, 
                               y = genes_fecal_filtered, 
                                  by.x = "row.names", by.y = "row.names")
jaccfg <- ggplot(mds_jacc_f_gene_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_f_gene_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Fecal") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, r<sup>2</sup> = 4.2% (All)</b><br><i>p</i> = 0.243, r<sup>2</sup> = 7.3% (Before)<br><b><i>p</i> = 0.004, r<sup>2</sup> = 4.3% (During)</b><br><i>p</i> = 0.092, r<sup>2</sup> = 5.3% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.145",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccfg

mds_jacc_o_gene<-metaMDS(jacco_gene, k=2, trymax=499, distance = "jaccard")
mds_jacc_o_gene_points<-mds_jacc_o_gene$points
mds_jacc_o_gene_points2<-merge(x=mds_jacc_o_gene_points, 
                          y = genes_oral_filtered, 
                                  by.x = "row.names", by.y = "row.names")
jaccog <- ggplot(mds_jacc_o_gene_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_o_gene_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Oral") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.058, r<sup>2</sup> = 5.3% (All)<br><i>p</i> = 0.289, r<sup>2</sup> = 6.9% (Before)<br><i>p</i> = 0.084, r<sup>2</sup> = 9.9% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.111",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccog

jaccfgs <- ggplot(mds_jacc_f_gene_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_f_gene_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.993, r<sup>2</sup> = 1.0% (Both)<br><i>p</i> = 0.997, r<sup>2</sup> = 1.5% (Black)<br><i>p</i> = 0.904, r<sup>2</sup> = 2.6% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.145",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccfgs

jaccogs <- ggplot(mds_jacc_o_gene_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","After"="#2171b5")) + 
  scale_fill_manual(values = c("Before"="#238b45","After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_o_gene_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Oral")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.985, r<sup>2</sup> = 1.4% (Both)<br><i>p</i> = 0.955, r<sup>2</sup> = 2.6% (Black)<br><i>p</i> = 0.440, r<sup>2</sup> = 7.4% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.111",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccogs
```

```{r nmds_save_jacc_gene}
tiff(file = "figures/nmds_plot_cog_jacc_e.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccfg + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfg + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccog + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()

tiff(file = "figures/nmds_plot_cog_jacc_s.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccfgs + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfgs + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccogs + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
### PERMANOVAs - Before samples

Filter samples from before the diet intervention and recalculate Bray-Curtis and Jaccard metrics
```{r filter_gene_before}
gene_fecal_before = genes_fecal_filtered %>% 
  filter(`Study period` == "Before")
gene_oral_before = genes_oral_filtered %>% 
  filter(`Study period` == "Before")

brayf_gene_b = vegdist(gene_fecal_before[,2:40168], method = "bray")
jaccf_gene_b = vegdist(gene_fecal_before[,2:40168], method = "jaccard")
brayo_gene_b = vegdist(gene_oral_before[,2:32391], method = "bray")
jacco_gene_b = vegdist(gene_oral_before[,2:32391], method = "jaccard")
```
#### Bray-Curtis - Before
Fecal
```{r permanova_brayf_gene_before}
adonis2(brayf_gene_b ~ Ethnicity, data = gene_fecal_before, 
        permutations = 4999)
```

Oral
```{r permanova_brayo_gene}
adonis2(brayo_gene_b ~ Ethnicity, data = gene_oral_before, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with all samples.
```{r nmds_bray_gene_before, echo=FALSE, error=FALSE}
mds_bray_f_geneb<-metaMDS(brayf_gene_b, k=2, trymax=499)
mds_bray_f_geneb_points<-mds_bray_f_geneb$points
mds_bray_f_geneb_points2<-merge(x=mds_bray_f_geneb_points, 
                               y = gene_fecal_before, 
                                  by.x = "row.names", by.y = "row.names")
brayfgb <- ggplot(mds_bray_f_geneb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity, 
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Bray-Curtis - Fecal - Before")
brayfgb

mds_bray_o_geneb<-metaMDS(brayo_gene_b, k=2, trymax=499)
mds_bray_o_geneb_points<-mds_bray_o_geneb$points
mds_bray_o_geneb_points2<-merge(x=mds_bray_o_geneb_points, 
                               y = gene_oral_before, 
                                  by.x = "row.names", by.y = "row.names")
brayogb <- ggplot(mds_bray_o_geneb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Bray-Curtis - Oral - Before")
brayogb
```

```{r nmds_save_bray_gene_before}
tiff(file = "figures/nmds_plot_cog_bray_before.tif", 
     res = 150, width = 18, height = 7, units = "in")
legend1 = get_legend(brayfgb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfgb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayogb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
#### Jaccard
Fecal
```{r permanova_jaccf_gene_before}
adonis2(jaccf_gene_b ~ Ethnicity, data = gene_fecal_before, 
        permutations = 4999)
```

Oral
```{r permanova_jacco_gene_before}
adonis2(jacco_gene_b ~ Ethnicity, data = gene_oral_before, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with all samples.
```{r nmds_jacc_before, echo=FALSE, error=FALSE}
mds_jacc_f_geneb<-metaMDS(jaccf_gene_b, k=2, trymax=999, distance = "jaccard")
mds_jacc_f_geneb_points<-mds_jacc_f_geneb$points
mds_jacc_f_geneb_points2<-merge(x=mds_jacc_f_geneb_points, 
                               y = gene_fecal_before, 
                                  by.x = "row.names", by.y = "row.names")
jaccfgb <- ggplot(mds_jacc_f_geneb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Fecal - Before")
jaccfgb

mds_jacc_o_geneb<-metaMDS(jacco_gene_b, k=2, trymax=499, distance = "jaccard")
mds_jacc_o_geneb_points<-mds_jacc_o_geneb$points
mds_jacc_o_geneb_points2<-merge(x=mds_jacc_o_geneb_points, 
                          y = gene_oral_before, 
                                  by.x = "row.names", by.y = "row.names")
jaccogb <- ggplot(mds_jacc_o_geneb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Oral - Before")
jaccogb
```
```{r nmds_save_jacc_gene_before}
tiff(file = "figures/nmds_plot_cog_jacc_before.tif", 
     res = 150, width = 18, height = 7, units = "in")
legend1 = get_legend(jaccfgb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfgb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccogb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
### PERMANOVAs - During samples

Filter samples from during the diet intervention and recalculate Bray-Curtis and Jaccard metrics
```{r filter_gene_during}
gene_fecal_during = genes_fecal_filtered %>% 
  filter(`Study period` == "During")

brayf_gene_d = vegdist(gene_fecal_during[,2:40168], method = "bray")
jaccf_gene_d = vegdist(gene_fecal_during[,2:40168], method = "jaccard")
```
#### Bray-Curtis - During
Fecal
```{r permanova_brayf_gene_during}
adonis2(brayf_gene_d ~ Ethnicity, data = gene_fecal_during, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with during samples.
```{r nmds_bray_gene_during, echo=FALSE, error=FALSE}
mds_bray_f_gened<-metaMDS(brayf_gene_d, k=2, trymax=499)
mds_bray_f_gened_points<-mds_bray_f_gened$points
mds_bray_f_gened_points2<-merge(x=mds_bray_f_gened_points, 
                               y = gene_fecal_during, 
                                  by.x = "row.names", by.y = "row.names")
brayfgd <- ggplot(mds_bray_f_gened_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity, 
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Bray-Curtis - Fecal - During")
brayfgd
```

```{r nmds_save_bray_gene_during}
tiff(file = "figures/nmds_plot_cog_bray_during.tif", 
     res = 150, width = 9, height = 7, units = "in")
brayfgd
dev.off()
```
#### Jaccard
Fecal
```{r permanova_jaccf_gene_during}
adonis2(jaccf_gene_d ~ Ethnicity, data = gene_fecal_during, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with during samples.
```{r nmds_jacc_during, echo=FALSE, error=FALSE}
mds_jacc_f_gened<-metaMDS(jaccf_gene_d, k=2, trymax=999, distance = "jaccard")
mds_jacc_f_gened_points<-mds_jacc_f_gened$points
mds_jacc_f_gened_points2<-merge(x=mds_jacc_f_gened_points, 
                               y = gene_fecal_during, 
                                  by.x = "row.names", by.y = "row.names")
jaccfgd <- ggplot(mds_jacc_f_gened_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Fecal - During")
jaccfgd
```

```{r nmds_save_jacc_gene_during}
tiff(file = "figures/nmds_plot_cog_jacc_during.tif", 
     res = 150, width = 9, height = 7, units = "in")
jaccfgd
dev.off()
```
### PERMANOVAs - After samples

Filter samples from after the diet intervention and recalculate Bray-Curtis and Jaccard metrics
```{r filter_gene_before}
gene_fecal_after = genes_fecal_filtered %>% 
  filter(`Study period` == "After")
gene_oral_after = genes_oral_filtered %>% 
  filter(`Study period` == "After")

brayf_gene_a = vegdist(gene_fecal_after[,2:40168], method = "bray")
jaccf_gene_a = vegdist(gene_fecal_after[,2:40168], method = "jaccard")
brayo_gene_a = vegdist(gene_oral_after[,2:32391], method = "bray")
jacco_gene_a = vegdist(gene_oral_after[,2:32391], method = "jaccard")
```
#### Bray-Curtis - After
Fecal
```{r permanova_brayf_gene_after}
adonis2(brayf_gene_a ~ Ethnicity, data = gene_fecal_after, 
        permutations = 4999)
```

Oral
```{r permanova_brayo_gene_after}
adonis2(brayo_gene_a ~ Ethnicity, data = gene_oral_after, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with after samples.
```{r nmds_bray_gene_after, echo=FALSE, error=FALSE}
mds_bray_f_genea<-metaMDS(brayf_gene_a, k=2, trymax=499)
mds_bray_f_genea_points<-mds_bray_f_genea$points
mds_bray_f_genea_points2<-merge(x=mds_bray_f_genea_points, 
                               y = gene_fecal_after, 
                                  by.x = "row.names", by.y = "row.names")
brayfga <- ggplot(mds_bray_f_genea_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity, 
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Bray-Curtis - Fecal - After")
brayfga

mds_bray_o_genea<-metaMDS(brayo_gene_a, k=2, trymax=499)
mds_bray_o_genea_points<-mds_bray_o_genea$points
mds_bray_o_genea_points2<-merge(x=mds_bray_o_genea_points, 
                               y = gene_oral_after, 
                                  by.x = "row.names", by.y = "row.names")
brayoga <- ggplot(mds_bray_o_genea_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Bray-Curtis - Oral - After")
brayoga
```

```{r nmds_save_bray_gene_after}
tiff(file = "figures/nmds_plot_cog_bray_after.tif", 
     res = 150, width = 18, height = 7, units = "in")
legend1 = get_legend(brayfga + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfga + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayoga + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
#### Jaccard - After
Fecal
```{r permanova_jaccf_gene_after}
adonis2(jaccf_gene_a ~ Ethnicity, data = gene_fecal_after, 
        permutations = 4999)
```

Oral
```{r permanova_jacco_gene_after}
adonis2(jacco_gene_a ~ Ethnicity, data = gene_oral_after, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with all samples.
```{r nmds_jacc_after, echo=FALSE, error=FALSE}
mds_jacc_f_genea<-metaMDS(jaccf_gene_a, k=2, trymax=999, distance = "jaccard")
mds_jacc_f_genea_points<-mds_jacc_f_genea$points
mds_jacc_f_genea_points2<-merge(x=mds_jacc_f_genea_points, 
                               y = gene_fecal_after, 
                                  by.x = "row.names", by.y = "row.names")
jaccfga <- ggplot(mds_jacc_f_genea_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Fecal - After")
jaccfga

mds_jacc_o_genea<-metaMDS(jacco_gene_a, k=2, trymax=499, distance = "jaccard")
mds_jacc_o_genea_points<-mds_jacc_o_genea$points
mds_jacc_o_genea_points2<-merge(x=mds_jacc_o_genea_points, 
                          y = gene_oral_after, 
                                  by.x = "row.names", by.y = "row.names")
jaccoga <- ggplot(mds_jacc_o_genea_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Oral - After")
jaccoga
```

```{r nmds_save_jacc_gene_after}
tiff(file = "figures/nmds_plot_cog_jacc_after.tif", 
     res = 150, width = 18, height = 7, units = "in")
legend1 = get_legend(jaccfga + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfga + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccoga + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

# Pathway Abundances

### Set up the environment
Load the necessary libraries.
```{r libraries_path, results=FALSE, error=FALSE}
library(tidyverse)
library(vegan)
library(ggplot2)
library(cowplot)
library(car)
library(multcomp)
library(ggpubr)
library(fdrtool)
```

### Importing data
Import metadata
```{r import_meta_path, error=FALSE}
metadata_path = read_tsv("year2_sample_mapping_pathways.tsv") %>% 
  mutate(Ethnicity = case_when(Ethnicity == "African_American" ~ "Black", 
                               Ethnicity == "Caucasian" ~ "White"))
```

Import normalized, unstratified gene abundance table with outliers (30f5, 37f1, and 33f1) removed. Transpose.
```{r import_path, error=FALSE}
path_table = read_tsv("year2_pathabundance_relab_unstratified_noout.tsv")
paths = path_table %>% gather(file, count, 2:159) %>% 
  spread(Pathway, count)
```

### Filter samples

Filter out samples that have no identified species or are controls (blanks and mock communities). Separate oral samples from fecal samples.

```{r filter_path}
paths_fecal = paths %>% left_join(metadata_path, by = "file") %>% 
  filter(source == "fecal") 
paths_fecal_filtered = paths_fecal %>% 
  mutate(total = rowSums(paths_fecal[,2:515])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
write_csv(paths_fecal_filtered, "path_abundance_relab_fecal.csv")

paths_oral = paths %>% left_join(metadata_path, by = "file") %>% 
  filter(source == "oral")
paths_oral_filtered = paths_oral %>% 
  mutate(total = rowSums(paths_oral[,2:515])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
write_csv(paths_oral, "path_abundance_relab_oral.csv")
```

### Calculating beta diversity metrics
Vegan (2.5-7) is used to calculate the Bray-Curtis dissimilarity and Jaccard similarity metrics.
```{r beta_metrics_path, error=FALSE}
brayf_path = vegdist(paths_fecal_filtered[,2:425], method = "bray")
jaccf_path = vegdist(paths_fecal_filtered[,2:425], method = "jaccard")
brayo_path = vegdist(paths_oral_filtered[,2:348], method = "bray")
jacco_path = vegdist(paths_oral_filtered[,2:348], method = "jaccard")

paths_fecal_filtered_black = paths_fecal_filtered %>% filter(Ethnicity == "Black")
paths_fecal_filtered_white = paths_fecal_filtered %>% filter(Ethnicity == "White")
brayf_path_black = vegdist(paths_fecal_filtered_black[,2:425], method = "bray")
jaccf_path_black = vegdist(paths_fecal_filtered_black[,2:425], method = "jaccard")
brayf_path_white = vegdist(paths_fecal_filtered_white[,2:425], method = "bray")
jaccf_path_white = vegdist(paths_fecal_filtered_white[,2:425], method = "jaccard")

paths_oral_filtered_black = paths_oral_filtered %>% filter(Ethnicity == "Black")
paths_oral_filtered_white = paths_oral_filtered %>% filter(Ethnicity == "White")
brayo_path_black = vegdist(paths_oral_filtered_black[,2:348], method = "bray")
jacco_path_black = vegdist(paths_oral_filtered_black[,2:348], method = "jaccard")
brayo_path_white = vegdist(paths_oral_filtered_white[,2:348], method = "bray")
jacco_path_white = vegdist(paths_oral_filtered_white[,2:348], method = "jaccard")
```

### PERMANOVAs - All Samples
Differences in the pathway abundance profiles of the community associated with self-reported Ethnicity is assessed using permutational multivariate analysis of variance.

#### Bray-Curtis
Fecal
```{r permanova_brayf_path}
adonis2(brayf_path ~ Ethnicity, data = paths_fecal_filtered, 
        permutations = 4999)
adonis2(brayf_path ~ `Study period`, data = paths_fecal_filtered, 
        permutations = 4999)
adonis2(brayf_path_black ~ `Study period`, data = paths_fecal_filtered_black, 
        permutations = 4999)
adonis2(brayf_path_white ~ `Study period`, data = paths_fecal_filtered_white, 
        permutations = 4999)
```

Oral
```{r permanova_brayo_path}
adonis2(brayo_path ~ Ethnicity, data = paths_oral_filtered, 
        permutations = 4999)
adonis2(brayo_path ~ `Study period`, data = paths_oral_filtered, 
        permutations = 4999)
adonis2(brayo_path_black ~ `Study period`, data = paths_oral_filtered_black, 
        permutations = 4999)
adonis2(brayo_path_white ~ `Study period`, data = paths_oral_filtered_white, 
        permutations = 4999)
```
NMDS visualization of Bray-Curtis dissimilarity with all samples.
```{r nmds_bray_path, echo=FALSE, error=FALSE}
mds_bray_f_path<-metaMDS(brayf_path, k=2, trymax=499)
mds_bray_f_path_points<-mds_bray_f_path$points
mds_bray_f_path_points2<-merge(x=mds_bray_f_path_points, 
                               y = paths_fecal_filtered, 
                                  by.x = "row.names", by.y = "row.names")
brayfp <- ggplot(mds_bray_f_path_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_f_path_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Bray-Curtis - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.294, r<sup>2</sup> = 1.1% (All)<br><i>p</i> = 0.909, r<sup>2</sup> = 0.7% (Before)<br><i>p</i> = 0.256, r<sup>2</sup> = 2.2% (During)<br><i>p</i> = 0.878, r<sup>2</sup> = 0.3% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.018",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayfp

mds_bray_o_path<-metaMDS(brayo_path, k=2, trymax=499)
mds_bray_o_path_points<-mds_bray_o_path$points
mds_bray_o_path_points2<-merge(x=mds_bray_o_path_points, 
                               y = paths_oral_filtered, 
                                  by.x = "row.names", by.y = "row.names")
brayop <- ggplot(mds_bray_o_path_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_o_path_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Bray-Curtis - Oral")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.129, r<sup>2</sup> = 4.7% (All)<br><i>p</i> = 0.362, r<sup>2</sup> = 6.6% (Before)<br><i>p</i> = 0.128, r<sup>2</sup> = 10.0% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.115",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayop

brayfps <- ggplot(mds_bray_f_path_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_f_path_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Bray-Curtis - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.951, r<sup>2</sup> = 0.2% (Both)<br><i>p</i> = 0.976, r<sup>2</sup> = 0.3% (Black)<br><i>p</i> = 0.886, r<sup>2</sup> = 0.8% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.018",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayfps

brayops <- ggplot(mds_bray_o_path_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","After"="#2171b5")) + 
  scale_fill_manual(values = c("Before"="#238b45","After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_o_path_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Bray-Curtis - Oral")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.873, r<sup>2</sup> = 1.5% (Both)<br><i>p</i> = 0.648, r<sup>2</sup> = 3.5% (Black)<br><i>p</i> = 0.376, r<sup>2</sup> = 8.0% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.115",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayops
```

```{r nmds_save_bray_path}
tiff(file = "figures/nmds_plot_path_bray_e.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayfp + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfp + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayop + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()

tiff(file = "figures/nmds_plot_path_bray_s.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayfps + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfps + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayops + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
#### Jaccard
Fecal
```{r permanova_jaccf_path}
adonis2(jaccf_path ~ Ethnicity, data = paths_fecal_filtered, 
        permutations = 4999)
adonis2(jaccf_path ~ `Study period`, data = paths_fecal_filtered, 
        permutations = 4999)
adonis2(jaccf_path_black ~ `Study period`, data = paths_fecal_filtered_black, 
        permutations = 4999)
adonis2(jaccf_path_white ~ `Study period`, data = paths_fecal_filtered_white, 
        permutations = 4999)
```

Oral
```{r permanova_jacco_path}
adonis2(jacco_path ~ Ethnicity, data = paths_oral_filtered, 
        permutations = 4999)
adonis2(jacco_path ~ `Study period`, data = paths_oral_filtered, 
        permutations = 4999)
adonis2(jacco_path_black ~ `Study period`, data = paths_oral_filtered_black, 
        permutations = 4999)
adonis2(jacco_path_white ~ `Study period`, data = paths_oral_filtered_white, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with all samples.
```{r nmds_jacc_path, echo=FALSE, error=FALSE}
mds_jacc_f_path<-metaMDS(jaccf_path, k=2, trymax=999, distance = "jaccard")
mds_jacc_f_path_points<-mds_jacc_f_path$points
mds_jacc_f_path_points2<-merge(x=mds_jacc_f_path_points, 
                               y = paths_fecal_filtered, 
                                  by.x = "row.names", by.y = "row.names")
jaccfp <- ggplot(mds_jacc_f_path_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_f_path_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Jaccard - Fecal") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.285, r<sup>2</sup> = 1.0% (All)<br><i>p</i> = 0.932, r<sup>2</sup> = 1.0% (Before)<br><i>p</i> = 0.282, r<sup>2</sup> = 2.0% (During)<br><i>p</i> = 0.906, r<sup>2</sup> = 0.4% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.018",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccfp

mds_jacc_o_path<-metaMDS(jacco_path, k=2, trymax=499, distance = "jaccard")
mds_jacc_o_path_points<-mds_jacc_o_path$points
mds_jacc_o_path_points2<-merge(x=mds_jacc_o_path_points, 
                          y = paths_oral_filtered, 
                                  by.x = "row.names", by.y = "row.names")
jaccop <- ggplot(mds_jacc_o_path_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_o_path_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Jaccard - Oral") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.152, r<sup>2</sup> = 4.3% (All)<br><i>p</i> = 0.367, r<sup>2</sup> = 6.5% (Before)<br><i>p</i> = 0.134, r<sup>2</sup> = 9.2% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.113",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccop

jaccfps <- ggplot(mds_jacc_f_path_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_f_path_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Jaccard - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.953, r<sup>2</sup> = 0.3% (Both)<br><i>p</i> = 0.977, r<sup>2</sup> = 0.4% (Black)<br><i>p</i> = 0.901, r<sup>2</sup> = 0.8% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.018",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccfps

jaccops <- ggplot(mds_jacc_o_path_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","After"="#2171b5")) + 
  scale_fill_manual(values = c("Before"="#238b45","After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_o_path_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Jaccard - Oral")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.921, r<sup>2</sup> = 1.5% (Both)<br><i>p</i> = 0.777, r<sup>2</sup> = 3.3% (Black)<br><i>p</i> = 0.344, r<sup>2</sup> = 8.2% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.113",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccops
```

```{r nmds_save_jacc_path}
tiff(file = "figures/nmds_plot_path_jacc_e.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccfp + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfp + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccop + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()

tiff(file = "figures/nmds_plot_path_jacc_s.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccfps + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfps + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccops + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

### PERMANOVAs - Before samples

Filter samples from before the diet intervention and recalculate Bray-Curtis and Jaccard metrics
```{r filter_path_before}
path_fecal_before = paths_fecal_filtered %>% 
  filter(`Study period` == "Before")
path_oral_before = paths_oral_filtered %>% 
  filter(`Study period` == "Before")

brayf_path_b = vegdist(path_fecal_before[,2:425], method = "bray")
jaccf_path_b = vegdist(path_fecal_before[,2:425], method = "jaccard")
brayo_path_b = vegdist(path_oral_before[,2:348], method = "bray")
jacco_path_b = vegdist(path_oral_before[,2:348], method = "jaccard")
```
#### Bray-Curtis - Before
Fecal
```{r permanova_brayf_path_before}
adonis2(brayf_path_b ~ Ethnicity, data = path_fecal_before, 
        permutations = 4999)
```

Oral
```{r permanova_brayo_path}
adonis2(brayo_path_b ~ Ethnicity, data = path_oral_before, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with all samples.
```{r nmds_bray_path_before, echo=FALSE, error=FALSE}
mds_bray_f_pathb<-metaMDS(brayf_path_b, k=2, trymax=499)
mds_bray_f_pathb_points<-mds_bray_f_pathb$points
mds_bray_f_pathb_points2<-merge(x=mds_bray_f_pathb_points, 
                               y = path_fecal_before, 
                                  by.x = "row.names", by.y = "row.names")
brayfpb <- ggplot(mds_bray_f_pathb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity, 
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Bray-Curtis - Fecal - Before")
brayfpb

mds_bray_o_pathb<-metaMDS(brayo_path_b, k=2, trymax=499)
mds_bray_o_pathb_points<-mds_bray_o_pathb$points
mds_bray_o_pathb_points2<-merge(x=mds_bray_o_pathb_points, 
                               y = path_oral_before, 
                                  by.x = "row.names", by.y = "row.names")
brayopb <- ggplot(mds_bray_o_pathb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Bray-Curtis - Oral - Before")
brayopb
```

```{r nmds_save_bray_path_before}
tiff(file = "figures/nmds_plot_path_bray_before.tif", 
     res = 150, width = 18, height = 7, units = "in")
legend1 = get_legend(brayfpb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfpb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayopb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
#### Jaccard
Fecal
```{r permanova_jaccf_path_before}
adonis2(jaccf_path_b ~ Ethnicity, data = path_fecal_before, 
        permutations = 4999)
```

Oral
```{r permanova_jacco_path_before}
adonis2(jacco_path_b ~ Ethnicity, data = path_oral_before, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with all samples.
```{r nmds_jacc_before, echo=FALSE, error=FALSE}
mds_jacc_f_pathb<-metaMDS(jaccf_path_b, k=2, trymax=999, distance = "jaccard")
mds_jacc_f_pathb_points<-mds_jacc_f_pathb$points
mds_jacc_f_pathb_points2<-merge(x=mds_jacc_f_pathb_points, 
                               y = path_fecal_before, 
                                  by.x = "row.names", by.y = "row.names")
jaccfpb <- ggplot(mds_jacc_f_pathb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Jaccard - Fecal - Before")
jaccfpb

mds_jacc_o_pathb<-metaMDS(jacco_path_b, k=2, trymax=499, distance = "jaccard")
mds_jacc_o_pathb_points<-mds_jacc_o_pathb$points
mds_jacc_o_pathb_points2<-merge(x=mds_jacc_o_pathb_points, 
                          y = path_oral_before, 
                                  by.x = "row.names", by.y = "row.names")
jaccopb <- ggplot(mds_jacc_o_pathb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Jaccard - Oral - Before")
jaccopb
```
```{r nmds_save_jacc_path_before}
tiff(file = "figures/nmds_plot_path_jacc_before.tif", 
     res = 150, width = 18, height = 7, units = "in")
legend1 = get_legend(jaccfpb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfpb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccopb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
### PERMANOVAs - During samples

Filter samples from during the diet intervention and recalculate Bray-Curtis and Jaccard metrics
```{r filter_path_during}
path_fecal_during = paths_fecal_filtered %>% 
  filter(`Study period` == "During")

brayf_path_d = vegdist(path_fecal_during[,2:425], method = "bray")
jaccf_path_d = vegdist(path_fecal_during[,2:425], method = "jaccard")
```
#### Bray-Curtis - During
Fecal
```{r permanova_brayf_path_during}
adonis2(brayf_path_d ~ Ethnicity, data = path_fecal_during, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with during samples.
```{r nmds_bray_path_during, echo=FALSE, error=FALSE}
mds_bray_f_pathd<-metaMDS(brayf_path_d, k=2, trymax=499)
mds_bray_f_pathd_points<-mds_bray_f_pathd$points
mds_bray_f_pathd_points2<-merge(x=mds_bray_f_pathd_points, 
                               y = path_fecal_during, 
                                  by.x = "row.names", by.y = "row.names")
brayfpd <- ggplot(mds_bray_f_pathd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity, 
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Bray-Curtis - Fecal - During")
brayfpd
```

```{r nmds_save_bray_path_during}
tiff(file = "figures/nmds_plot_path_bray_during.tif", 
     res = 150, width = 9, height = 7, units = "in")
brayfpd
dev.off()
```
#### Jaccard
Fecal
```{r permanova_jaccf_path_during}
adonis2(jaccf_path_d ~ Ethnicity, data = path_fecal_during, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with during samples.
```{r nmds_jacc_during, echo=FALSE, error=FALSE}
mds_jacc_f_pathd<-metaMDS(jaccf_path_d, k=2, trymax=999, distance = "jaccard")
mds_jacc_f_pathd_points<-mds_jacc_f_pathd$points
mds_jacc_f_pathd_points2<-merge(x=mds_jacc_f_pathd_points, 
                               y = path_fecal_during, 
                                  by.x = "row.names", by.y = "row.names")
jaccfpd <- ggplot(mds_jacc_f_pathd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("COGs - Jaccard - Fecal - During")
jaccfpd
```

```{r nmds_save_jacc_path_during}
tiff(file = "figures/nmds_plot_path_jacc_during.tif", 
     res = 150, width = 9, height = 7, units = "in")
jaccfpd
dev.off()
```
### PERMANOVAs - After samples

Filter samples from after the diet intervention and recalculate Bray-Curtis and Jaccard metrics
```{r filter_path_before}
path_fecal_after = paths_fecal_filtered %>% 
  filter(`Study period` == "After")
path_oral_after = paths_oral_filtered %>% 
  filter(`Study period` == "After")

brayf_path_a = vegdist(path_fecal_after[,2:425], method = "bray")
jaccf_path_a = vegdist(path_fecal_after[,2:425], method = "jaccard")
brayo_path_a = vegdist(path_oral_after[,2:348], method = "bray")
jacco_path_a = vegdist(path_oral_after[,2:348], method = "jaccard")
```
#### Bray-Curtis - After
Fecal
```{r permanova_brayf_path_after}
adonis2(brayf_path_a ~ Ethnicity, data = path_fecal_after, 
        permutations = 4999)
```

Oral
```{r permanova_brayo_path_after}
adonis2(brayo_path_a ~ Ethnicity, data = path_oral_after, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with after samples.
```{r nmds_bray_path_after, echo=FALSE, error=FALSE}
mds_bray_f_patha<-metaMDS(brayf_path_a, k=2, trymax=499)
mds_bray_f_patha_points<-mds_bray_f_patha$points
mds_bray_f_patha_points2<-merge(x=mds_bray_f_patha_points, 
                               y = path_fecal_after, 
                                  by.x = "row.names", by.y = "row.names")
brayfpa <- ggplot(mds_bray_f_patha_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity, 
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Bray-Curtis - Fecal - After")
brayfpa

mds_bray_o_patha<-metaMDS(brayo_path_a, k=2, trymax=499)
mds_bray_o_patha_points<-mds_bray_o_patha$points
mds_bray_o_patha_points2<-merge(x=mds_bray_o_patha_points, 
                               y = path_oral_after, 
                                  by.x = "row.names", by.y = "row.names")
brayopa <- ggplot(mds_bray_o_patha_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Bray-Curtis - Oral - After")
brayopa
```

```{r nmds_save_bray_path_after}
tiff(file = "figures/nmds_plot_path_bray_after.tif", 
     res = 150, width = 18, height = 7, units = "in")
legend1 = get_legend(brayfpa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfpa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayopa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
#### Jaccard - After
Fecal
```{r permanova_jaccf_path_after}
adonis2(jaccf_path_a ~ Ethnicity, data = path_fecal_after, 
        permutations = 4999)
```

Oral
```{r permanova_jacco_path_after}
adonis2(jacco_path_a ~ Ethnicity, data = path_oral_after, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with all samples.
```{r nmds_jacc_after, echo=FALSE, error=FALSE}
mds_jacc_f_patha<-metaMDS(jaccf_path_a, k=2, trymax=999, distance = "jaccard")
mds_jacc_f_patha_points<-mds_jacc_f_patha$points
mds_jacc_f_patha_points2<-merge(x=mds_jacc_f_patha_points, 
                               y = path_fecal_after, 
                                  by.x = "row.names", by.y = "row.names")
jaccfpa <- ggplot(mds_jacc_f_patha_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Jaccard - Fecal - After")
jaccfpa

mds_jacc_o_patha<-metaMDS(jacco_path_a, k=2, trymax=499, distance = "jaccard")
mds_jacc_o_patha_points<-mds_jacc_o_patha$points
mds_jacc_o_patha_points2<-merge(x=mds_jacc_o_patha_points, 
                          y = path_oral_after, 
                                  by.x = "row.names", by.y = "row.names")
jaccopa <- ggplot(mds_jacc_o_patha_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("MetaCyc - Jaccard - Oral - After")
jaccopa
```

```{r nmds_save_jacc_path_after}
tiff(file = "figures/nmds_plot_path_jacc_after.tif", 
     res = 150, width = 18, height = 7, units = "in")
legend1 = get_legend(jaccfpa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfpa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccopa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

## ARGS

Antibiotic resistance genes were identified and quantified with ShortBRED.

### Set up the environment
Load the necessary libraries.
```{r libraries_args, results=FALSE, error=FALSE}
library(tidyverse)
library(vegan)
library(ggplot2)
library(ggtext)
library(cowplot)
library(car)
library(multcomp)
library(ggpubr)
library(fdrtool)
```

### Importing data
Import metadata
```{r import_meta_args, error=FALSE}
metadata_args = read_tsv("tables/year2_sample_mapping_args_noout.tsv") %>% 
  mutate(Ethnicity = case_when(Ethnicity == "African_American" ~ "Black", 
                               Ethnicity == "Caucasian" ~ "White"))
```

Import normalized, arg abundance table with subject 21 removed. Transpose.
```{r import_args, error=FALSE}
arg_table = read_tsv("all_shortbred_abundance_year2.tsv")
args = arg_table %>% gather(file, count, 2:165) %>% 
  spread(Family, count)
```

### Filter samples

Filter out samples that have no identified species or are controls (blanks and mock communities). Separate oral samples from fecal samples.

```{r filter_arg}
args_fecal = args %>% left_join(metadata_args, by = "file") %>% 
  filter(fecal_use == 1)
args_fecal_filtered = args_fecal %>% 
  mutate(total = rowSums(args_fecal[,2:889])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0) 
write_csv(args_fecal_filtered, "arg_fecal_year2.csv")
args_fecal_filtered = read_csv("tables/arg_fecal_year2.csv")

args_oral = args %>% left_join(metadata_args, by = "file") %>% 
  filter(oral_use == 1)
args_oral_filtered = args_oral %>% 
  mutate(total = rowSums(args_oral[,2:889])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0) 
write_csv(args_oral, "arg_oral_year2.csv")
args_oral_filtered = read_csv("tables/arg_oral_year2.csv")
```

### Calculating beta diversity metrics
Vegan (2.5-7) is used to calculate the Bray-Curtis dissimilarity and Jaccard similarity metrics.
```{r beta_metrics_arg, error=FALSE}
brayf_arg = vegdist(args_fecal_filtered[,2:186], method = "bray")
brayo_arg = vegdist(args_oral_filtered[,2:66], method = "bray")
jaccf_arg = vegdist(args_fecal_filtered[,2:186], method = "jaccard",
                    binary = T)
jacco_arg = vegdist(args_oral_filtered[,2:66], method = "jaccard",
                    binary = T)

args_fecal_filtered_black = args_fecal_filtered %>% 
  filter(Ethnicity == "Black")
args_fecal_filtered_white = args_fecal_filtered %>% 
  filter(Ethnicity == "White")
brayf_arg_black = vegdist(args_fecal_filtered_black[,2:186], 
                          method = "bray")
jaccf_arg_black = vegdist(args_fecal_filtered_black[,2:186], 
                          method = "jaccard")
brayf_arg_white = vegdist(args_fecal_filtered_white[,2:186], 
                          method = "bray")
jaccf_arg_white = vegdist(args_fecal_filtered_white[,2:186], 
                          method = "jaccard")

args_oral_filtered_black = args_oral_filtered %>% 
  filter(Ethnicity == "Black")
args_oral_filtered_white = args_oral_filtered %>% 
  filter(Ethnicity == "White")
brayo_arg_black = vegdist(args_oral_filtered_black[,2:66], 
                          method = "bray")
jacco_arg_black = vegdist(args_oral_filtered_black[,2:66], 
                          method = "jaccard")
brayo_arg_white = vegdist(args_oral_filtered_white[,2:66], 
                          method = "bray")
jacco_arg_white = vegdist(args_oral_filtered_white[,2:66], 
                          method = "jaccard")
```

### PERMANOVAS

```{r permanova_fecal_arg}
adonis2(brayf_arg ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = args_fecal_filtered, 
        permutations = 4999)

anova(betadisper(brayf_arg, group = args_fecal_filtered$Ethnicity))

anova(betadisper(brayf_arg, group = args_fecal_filtered$`Study period`))

anova(betadisper(brayf_arg, group = args_fecal_filtered$`Antibiotic use`))

anova(betadisper(brayf_arg, group = args_fecal_filtered$`Contraceptive use`))

anova(betadisper(brayf_arg, group = args_fecal_filtered$subject_id))

adonis2(jaccf_arg ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = args_fecal_filtered, 
        permutations = 4999)

anova(betadisper(jaccf_arg, group = args_fecal_filtered$Ethnicity))

anova(betadisper(jaccf_arg, group = args_fecal_filtered$`Study period`))

anova(betadisper(jaccf_arg, group = args_fecal_filtered$`Antibiotic use`))

anova(betadisper(jaccf_arg, group = args_fecal_filtered$`Contraceptive use`))

anova(betadisper(jaccf_arg, group = args_fecal_filtered$subject_id))
```

```{r permanova_oral_arg}
adonis2(brayo_arg ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = args_oral_filtered, 
        permutations = 4999)

anova(betadisper(brayo_arg, group = args_oral_filtered$Ethnicity))

anova(betadisper(brayo_arg, group = args_oral_filtered$`Study period`))

anova(betadisper(brayo_arg, group = args_oral_filtered$`Antibiotic use`))

anova(betadisper(brayo_arg, group = args_oral_filtered$`Contraceptive use`))

anova(betadisper(brayo_arg, group = args_oral_filtered$subject_id))

adonis2(jacco_arg ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = args_oral_filtered, 
        permutations = 4999)

anova(betadisper(jacco_arg, group = args_oral_filtered$Ethnicity))

anova(betadisper(jacco_arg, group = args_oral_filtered$`Study period`))

anova(betadisper(jacco_arg, group = args_oral_filtered$`Antibiotic use`))

anova(betadisper(jacco_arg, group = args_oral_filtered$`Contraceptive use`))

anova(betadisper(jacco_arg, group = args_oral_filtered$subject_id))
```

NMDS visualization of Bray-Curtis dissimilarity with all samples.
```{r nmds_bray_arg, echo=FALSE, error=FALSE}
mds_bray_f_arg<-metaMDS(brayf_arg, k=2, trymax=499)
mds_bray_f_arg_points<-mds_bray_f_arg$points
mds_bray_f_arg_points2<-merge(x=mds_bray_f_arg_points, 
                               y = args_fecal_filtered, 
                                  by.x = "row.names", by.y = "row.names")
brayfargy2 <- ggplot(mds_bray_f_arg_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_f_arg_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, r<sup>2</sup> = 9.9% (All)<br><i>p</i> = 0.002, r<sup>2</sup> = 11.5% (Before)<br><i>p</i> < 0.001, r<sup>2</sup> = 9.6% (During)<br><i>p</i> = 0.002, r<sup>2</sup> = 13.0% (After)</b><br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.192",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayfargy2

mds_bray_o_arg<-metaMDS(brayo_arg, k=2, trymax=499)
mds_bray_o_arg_points<-mds_bray_o_arg$points
mds_bray_o_arg_points2<-merge(x=mds_bray_o_arg_points, 
                               y = args_oral_filtered, 
                                  by.x = "row.names", by.y = "row.names")
brayoargy2 <- ggplot(mds_bray_o_arg_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_o_arg_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.003, r<sup>2</sup> = 7.6% (All)<br><i>p</i> = 0.034, r<sup>2</sup> = 9.7% (Before)</b><br><i>p</i> = 0.202, r<sup>2</sup> = 6.7% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.124",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayoargy2

brayfargy2s <- ggplot(mds_bray_f_arg_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_f_arg_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.914, r<sup>2</sup> = 1.0% (Both)<br><i>p</i> = 0.972, r<sup>2</sup> = 1.5% (Black)<br><i>p</i> = 0.800, r<sup>2</sup> = 2.1% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.203",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayfargy2s

brayoargy2s <- ggplot(mds_bray_o_arg_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","After"="#2171b5")) + 
  scale_fill_manual(values = c("Before"="#238b45","After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_o_arg_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.976, r<sup>2</sup> = 0.8% (Both)<br><i>p</i> = 0.817, r<sup>2</sup> = 3.9% (Black)<br><i>p</i> = 0.871, r<sup>2</sup> = 2.6% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.182",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayoargy2s
```

