---
title: "Cohort 2 Taxonomy"
author: "Liz Mallott"
date: "1/21/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Overview of analysis

This analysis examines if overall community composition differs based on self-reported ethnicity and what specific taxa are differentially abundant. 
MetaPhlAn (v3.0.7) was used to profile the taxonomic composition of metagenomes and calculate unweighted and weighted UniFrac distances. 

### Setting up the environment

Load the necessary libraries.
```{r libraries, results=FALSE, error=FALSE}
library(tidyverse)
library(vegan)
library(ggplot2)
library(cowplot)
library(car)
library(multcomp)
library(ggpubr)
library(fdrtool)
```

### Importing data

Import metadata
```{r import_meta, error=FALSE}
metadata = read_tsv("year2_sample_mapping_metaphlan.tsv")
```

Import species-level relative abundance table, transpose, and recalculate relative abundance to sum to 1 instead of 100.
```{r import_relab, error=FALSE}
relab = read_tsv("year2_merged_abundance_table_species.txt")
relab1 = relab %>% mutate_if(is.numeric, ~./100)
relab1_t = relab1 %>% gather(file, count, 2:164) %>% 
  spread(body_site, count)
```

Import UniFrac matrices
```{r import_unifrac, error=FALSE}
wuni_fecal = as.dist(
  read.table("year2_merged_abundance_wunifrac_fecal.tsv"))
uuni_fecal = as.dist(
  read.table("year2_merged_abundance_uunifrac_fecal.tsv"))
wuni_oral = as.dist(
  read.table("year2_merged_abundance_wunifrac_oral.tsv"))
uuni_oral = as.dist(
  read.table("year2_merged_abundance_uunifrac_oral.tsv"))

wuni_fecalb = as.dist(
  read.table("year2_merged_abundance_wunifrac_fecalb.tsv"))
uuni_fecalb = as.dist(
  read.table("year2_merged_abundance_uunifrac_fecalb.tsv"))
wuni_oralb = as.dist(
  read.table("year2_merged_abundance_wunifrac_oralb.tsv"))
uuni_oralb = as.dist(
  read.table("year2_merged_abundance_uunifrac_oralb.tsv"))

wuni_fecala = as.dist(
  read.table("year2_merged_abundance_wunifrac_fecala.tsv"))
uuni_fecala = as.dist(
  read.table("year2_merged_abundance_uunifrac_fecala.tsv"))
wuni_orala = as.dist(
  read.table("year2_merged_abundance_wunifrac_orala.tsv"))
uuni_orala = as.dist(
  read.table("year2_merged_abundance_uunifrac_orala.tsv"))

wuni_fecald = as.dist(
  read.table("year2_merged_abundance_wunifrac_fecald.tsv"))
uuni_fecald = as.dist(
  read.table("year2_merged_abundance_uunifrac_fecald.tsv"))
```

### Filter samples

Filter out samples that have no identified species or are controls (blanks and mock communities). Separate oral samples from fecal samples.

```{r filter}
relab_fecal = relab1_t %>% left_join(metadata, by = "file") %>% 
  filter(source == "fecal") 
relab_fecal_filtered = relab_fecal %>% 
  mutate(total = rowSums(relab_fecal[,2:496])) %>% 
  filter(total > 0)
relab_oral = relab1_t %>% left_join(metadata, by = "file") %>% 
  filter(source == "oral")
```

### Calculating beta diversity metrics

UniFrac distances are calculated by a MetaPhlAn helper function. Vegan (2.5-7) is used to calculate the Bray-Curtis dissimilarity and Jaccard similarity metrics.
```{r beta_metrics, error=FALSE}
brayf = vegdist(relab_fecal_filtered[,2:496], method = "bray")
jaccf = vegdist(relab_fecal_filtered[,2:496], method = "jaccard")
brayo = vegdist(relab_oral[,2:496], method = "bray")
jacco = vegdist(relab_oral[,2:496], method = "jaccard")
```

### PERMANOVAs - All Samples
Differences in the taxonomic structure of the community associated with self-reported ethnicity is assessed using permutational multivariate analysis of variance.

#### Bray-Curtis
Fecal
```{r permanova_brayf}
adonis2(brayf ~ ethnicity, data = relab_fecal_filtered, 
        permutations = 4999)
```

Oral
```{r permanova_brayo}
adonis2(brayo ~ ethnicity, data = relab_oral, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with all samples.
```{r nmds_bray, echo=FALSE, error=FALSE}
mds_bray_f<-metaMDS(brayf, k=2, trymax=500)
mds_bray_f_points<-mds_bray_f$points
mds_bray_f_points2<-merge(x=mds_bray_f_points, y = relab_fecal_filtered, 
                                  by.x = "row.names", by.y = "row.names")
brayf <- ggplot(mds_bray_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity, 
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Bray-Curtis - Fecal")
brayf

mds_bray_o<-metaMDS(brayo, k=2, trymax=500)
mds_bray_o_points<-mds_bray_o$points
mds_bray_o_points2<-merge(x=mds_bray_o_points, y = relab_oral, 
                                  by.x = "row.names", by.y = "row.names")
brayo <- ggplot(mds_bray_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Bray-Curtis - Oral")
brayo
```

```{r nmds_save_bray}
tiff(file = "figures/nmds_plot_taxa_bray.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayf + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayf + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayo + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Jaccard
Fecal
```{r permanova_jaccf}
adonis2(jaccf ~ ethnicity, data = relab_fecal_filtered, 
        permutations = 4999)
```

Oral
```{r permanova_jacco}
adonis2(jacco ~ ethnicity, data = relab_oral, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with all samples.
```{r nmds_jacc, echo=FALSE, error=FALSE}
mds_jacc_f<-metaMDS(jaccf, k=2, trymax=500, distance = "jaccard")
mds_jacc_f_points<-mds_jacc_f$points
mds_jacc_f_points2<-merge(x=mds_jacc_f_points, y = relab_fecal_filtered, 
                                  by.x = "row.names", by.y = "row.names")
jaccf <- ggplot(mds_jacc_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Fecal")
jaccf

mds_jacc_o<-metaMDS(jacco, k=2, trymax=500, distance = "jaccard")
mds_jacc_o_points<-mds_jacc_o$points
mds_jacc_o_points2<-merge(x=mds_jacc_o_points, y = relab_oral, 
                                  by.x = "row.names", by.y = "row.names")
jacco <- ggplot(mds_jacc_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Oral")
jacco
```

```{r nmds_save_jacc}
tiff(file = "figures/nmds_plot_taxa_jacc.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccf + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccf + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jacco + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Weighted UniFrac
Fecal
```{r permanova_wunif}
adonis2(wuni_fecal ~ ethnicity, data = relab_fecal_filtered, 
        permutations = 4999)
```

Oral
```{r permanova_wunio}
adonis2(wuni_oral ~ ethnicity, data = relab_oral, 
        permutations = 4999)
```

NMDS visualization of weighted Unifrac distances with all samples.
```{r nmds_wuni, echo=FALSE, error=FALSE}
mds_wuni_f<-metaMDS(wuni_fecal, k=2, trymax=500)
mds_wuni_f_points<-mds_wuni_f$points
mds_wuni_f_points2<-merge(x=mds_wuni_f_points, y = relab_fecal_filtered, 
                                  by.x = "row.names", by.y = "file")
wunif <- ggplot(mds_wuni_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Weighted Unifrac - Fecal")
wunif

mds_wuni_o<-metaMDS(wuni_oral, k=2, trymax=500)
mds_wuni_o_points<-mds_wuni_o$points
mds_wuni_o_points2<-merge(x=mds_wuni_o_points, y = relab_oral, 
                                  by.x = "row.names", by.y = "file")
wunio <- ggplot(mds_wuni_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Weighted UniFrac - Oral")
wunio
```

```{r nmds_save_wuni}
tiff(file = "figures/nmds_plot_taxa_wuni.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(wunif + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(wunif + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(wunio + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Unweighted UniFrac
Fecal
```{r permanova_uunif}
adonis2(uuni_fecal ~ ethnicity, data = relab_fecal_filtered, 
        permutations = 4999)
```

Oral
```{r permanova_uunio}
adonis2(uuni_oral ~ ethnicity, data = relab_oral, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with all samples.
```{r nmds_uuni, echo=FALSE, error=FALSE}
mds_uuni_f<-metaMDS(uuni_fecal, k=2, trymax=500)
mds_uuni_f_points<-mds_uuni_f$points
mds_uuni_f_points2<-merge(x=mds_uuni_f_points, y = relab_fecal_filtered, 
                                  by.x = "row.names", by.y = "file")
uunif <- ggplot(mds_uuni_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Unweighted Unifrac - Fecal")
uunif

mds_uuni_o<-metaMDS(uuni_oral, k=2, trymax=500)
mds_uuni_o_points<-mds_uuni_o$points
mds_uuni_o_points2<-merge(x=mds_uuni_o_points, y = relab_oral, 
                                  by.x = "row.names", by.y = "file")
uunio <- ggplot(mds_uuni_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Unweighted UniFrac - Oral")
uunio
```

```{r nmds_save_uuni}
tiff(file = "figures/nmds_plot_taxa_uuni.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(uunif + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(uunif + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(uunio + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

### PERMANOVAs - Before samples

#### Filtering samples
Filter samples from before the diet intervention and recalculate Bray-Curtis and Jaccard metrics

```{r filter_before}
relab_fecal_filtered_before = relab_fecal_filtered %>% 
  filter(Study_period == "Before")
relab_oral_before = relab_oral %>% 
  filter(Study_period == "Before")

brayfb = vegdist(relab_fecal_filtered_before[,2:496], method = "bray")
jaccfb = vegdist(relab_fecal_filtered_before[,2:496], method = "jaccard")
brayob = vegdist(relab_oral_before[,2:496], method = "bray")
jaccob = vegdist(relab_oral_before[,2:496], method = "jaccard")
```


#### Bray-Curtis
Fecal
```{r permanova_brayfb}
adonis2(brayfb ~ ethnicity, data = relab_fecal_filtered_before, 
        permutations = 4999)
```

Oral
```{r permanova_brayob}
adonis2(brayob ~ ethnicity, data = relab_oral_before, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with before samples.
```{r nmds_bray_before, echo=FALSE, error=FALSE}
mds_bray_fb<-metaMDS(brayfb, k=2, trymax=500)
mds_bray_fb_points<-mds_bray_fb$points
mds_bray_fb_points2<-merge(x=mds_bray_fb_points, 
                           y = relab_fecal_filtered_before, 
                                  by.x = "row.names", by.y = "row.names")
brayfb <- ggplot(mds_bray_fb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity, 
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Bray-Curtis - Fecal")
brayfb

mds_bray_ob<-metaMDS(brayob, k=2, trymax=500)
mds_bray_ob_points<-mds_bray_ob$points
mds_bray_ob_points2<-merge(x=mds_bray_ob_points, y = relab_oral_before, 
                                  by.x = "row.names", by.y = "row.names")
brayob <- ggplot(mds_bray_ob_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Bray-Curtis - Oral")
brayob
```

```{r nmds_save_bray_before}
tiff(file = "figures/nmds_plot_taxa_bray_before.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayfb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayob + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Jaccard
Fecal
```{r permanova_jaccfb}
adonis2(jaccfb ~ ethnicity, data = relab_fecal_filtered_before, 
        permutations = 4999)
```

Oral
```{r permanova_jaccob}
adonis2(jaccob ~ ethnicity, data = relab_oral_before, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with before samples.
```{r nmds_jacc_before, echo=FALSE, error=FALSE}
mds_jacc_fb<-metaMDS(jaccfb, k=2, trymax=500, distance = "jaccard")
mds_jacc_fb_points<-mds_jacc_fb$points
mds_jacc_fb_points2<-merge(x=mds_jacc_fb_points, 
                           y = relab_fecal_filtered_before, 
                                  by.x = "row.names", by.y = "row.names")
jaccfb <- ggplot(mds_jacc_fb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Fecal")
jaccfb

mds_jacc_ob<-metaMDS(jaccob, k=2, trymax=500, distance = "jaccard")
mds_jacc_ob_points<-mds_jacc_ob$points
mds_jacc_ob_points2<-merge(x=mds_jacc_ob_points, y = relab_oral_before, 
                                  by.x = "row.names", by.y = "row.names")
jaccob <- ggplot(mds_jacc_ob_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Oral")
jaccob
```

```{r nmds_save_jacc_before}
tiff(file = "figures/nmds_plot_taxa_jacc_before.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccfb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccob + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Weighted UniFrac
Fecal
```{r permanova_wunif_before}
adonis2(wuni_fecalb ~ ethnicity, data = relab_fecal_filtered_before, 
        permutations = 4999)
```

Oral
```{r permanova_wunio_before}
adonis2(wuni_oralb ~ ethnicity, data = relab_oral_before, 
        permutations = 4999)
```

NMDS visualization of weighted Unifrac distances with before samples.
```{r nmds_wuni_before, echo=FALSE, error=FALSE}
mds_wuni_fb<-metaMDS(wuni_fecalb, k=2, trymax=500)
mds_wuni_fb_points<-mds_wuni_fb$points
mds_wuni_fb_points2<-merge(x=mds_wuni_fb_points, 
                           y = relab_fecal_filtered_before, 
                                  by.x = "row.names", by.y = "file")
wunifb <- ggplot(mds_wuni_fb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Weighted Unifrac - Fecal")
wunifb

mds_wuni_ob<-metaMDS(wuni_oralb, k=2, trymax=500)
mds_wuni_ob_points<-mds_wuni_ob$points
mds_wuni_ob_points2<-merge(x=mds_wuni_ob_points, y = relab_oral_before, 
                                  by.x = "row.names", by.y = "file")
wuniob <- ggplot(mds_wuni_ob_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Weighted UniFrac - Oral")
wuniob
```

```{r nmds_save_wuni_before}
tiff(file = "figures/nmds_plot_taxa_wuni_before.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(wunifb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(wunifb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(wuniob + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Unweighted UniFrac
Fecal
```{r permanova_uunif_before}
adonis2(uuni_fecalb ~ ethnicity, data = relab_fecal_filtered_before, 
        permutations = 4999)
```

Oral
```{r permanova_uunio_before}
adonis2(uuni_oralb ~ ethnicity, data = relab_oral_before, 
        permutations = 4999)
```

NMDS visualization of unweighted UniFrac distances with before samples.
```{r nmds_uuni_before, echo=FALSE, error=FALSE}
mds_uuni_fb<-metaMDS(uuni_fecalb, k=2, trymax=500)
mds_uuni_fb_points<-mds_uuni_fb$points
mds_uuni_fb_points2<-merge(x=mds_uuni_fb_points, 
                           y = relab_fecal_filtered_before, 
                                  by.x = "row.names", by.y = "file")
uunifb <- ggplot(mds_uuni_fb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Unweighted Unifrac - Fecal")
uunifb

mds_uuni_ob<-metaMDS(uuni_oralb, k=2, trymax=500)
mds_uuni_ob_points<-mds_uuni_ob$points
mds_uuni_ob_points2<-merge(x=mds_uuni_ob_points, y = relab_oral_before, 
                                  by.x = "row.names", by.y = "file")
uuniob <- ggplot(mds_uuni_ob_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Unweighted UniFrac - Oral")
uuniob
```

```{r nmds_save_uuni_before}
tiff(file = "figures/nmds_plot_taxa_uuni_before.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(uunifb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(uunifb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(uuniob + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
### PERMANOVAs - During samples

#### Filter samples
Filter samples from during the diet intervention and recalculate Bray-Curtis and Jaccard metrics
```{r filter_during}
relab_fecal_filtered_during = relab_fecal_filtered %>% 
  filter(Study_period == "During")

brayfd = vegdist(relab_fecal_filtered_during[,2:496], method = "bray")
jaccfd = vegdist(relab_fecal_filtered_during[,2:496], method = "jaccard")
```

#### Bray-Curtis
Fecal
```{r permanova_brayfd}
adonis2(brayfd ~ ethnicity, data = relab_fecal_filtered_during, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with during samples.
```{r nmds_bray_during, echo=FALSE, error=FALSE}
mds_bray_fd<-metaMDS(brayfd, k=2, trymax=500)
mds_bray_fd_points<-mds_bray_fd$points
mds_bray_fd_points2<-merge(x=mds_bray_fd_points, 
                           y = relab_fecal_filtered_during, 
                                  by.x = "row.names", by.y = "row.names")
brayfd <- ggplot(mds_bray_fd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity, 
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Bray-Curtis - Fecal")
brayfd
```

```{r nmds_save_bray_during}
tiff(file = "figures/nmds_plot_taxa_bray_during.tif", res = 150, width = 11, 
     height = 7, units = "in")
brayfd
dev.off()
```

#### Jaccard
Fecal
```{r permanova_jaccfd}
adonis2(jaccfd ~ ethnicity, data = relab_fecal_filtered_during, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with during samples.
```{r nmds_jacc_during, echo=FALSE, error=FALSE}
mds_jacc_fd<-metaMDS(jaccfd, k=2, trymax=500, distance = "jaccard")
mds_jacc_fd_points<-mds_jacc_fd$points
mds_jacc_fd_points2<-merge(x=mds_jacc_fd_points, 
                           y = relab_fecal_filtered_during, 
                                  by.x = "row.names", by.y = "row.names")
jaccfd <- ggplot(mds_jacc_fd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Fecal")
jaccfd
```

```{r nmds_save_jacc_during}
tiff(file = "figures/nmds_plot_taxa_jacc_during.tif", res = 150, width = 11, 
     height = 7, units = "in")
jaccfd
dev.off()
```

#### Weighted UniFrac
Fecal
```{r permanova_wunif_during}
adonis2(wuni_fecald ~ ethnicity, data = relab_fecal_filtered_during, 
        permutations = 4999)
```

NMDS visualization of weighted Unifrac distances with during samples.
```{r nmds_wuni_during, echo=FALSE, error=FALSE}
mds_wuni_fd<-metaMDS(wuni_fecald, k=2, trymax=500)
mds_wuni_fd_points<-mds_wuni_fd$points
mds_wuni_fd_points2<-merge(x=mds_wuni_fd_points, 
                           y = relab_fecal_filtered_during, 
                                  by.x = "row.names", by.y = "file")
wunifd <- ggplot(mds_wuni_fd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Weighted Unifrac - Fecal")
wunifd
```

```{r nmds_save_wuni_during}
tiff(file = "figures/nmds_plot_taxa_wuni_during.tif", res = 150, width = 11, 
     height = 7, units = "in")
wunif
dev.off()
```

#### Unweighted UniFrac
Fecal
```{r permanova_uunif_during}
adonis2(uuni_fecald ~ ethnicity, data = relab_fecal_filtered_during, 
        permutations = 4999)
```

NMDS visualization of unweighted Unifrac distance with during samples.
```{r nmds_uuni_during, echo=FALSE, error=FALSE}
mds_uuni_fd<-metaMDS(uuni_fecald, k=2, trymax=500)
mds_uuni_fd_points<-mds_uuni_fd$points
mds_uuni_fd_points2<-merge(x=mds_uuni_fd_points, 
                           y = relab_fecal_filtered_during, 
                                  by.x = "row.names", by.y = "file")
uunifd <- ggplot(mds_uuni_fd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Unweighted Unifrac - Fecal")
uunifd
```

```{r nmds_save_uuni_during}
tiff(file = "figures/nmds_plot_taxa_uuni_during.tif", res = 150, width = 18, 
     height = 7, units = "in")
uunifd
dev.off()
```
### PERMANOVAs - After samples

#### Filter samples

Filter after samples and recalculate Bray-Curtis and Jaccard metrics
```{r filter_after}
relab_fecal_filtered_after = relab_fecal_filtered %>% 
  filter(Study_period == "After")
relab_oral_after = relab_oral %>% 
  filter(Study_period == "After")

brayfa = vegdist(relab_fecal_filtered_after[,2:496], method = "bray")
jaccfa = vegdist(relab_fecal_filtered_after[,2:496], method = "jaccard")
brayoa = vegdist(relab_oral_after[,2:496], method = "bray")
jaccoa = vegdist(relab_oral_after[,2:496], method = "jaccard")
```
#### Bray-Curtis
Fecal
```{r permanova_brayfa}
adonis2(brayfa ~ ethnicity, data = relab_fecal_filtered_after, 
        permutations = 4999)
```

Oral
```{r permanova_brayoa}
adonis2(brayoa ~ ethnicity, data = relab_oral_after, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with after samples.
```{r nmds_bray_after, echo=FALSE, error=FALSE}
mds_bray_fa<-metaMDS(brayfa, k=2, trymax=500)
mds_bray_fa_points<-mds_bray_fa$points
mds_bray_fa_points2<-merge(x=mds_bray_fa_points, 
                           y = relab_fecal_filtered_after, 
                                  by.x = "row.names", by.y = "row.names")
brayfa <- ggplot(mds_bray_fa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity, 
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Bray-Curtis - Fecal")
brayfa

mds_bray_oa<-metaMDS(brayoa, k=2, trymax=500)
mds_bray_oa_points<-mds_bray_oa$points
mds_bray_oa_points2<-merge(x=mds_bray_oa_points, y = relab_oral_after, 
                                  by.x = "row.names", by.y = "row.names")
brayoa <- ggplot(mds_bray_oa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Bray-Curtis - Oral")
brayoa
```

```{r nmds_save_bray_after}
tiff(file = "figures/nmds_plot_taxa_bray_after.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayfa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayoa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Jaccard
Fecal
```{r permanova_jaccfa}
adonis2(jaccfa ~ ethnicity, data = relab_fecal_filtered_after, 
        permutations = 4999)
```

Oral
```{r permanova_jaccoa}
adonis2(jaccoa ~ ethnicity, data = relab_oral_after, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with after samples.
```{r nmds_jacc_after, echo=FALSE, error=FALSE}
mds_jacc_fa<-metaMDS(jaccfa, k=2, trymax=500, distance = "jaccard")
mds_jacc_fa_points<-mds_jacc_fa$points
mds_jacc_fa_points2<-merge(x=mds_jacc_fa_points, 
                           y = relab_fecal_filtered_after, 
                                  by.x = "row.names", by.y = "row.names")
jaccfa <- ggplot(mds_jacc_fa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Fecal")
jaccfa

mds_jacc_oa<-metaMDS(jaccoa, k=2, trymax=500, distance = "jaccard")
mds_jacc_oa_points<-mds_jacc_oa$points
mds_jacc_oa_points2<-merge(x=mds_jacc_oa_points, y = relab_oral_after, 
                                  by.x = "row.names", by.y = "row.names")
jaccoa <- ggplot(mds_jacc_oa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Oral")
jaccoa
```

```{r nmds_save_jacc_after}
tiff(file = "figures/nmds_plot_taxa_jacc_after.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccfa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccoa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Weighted UniFrac
Fecal
```{r permanova_wunif_after}
adonis2(wuni_fecala ~ ethnicity, data = relab_fecal_filtered_after, 
        permutations = 4999)
```

Oral
```{r permanova_wunio_after}
adonis2(wuni_orala ~ ethnicity, data = relab_oral_after, 
        permutations = 4999)
```

NMDS visualization of weighted Unifrac distances with after samples.
```{r nmds_wuni_after, echo=FALSE, error=FALSE}
mds_wuni_fa<-metaMDS(wuni_fecala, k=2, trymax=500)
mds_wuni_fa_points<-mds_wuni_fa$points
mds_wuni_fa_points2<-merge(x=mds_wuni_fa_points, 
                           y = relab_fecal_filtered_after, 
                                  by.x = "row.names", by.y = "file")
wunifa <- ggplot(mds_wuni_fa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Weighted Unifrac - Fecal")
wunifa

mds_wuni_oa<-metaMDS(wuni_orala, k=2, trymax=500)
mds_wuni_oa_points<-mds_wuni_oa$points
mds_wuni_oa_points2<-merge(x=mds_wuni_oa_points, y = relab_oral_after, 
                                  by.x = "row.names", by.y = "file")
wunioa <- ggplot(mds_wuni_oa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Weighted UniFrac - Oral")
wunioa
```

```{r nmds_save_wuni_after}
tiff(file = "figures/nmds_plot_taxa_wuni_after.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(wunifa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(wunifa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(wunioa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Unweighted UniFrac
Fecal
```{r permanova_uunif_after}
adonis2(uuni_fecala ~ ethnicity, data = relab_fecal_filtered_after, 
        permutations = 4999)
```

Oral
```{r permanova_uunio_after}
adonis2(uuni_orala ~ ethnicity, data = relab_oral_after, 
        permutations = 4999)
```

NMDS visualization of unweighted UniFrac distances with after samples.
```{r nmds_uuni_after, echo=FALSE, error=FALSE}
mds_uuni_fa<-metaMDS(uuni_fecala, k=2, trymax=500)
mds_uuni_fa_points<-mds_uuni_fa$points
mds_uuni_fa_points2<-merge(x=mds_uuni_fa_points, 
                           y = relab_fecal_filtered_after, 
                                  by.x = "row.names", by.y = "file")
uunifa <- ggplot(mds_uuni_fa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Unweighted Unifrac - Fecal")
uunifa

mds_uuni_oa<-metaMDS(uuni_orala, k=2, trymax=500)
mds_uuni_oa_points<-mds_uuni_oa$points
mds_uuni_oa_points2<-merge(x=mds_uuni_oa_points, y = relab_oral_after, 
                                  by.x = "row.names", by.y = "file")
uunioa <- ggplot(mds_uuni_oa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = ethnicity,
                                        shape = Study_period)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Caucasian"="#87ceeb","African_American"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = ethnicity, 
                   linetype = ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Unweighted UniFrac - Oral")
uunioa
```

```{r nmds_save_uuni_after}
tiff(file = "figures/nmds_plot_taxa_uuni_after.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(uunifa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(uunifa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(uunioa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

### Alpha Diversity

Alpha diversity was calculated using species-level metaphlan results in vegan (2.5-7). Shannon diversity and Shannon-Weaver evenness (Pielou's evenness) were calculated.

```{r alpha_d}
shannond_fecal = diversity(relab_fecal_filtered[,2:496], index = "shannon")
shannond_oral = diversity(relab_oral[,2:496], index = "shannon")

shannone_fecal = shannond_fecal/log(specnumber(relab_fecal_filtered[,2:496]))
shannone_oral = shannond_oral/log(specnumber(relab_oral[,2:496]))

fecal_diversity = cbind(relab_fecal_filtered[,1], relab_fecal_filtered[,497:520], shannond_fecal, shannone_fecal)
fecal_diversity$Study_period = as.factor(fecal_diversity$Study_period)
oral_diversity = cbind(relab_oral[,1], relab_oral[,497:519], shannond_oral, shannone_oral)
oral_diversity$Study_period = as.factor(oral_diversity$Study_period)
```

A linear model was used to examine differences between self-reported ethnicities and study period (before/during/after) in diversity. 
```{r alpha_models_fecal}
div_fecal = lm(shannond_fecal ~ ethnicity*Study_period, 
                         data = fecal_diversity)
summary(div_fecal)
Anova(div_fecal)
summary(glht(div_fecal,linfct=mcp(Study_period="Tukey")))

even_fecal = lm(shannone_fecal ~ ethnicity*Study_period, 
                          data = fecal_diversity)
summary(even_fecal)
Anova(even_fecal)
summary(glht(even_fecal,linfct=mcp(Study_period="Tukey")))
```

```{r alpha_models_oral}
div_oral = lm(shannond_oral ~ ethnicity*Study_period, data = oral_diversity)
summary(div_oral)
Anova(div_oral)

even_oral = lm(shannone_oral ~ ethnicity*Study_period, data = oral_diversity)
summary(even_oral)
Anova(even_oral)
```
Shannon diversity and evenness did not vary across self-reported ethnicities or study periods in either fecal or oral samples.

```{r alpha_graphs, echo=FALSE}
fecal_diversity_before = fecal_diversity %>% 
  filter(Study_period == "Before")
oral_diversity_before = oral_diversity %>% 
  filter(Study_period == "Before")
fecal_diversity_during = fecal_diversity %>% 
  filter(Study_period == "During")
fecal_diversity_after = fecal_diversity %>% 
  filter(Study_period == "After")
oral_diversity_after = oral_diversity %>% 
  filter(Study_period == "After")

fecal_div_all = ggboxplot(fecal_diversity, x = "ethnicity", 
                  y = "shannond_fecal", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
fecal_div_all = ggpar(fecal_div_all, legend = "right", 
                      title = "Fecal samples - All") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

oral_div_all = ggboxplot(oral_diversity, x = "ethnicity", 
                  y = "shannond_oral", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
oral_div_all = ggpar(oral_div_all, legend = "right", 
                      title = "Oral samples - All") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

fecal_div_before = ggboxplot(fecal_diversity_before, x = "ethnicity", 
                  y = "shannond_fecal", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
fecal_div_before = ggpar(fecal_div_before, legend = "right", 
                      title = "Fecal samples - Before") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

oral_div_before = ggboxplot(oral_diversity_before, x = "ethnicity", 
                  y = "shannond_oral", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
oral_div_before = ggpar(oral_div_before, legend = "right", 
                      title = "Oral samples - Before") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

fecal_div_during = ggboxplot(fecal_diversity_during, x = "ethnicity", 
                  y = "shannond_fecal", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
fecal_div_during = ggpar(fecal_div_during, legend = "right", 
                      title = "Fecal samples - During") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

fecal_div_after = ggboxplot(fecal_diversity_after, x = "ethnicity", 
                  y = "shannond_fecal", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
fecal_div_after = ggpar(fecal_div_after, legend = "right", 
                      title = "Fecal samples - After") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

oral_div_after = ggboxplot(oral_diversity_after, x = "ethnicity", 
                  y = "shannond_oral", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Diversity") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
oral_div_after = ggpar(oral_div_after, legend = "right", 
                      title = "Oral samples - After") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

fecal_even_all = ggboxplot(fecal_diversity, x = "ethnicity", 
                  y = "shannone_fecal", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Evenness") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
fecal_even_all = ggpar(fecal_even_all, legend = "right", 
                      title = "Fecal samples - All") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

oral_even_all = ggboxplot(oral_diversity, x = "ethnicity", 
                  y = "shannone_oral", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Evenness") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
oral_even_all = ggpar(oral_even_all, legend = "right", 
                      title = "Oral samples - All") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

fecal_even_before = ggboxplot(fecal_diversity_before, x = "ethnicity", 
                  y = "shannone_fecal", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Evenness") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
fecal_even_before = ggpar(fecal_even_before, legend = "right", 
                      title = "Fecal samples - Before") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

oral_even_before = ggboxplot(oral_diversity_before, x = "ethnicity", 
                  y = "shannone_oral", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Evenness") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
oral_even_before = ggpar(oral_even_before, legend = "right", 
                      title = "Oral samples - Before") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

fecal_even_during = ggboxplot(fecal_diversity_during, x = "ethnicity", 
                  y = "shannone_fecal", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Evenness") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
fecal_even_during = ggpar(fecal_even_during, legend = "right", 
                      title = "Fecal samples - During") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

fecal_even_after = ggboxplot(fecal_diversity_after, x = "ethnicity", 
                  y = "shannone_fecal", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Evenness") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
fecal_even_after = ggpar(fecal_even_after, legend = "right", 
                      title = "Fecal samples - After") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

oral_even_after = ggboxplot(oral_diversity_after, x = "ethnicity", 
                  y = "shannone_oral", color = "ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Shannon Evenness") + 
  scale_color_manual(values = c("Caucasian" = "#87ceeb", 
                                "African_American" = "#f5c4ca"))
oral_even_after = ggpar(oral_even_after, legend = "right", 
                      title = "Oral samples - After") + 
  rremove("xlab") + rremove("x.text") + 
  rremove("legend.title")

tiff(file="figures/fecal_alpha_diversity.tif", res=150, width=15, height=6, units="in")
ggarrange(fecal_div_all, fecal_div_before, fecal_div_during, fecal_div_after,
          fecal_even_all, fecal_even_before, fecal_even_during, 
          fecal_even_after, ncol = 4, nrow=2, 
          common.legend = TRUE, legend = "right", align = "hv", 
          labels = c(""))
dev.off()

tiff(file="figures/oral_alpha_diversity.tif", res=150, width=12, height=6, units="in")
ggarrange(oral_div_all, oral_div_before, oral_div_after,
          oral_even_all, oral_even_before, oral_even_after, 
          ncol = 3, nrow=2, 
          common.legend = TRUE, legend = "right", align = "hv", 
          labels = c(""))
dev.off()
```

### Taxonomic abundance

Generalized linear mixed effects models using a negative binomial distribution were used to assess if self-reported ethnicity or study period is associated with the abundance of individual species.

```{r taxa_models_fecal}
ethnicity_estimatematrix = mat.or.vec(495,2)
ethnicity_pvaluematrix = mat.or.vec(495,2)
studyp_estimatematrix = mat.or.vec(495,2)
studyp_pvaluematrix = mat.or.vec(495,2)

relab_fecal_filtered = as.data.frame(relab_fecal_filtered)

for(i in 2:496) {
  variable = relab_fecal_filtered[,i]
  b = try(lm(variable ~ ethnicity + Study_period, 
                  data = relab_fecal_filtered))
  anova = anova(b)
  ethnicity_estimatematrix[i-1,2] = anova[1,4]
  ethnicity_pvaluematrix[i-1,2] = anova[1,5]
  studyp_estimatematrix[i-1,2] = anova[2,4]
  studyp_pvaluematrix[i-1,2] = anova[2,5]
  ethnicity_estimatematrix[i-1,1] = names(relab_fecal_filtered)[i]
  ethnicity_pvaluematrix[i-1,1] = names(relab_fecal_filtered)[i]
  studyp_estimatematrix[i-1,1] = names(relab_fecal_filtered)[i]
  studyp_pvaluematrix[i-1,1] = names(relab_fecal_filtered)[i]
}


taxa_ethnicity = bind_cols(as.data.frame(ethnicity_estimatematrix[,1:2]), 
                     as.data.frame(ethnicity_pvaluematrix[,2]))
write.csv(taxa_ethnicity, "tables/taxa_fecal_ethnicity.csv")
taxa_ethnicity_nona = filter(taxa_ethnicity, taxa_ethnicity[,3] != "NaN") 
taxa_ethnicity_nona[,3] = as.numeric(as.character(taxa_ethnicity_nona[,3]))
taxa_ethnicity_corrected = bind_cols(taxa_ethnicity_nona, 
                               as.data.frame(fdrtool(taxa_ethnicity_nona[,3], 
                                                     statistic = "pvalue", 
                                                     plot = F)))
write.csv(taxa_ethnicity_corrected, "tables/taxa_fecal_ethnicity_corrected.csv")

taxa_studyp = bind_cols(as.data.frame(studyp_estimatematrix[,1:2]), 
                      as.data.frame(studyp_pvaluematrix[,2]))
write.csv(taxa_studyp, "tables/taxa_fecal_studyp.csv")
taxa_studyp_nona = filter(taxa_studyp, taxa_studyp[,3] != "NaN") 
taxa_studyp_nona[,3] = as.numeric(as.character(taxa_studyp_nona[,3]))
taxa_studyp_corrected = bind_cols(taxa_studyp_nona, 
                                as.data.frame(fdrtool(taxa_studyp_nona[,3], 
                                                      statistic = "pvalue", 
                                                      plot = F)))
write.csv(taxa_studyp_corrected, "tables/taxa_fecal_studyp_corrected.csv")
```

```{r taxa_models_oral}
ethnicity_estimatematrix = mat.or.vec(495,2)
ethnicity_pvaluematrix = mat.or.vec(495,2)
studyp_estimatematrix = mat.or.vec(495,2)
studyp_pvaluematrix = mat.or.vec(495,2)

relab_oral = as.data.frame(relab_oral)

for(i in 2:496) {
  variable = relab_oral[,i]
  b = try(lm(variable ~ ethnicity + Study_period, 
                  data = relab_oral))
  anova = anova(b)
  ethnicity_estimatematrix[i-1,2] = anova[1,4]
  ethnicity_pvaluematrix[i-1,2] = anova[1,5]
  studyp_estimatematrix[i-1,2] = anova[2,4]
  studyp_pvaluematrix[i-1,2] = anova[2,5]
  ethnicity_estimatematrix[i-1,1] = names(relab_oral)[i]
  ethnicity_pvaluematrix[i-1,1] = names(relab_oral)[i]
  studyp_estimatematrix[i-1,1] = names(relab_oral)[i]
  studyp_pvaluematrix[i-1,1] = names(relab_oral)[i]
}


taxa_ethnicity = bind_cols(as.data.frame(ethnicity_estimatematrix[,1:2]), 
                     as.data.frame(ethnicity_pvaluematrix[,2]))
write.csv(taxa_ethnicity, "tables/taxa_oral_ethnicity.csv")
taxa_ethnicity_nona = filter(taxa_ethnicity, taxa_ethnicity[,3] != "NaN") 
taxa_ethnicity_nona[,3] = as.numeric(as.character(taxa_ethnicity_nona[,3]))
taxa_ethnicity_corrected = bind_cols(taxa_ethnicity_nona, 
                               as.data.frame(fdrtool(taxa_ethnicity_nona[,3], 
                                                     statistic = "pvalue", 
                                                     plot = F)))
write.csv(taxa_ethnicity_corrected, "tables/taxa_oral_ethnicity_corrected.csv")

taxa_studyp = bind_cols(as.data.frame(studyp_estimatematrix[,1:2]), 
                      as.data.frame(studyp_pvaluematrix[,2]))
write.csv(taxa_studyp, "tables/taxa_oral_studyp.csv")
taxa_studyp_nona = filter(taxa_studyp, taxa_studyp[,3] != "NaN") 
taxa_studyp_nona[,3] = as.numeric(as.character(taxa_studyp_nona[,3]))
taxa_studyp_corrected = bind_cols(taxa_studyp_nona, 
                                as.data.frame(fdrtool(taxa_studyp_nona[,3], 
                                                      statistic = "pvalue", 
                                                      plot = F)))
write.csv(taxa_studyp_corrected, "tables/taxa_oral_studyp_corrected.csv")
```