---
title: "Cohort 2 Taxonomy"
author: "Liz Mallott"
date: "1/21/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Overview of analysis

This analysis examines if overall community composition differs based on self-reported Ethnicity and what specific taxa are differentially abundant. 
MetaPhlAn (v3.0.7) was used to profile the taxonomic composition of metagenomes and calculate unweighted and weighted UniFrac distances. 

### Setting up the environment

Load the necessary libraries.
```{r libraries, results=FALSE, error=FALSE}
library(tidyverse)
library(vegan)
library(pairwiseAdonis)
library(ggplot2)
library(ggtext)
library(cowplot)
library(car)
library(multcomp)
library(ggpubr)
library(fdrtool)
library(gridExtra)
library(lme4)
library(multcomp)
```

### Importing data

Import metadata
```{r import_meta, error=FALSE}
metadata = read_tsv("year1_sample_mapping_metaphlan_noout.tsv")
metadata2 = metadata %>% 
  mutate(Ethnicity = case_when(Ethnicity == "African_American" ~ "Black", 
                               Ethnicity == "Caucasian" ~ "White"))
```

Import species-level relative abundance table with outliers (30f5, 37f1, and 33f1) removed. Transpose and recalculate relative abundance to sum to 1 instead of 100.
```{r import_relab, error=FALSE}
relab = read_tsv("year1_merged_abundance_table_species.txt")
relab1 = relab %>% mutate_if(is.numeric, ~./100)
relab1_t = relab1 %>% gather(file, count, 2:193) %>% 
  spread(body_site, count)
```

Import UniFrac matrices
```{r import_unifrac, error=FALSE}
wuni_fecal = as.dist(
  read.table("year1_merged_abundance_wunifrac_fecal.tsv"))
uuni_fecal = as.dist(
  read.table("year1_merged_abundance_uunifrac_fecal.tsv"))
wuni_oral = as.dist(
  read.table("year1_merged_abundance_wunifrac_oral.tsv"))
uuni_oral = as.dist(
  read.table("year1_merged_abundance_uunifrac_oral.tsv"))

wuni_fecal_black = as.dist(
  read.table("year1_merged_abundance_wunifrac_fecal_black.tsv"))
uuni_fecal_black = as.dist(
  read.table("year1_merged_abundance_uunifrac_fecal_black.tsv"))
wuni_oral_black = as.dist(
  read.table("year1_merged_abundance_wunifrac_oral_black.tsv"))
uuni_oral_black = as.dist(
  read.table("year1_merged_abundance_uunifrac_oral_black.tsv"))

wuni_fecal_white = as.dist(
  read.table("year1_merged_abundance_wunifrac_fecal_white.tsv"))
uuni_fecal_white = as.dist(
  read.table("year1_merged_abundance_uunifrac_fecal_white.tsv"))
wuni_oral_white = as.dist(
  read.table("year1_merged_abundance_wunifrac_oral_white.tsv"))
uuni_oral_white = as.dist(
  read.table("year1_merged_abundance_uunifrac_oral_white.tsv"))

wuni_fecalb = as.dist(
  read.table("year1_merged_abundance_wunifrac_fecalb.tsv"))
uuni_fecalb = as.dist(
  read.table("year1_merged_abundance_uunifrac_fecalb.tsv"))
wuni_oralb = as.dist(
  read.table("year1_merged_abundance_wunifrac_oralb.tsv"))
uuni_oralb = as.dist(
  read.table("year1_merged_abundance_uunifrac_oralb.tsv"))

wuni_fecala = as.dist(
  read.table("year1_merged_abundance_wunifrac_fecala.tsv"))
uuni_fecala = as.dist(
  read.table("year1_merged_abundance_uunifrac_fecala.tsv"))
wuni_orala = as.dist(
  read.table("year1_merged_abundance_wunifrac_orala.tsv"))
uuni_orala = as.dist(
  read.table("year1_merged_abundance_uunifrac_orala.tsv"))

wuni_fecald = as.dist(
  read.table("year1_merged_abundance_wunifrac_fecald.tsv"))
uuni_fecald = as.dist(
  read.table("year1_merged_abundance_uunifrac_fecald.tsv"))
```

### Filter samples

Filter out samples that have no identified species or are controls (blanks and mock communities). Separate oral samples from fecal samples. Remove species that don't occur in oral samples. Remove species that don't occur in fecal samples.

```{r filter}
relab_fecal = relab1_t %>% left_join(metadata2, by = "file") %>% 
  filter(fecal_use == 1) 
relab_fecal_filtered = relab_fecal %>% 
  mutate(total = rowSums(relab_fecal[,2:555])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
relab_fecal_filtered_b = relab_fecal_filtered %>% filter(Ethnicity == "Black")
relab_fecal_filtered_w = relab_fecal_filtered %>% filter(Ethnicity == "White")
relab_oral = relab1_t %>% left_join(metadata2, by = "file") %>% 
  filter(oral_use == 1) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
relab_oral_b = relab_oral %>% filter(Ethnicity == "Black")
relab_oral_w = relab_oral %>% filter(Ethnicity == "White")
```

### Calculating beta diversity metrics

UniFrac distances are calculated by a MetaPhlAn helper function. Vegan (2.5-7) is used to calculate the Bray-Curtis dissimilarity and Jaccard similarity metrics.
```{r beta_metrics, error=FALSE}
brayf = vegdist(relab_fecal_filtered[,2:337], method = "bray")
jaccf = vegdist(relab_fecal_filtered[,2:337], method = "jaccard")
brayfb = vegdist(relab_fecal_filtered_b[,2:337], method = "bray")
brayfw = vegdist(relab_fecal_filtered_w[,2:337], method = "bray")
jaccfb = vegdist(relab_fecal_filtered_b[,2:337], method = "jaccard")
jaccfw = vegdist(relab_fecal_filtered_w[,2:337], method = "jaccard")
brayo = vegdist(relab_oral[,2:272], method = "bray")
jacco = vegdist(relab_oral[,2:272], method = "jaccard")
brayob = vegdist(relab_oral_b[,2:272], method = "bray")
brayow = vegdist(relab_oral_w[,2:272], method = "bray")
jaccob = vegdist(relab_oral_b[,2:272], method = "jaccard")
jaccow = vegdist(relab_oral_w[,2:272], method = "jaccard")
```

### PERMANOVAs - All Samples
Differences in the taxonomic structure of the community associated with self-reported Ethnicity is assessed using permutational multivariate analysis of variance.

#### Bray-Curtis
Fecal
```{r permanova_brayf}
adonis2(brayf ~ Ethnicity, 
        data = relab_fecal_filtered, 
        permutations = 4999)
perm = how(nperm = 4999)
setBlocks(perm) = with(relab_fecal_filtered, subject_id)
adonis2(brayf ~ Ethnicity + `Study period`,
        data = relab_fecal_filtered,
        permutations = perm)
adonis2(brayf ~ `Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999, by = "margin")
pairwise.adonis(brayf, 
                factors = relab_fecal_filtered$`Study period`, 
                perm = 4999, 
                p.adjust.m='holm')
adonis(brayfb ~ `Study period`, 
        data = relab_fecal_filtered_b, 
        permutations = 4999, by = "margin")
adonis(brayfw ~ `Study period`, 
        data = relab_fecal_filtered_w, 
        permutations = 4999, by = "margin")
```

Oral
```{r permanova_brayo}
adonis2(brayo ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, 
        data = relab_oral, 
        permutations = 4999)
perm = how(nperm = 4999)
setBlocks(perm) = with(relab_oral, subject_id)
adonis2(brayo ~ Ethnicity + `Study period`,
        data = relab_oral,
        permutations = perm)
adonis2(brayo ~ `Study period`, 
        data = relab_oral, 
        permutations = 4999, by = "margin")
adonis2(brayob ~ `Study period`, 
        data = relab_oral_b, 
        permutations = 4999, by = "margin")
adonis2(brayow ~ `Study period`, 
        data = relab_oral_w, 
        permutations = 4999, by = "margin")
```

NMDS visualization of Bray-Curtis dissimilarity with all samples.
```{r nmds_bray, echo=FALSE, error=FALSE}
mds_bray_f<-metaMDS(brayf, k=2, trymax=499)
mds_bray_f_points<-mds_bray_f$points
mds_bray_f_points2<-merge(x=mds_bray_f_points, y = relab_fecal_filtered, 
                                  by.x = "row.names", by.y = "row.names")

brayfy1 <- ggplot(mds_bray_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_f_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Bray-Curtis - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, r<sup>2</sup> = 6.6% (All)<br><i>p</i> = 0.040, r<sup>2</sup> = 6.0% (Before)<br><i>p</i> < 0.001, r<sup>2</sup> = 8.0% (During)<br><i>p</i> = 0.014, r<sup>2</sup> = 6.9% (After)</b><br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.180",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayfy1

mds_bray_o<-metaMDS(brayo, k=2, trymax=499)
mds_bray_o_points<-mds_bray_o$points
mds_bray_o_points2<-merge(x=mds_bray_o_points, y = relab_oral, 
                                  by.x = "row.names", by.y = "row.names")
brayoy1 <- ggplot(mds_bray_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_o_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Bray-Curtis - Oral")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, r<sup>2</sup> = 12.7% (All)<br><i>p</i> = 0.004, r<sup>2</sup> = 16.8% (Before)<br><i>p</i> = 0.044, r<sup>2</sup> = 11.3% (After)</b><br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.181",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayoy1

brayfsy1 <- ggplot(mds_bray_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_f_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.991, r<sup>2</sup> = 0.9% (Both)<br><i>p</i> = 0.999, r<sup>2</sup> = 1.2% (Black)<br><i>p</i> = 0.952, r<sup>2</sup> = 1.9% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.180",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayfsy1

brayosy1 <- ggplot(mds_bray_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","After"="#2171b5")) + 
  scale_fill_manual(values = c("Before"="#238b45","After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_bray_o_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.992, r<sup>2</sup> = 0.8% (Both)<br><i>p</i> = 0.950, r<sup>2</sup> = 1.9% (Black)<br><i>p</i> = 0.874, r<sup>2</sup> = 2.6% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.181",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
brayosy1
```

```{r nmds_save_bray}
tiff(file = "figures/nmds_plot_taxa_bray_e_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayf + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayf + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayo + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()

tiff(file = "figures/nmds_plot_taxa_bray_s_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayfs + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfs + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayos + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Jaccard
Fecal
```{r permanova_jaccf}
adonis2(jaccf ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, 
        data = relab_fecal_filtered, 
        permutations = 4999)
perm = how(nperm = 4999)
setBlocks(perm) = with(relab_fecal_filtered, subject_id)
adonis2(jaccf ~ Ethnicity + `Study period`, 
        data = relab_fecal_filtered, 
        permutations = perm)
adonis2(jaccf ~ `Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999, by = "margin")
pairwise.adonis(jaccf, 
                factors = relab_fecal_filtered$`Study period`, 
                perm = 4999, 
                p.adjust.m='holm')
adonis(jaccfb ~ `Study period`, 
        data = relab_fecal_filtered_b, 
        permutations = 4999, by = "margin")
adonis(jaccfw ~ `Study period`, 
        data = relab_fecal_filtered_w, 
        permutations = 4999, by = "margin")
```

Oral
```{r permanova_jacco}
adonis2(jacco ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, 
        data = relab_oral, 
        permutations = 4999)
perm = how(nperm = 4999)
setBlocks(perm) = with(relab_oral, subject_id)
adonis2(jacco ~ Ethnicity + `Study period`, 
        data = relab_oral, 
        permutations = perm)
adonis2(jacco ~ `Study period`, 
        data = relab_oral, 
        permutations = 4999, by = "margin")
adonis2(jaccob ~ `Study period`, 
        data = relab_oral_b, 
        permutations = 4999, by = "margin")
adonis2(jaccow ~ `Study period`, 
        data = relab_oral_w, 
        permutations = 4999, by = "margin")
```

NMDS visualization of Jaccard similarity with all samples.
```{r nmds_jacc, echo=FALSE, error=FALSE}
mds_jacc_f<-metaMDS(jaccf, k=2, trymax=499, distance = "jaccard")
mds_jacc_f_points<-mds_jacc_f$points
mds_jacc_f_points2<-merge(x=mds_jacc_f_points, y = relab_fecal_filtered, 
                                  by.x = "row.names", by.y = "row.names")
jaccf <- ggplot(mds_jacc_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_f_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, r<sup>2</sup> = 5.8% (All)<br><i>p</i> = 0.012, r<sup>2</sup> = 5.9% (Before)<br><i>p</i> < 0.001, r<sup>2</sup> = 7.1% (During)<br><i>p</i> = 0.004, r<sup>2</sup> = 6.6% (After)</b><br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.180",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccf

mds_jacc_o<-metaMDS(jacco, k=2, trymax=499, distance = "jaccard")
mds_jacc_o_points<-mds_jacc_o$points
mds_jacc_o_points2<-merge(x=mds_jacc_o_points, y = relab_oral, 
                                  by.x = "row.names", by.y = "row.names")
jacco <- ggplot(mds_jacc_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_o_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Oral")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, r<sup>2</sup> = 9.6% (All)<br><i>p</i> = 0.005, r<sup>2</sup> = 12.7% (Before)<br><i>p</i> = 0.038, r<sup>2</sup> = 9.6% (After)</b><br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.148",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jacco

jaccfs <- ggplot(mds_jacc_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_f_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.998, r<sup>2</sup> = 1.0% (Both)<br><i>p</i> = 0.996, r<sup>2</sup> = 1.8% (Black)<br><i>p</i> = 0.977, r<sup>2</sup> = 2.0% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.180",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccfs

jaccos <- ggplot(mds_jacc_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","After"="#2171b5")) + 
  scale_fill_manual(values = c("Before"="#238b45","After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_jacc_o_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Jaccard - Oral")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.999, r<sup>2</sup> = 1.0% (Both)<br><i>p</i> = 0.982, r<sup>2</sup> = 2.4% (Black)<br><i>p</i> = 0.941, r<sup>2</sup> = 2.9% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.148",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
jaccos
```

```{r nmds_save_jacc}
tiff(file = "figures/nmds_plot_taxa_jacc_e_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccf + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccf + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jacco + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()

tiff(file = "figures/nmds_plot_taxa_jacc_s_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccfs + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfs + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccos + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Weighted UniFrac
Fecal
```{r permanova_wunif}
adonis2(wuni_fecal ~ Ethnicity, data = relab_fecal_filtered, 
        permutations = 4999)
perm = how(nperm = 4999)
setBlocks(perm) = with(relab_fecal_filtered, subject_id)
adonis2(wuni_fecal ~ Ethnicity + `Study period`, 
        data = relab_fecal_filtered, 
        permutations = perm)
adonis2(wuni_fecal ~ `Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999, by = "margin")
pairwise.adonis(wuni_fecal, 
                factors = relab_fecal_filtered$`Study period`, 
                perm = 4999, 
                p.adjust.m='holm')
adonis2(wuni_fecal_black ~ `Study period`, 
        data = relab_fecal_filtered_b, 
        permutations = 4999, by = "margin")
adonis2(wuni_fecal_white ~ `Study period`, 
        data = relab_fecal_filtered_w, 
        permutations = 4999, by = "margin")
```

Oral
```{r permanova_wunio}
adonis2(wuni_oral ~ Ethnicity, data = relab_oral, 
        permutations = 4999)
perm = how(nperm = 4999)
setBlocks(perm) = with(relab_oral, subject_id)
adonis2(wuni_oral ~ Ethnicity + `Study period`, 
        data = relab_oral, 
        permutations = perm)
adonis2(wuni_oral ~ `Study period`, 
        data = relab_oral, 
        permutations = 4999, by = "margin")
adonis2(wuni_oral_black ~ `Study period`, 
        data = relab_oral_b, 
        permutations = 4999, by = "margin")
adonis2(wuni_oral_white ~ `Study period`, 
        data = relab_oral_w, 
        permutations = 4999, by = "margin")
```

NMDS visualization of weighted Unifrac distances with all samples.
```{r nmds_wuni, echo=FALSE, error=FALSE}
mds_wuni_f<-metaMDS(wuni_fecal, k=2, trymax=499)
mds_wuni_f_points<-mds_wuni_f$points
mds_wuni_f_points2<-merge(x=mds_wuni_f_points, y = relab_fecal_filtered, 
                                  by.x = "row.names", by.y = "file")
wunif <- ggplot(mds_wuni_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8)))  +
  stat_ellipse(data = mds_wuni_f_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Ethnicity`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Weighted Unifrac - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, r<sup>2</sup> = 5.4% (All)</b><br><i>p</i> = 0.100, r<sup>2</sup> = 6.2% (Before)<br><b><i>p</i> = 0.004, r<sup>2</sup> = 7.5% (During)</b><br><i>p</i> = 0.177, r<sup>2</sup> = 4.9% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.145",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
wunif

mds_wuni_o<-metaMDS(wuni_oral, k=2, trymax=499)
mds_wuni_o_points<-mds_wuni_o$points
mds_wuni_o_points2<-merge(x=mds_wuni_o_points, y = relab_oral, 
                                  by.x = "row.names", by.y = "file")
wunio <- ggplot(mds_wuni_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8)))  +
  stat_ellipse(data = mds_wuni_o_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Ethnicity`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Weighted UniFrac - Oral") +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.012, r<sup>2</sup> = 10.3% (All)<br><i>p</i> = 0.033, r<sup>2</sup> = 15.3% (Before)</b><br><i>p</i> = 0.209, r<sup>2</sup> = 7.5% (After)", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.086",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
wunio

wunifs <- ggplot(mds_wuni_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_wuni_f_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Weighted UniFrac - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.847, r<sup>2</sup> = 1.2% (Both)<br><i>p</i> = 0.745, r<sup>2</sup> = 2.8% (Black)<br><i>p</i> = 0.940, r<sup>2</sup> = 1.7% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.145",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
wunifs

wunios <- ggplot(mds_wuni_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","After"="#2171b5")) + 
  scale_fill_manual(values = c("Before"="#238b45","After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_wuni_o_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Weighted UniFrac - Oral")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.920, r<sup>2</sup> = 0.7% (Both)<br><i>p</i> = 0.970, r<sup>2</sup> = 0.6% (Black)<br><i>p</i> = 0.562, r<sup>2</sup> = 3.3% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.086",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
wunios
```

```{r nmds_save_wuni}
tiff(file = "figures/nmds_plot_taxa_wuni_e_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(wunif + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(wunif + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(wunio + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()

tiff(file = "figures/nmds_plot_taxa_wuni_s_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(wunifs + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(wunifs + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(wunios + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Unweighted UniFrac
Fecal
```{r permanova_uunif}
adonis2(uuni_fecal ~ Ethnicity, data = relab_fecal_filtered, 
        permutations = 4999)
perm = how(nperm = 4999)
setBlocks(perm) = with(relab_fecal_filtered, subject_id)
adonis2(uuni_fecal ~ Ethnicity + `Study period`, 
        data = relab_fecal_filtered, 
        permutations = perm)
adonis2(uuni_fecal ~ `Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999, by = "margin")
pairwise.adonis(brayf, 
                factors = relab_fecal_filtered$`Study period`, 
                perm = 4999, 
                p.adjust.m='holm')
adonis2(uuni_fecal_black ~ `Study period`, 
        data = relab_fecal_filtered_b, 
        permutations = 4999, by = "margin")
adonis2(uuni_fecal_white ~ `Study period`, 
        data = relab_fecal_filtered_w, 
        permutations = 4999, by = "margin")
```

Oral
```{r permanova_uunio}
adonis2(uuni_oral ~ Ethnicity, data = relab_oral, 
        permutations = 4999)
perm = how(nperm = 4999)
setBlocks(perm) = with(relab_oral, subject_id)
adonis2(uuni_oral ~ Ethnicity + `Study period`, 
        data = relab_oral, 
        permutations = perm)
adonis2(uuni_oral ~ `Study period`, 
        data = relab_oral, 
        permutations = 4999, by = "margin")
pairwise.adonis(brayf, 
                factors = relab_fecal_filtered$`Study period`, 
                perm = 4999, 
                p.adjust.m='holm')
adonis2(uuni_oral_black ~ `Study period`, 
        data = relab_oral_b, 
        permutations = 4999, by = "margin")
adonis2(uuni_oral_white ~ `Study period`, 
        data = relab_oral_w, 
        permutations = 4999, by = "margin")
```

NMDS visualization of Bray-Curtis dissimilarity with all samples.
```{r nmds_uuni, echo=FALSE, error=FALSE}
mds_uuni_f<-metaMDS(uuni_fecal, k=2, trymax=499)
mds_uuni_f_points<-mds_uuni_f$points
mds_uuni_f_points2<-merge(x=mds_uuni_f_points, y = relab_fecal_filtered, 
                                  by.x = "row.names", by.y = "file")
uunif <- ggplot(mds_uuni_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_uuni_f_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Unweighted UniFrac - Fecal")   +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, r<sup>2</sup> = 7.7% (All)<br><i>p</i> = 0.015, r<sup>2</sup> = 6.9% (Before)<br><i>p</i> < 0.001, r<sup>2</sup> = 9.2% (During)<br><i>p</i> = 0.002, r<sup>2</sup> = 8.4% (After)</b>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.227",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
uunif

mds_uuni_o<-metaMDS(uuni_oral, k=2, trymax=499)
mds_uuni_o_points<-mds_uuni_o$points
mds_uuni_o_points2<-merge(x=mds_uuni_o_points, y = relab_oral, 
                                  by.x = "row.names", by.y = "file")
uunio <- ggplot(mds_uuni_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_uuni_o_points2, aes(x = MDS1, y = MDS2, 
                                              fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Unweighted UniFrac - Oral")   +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.010, r<sup>2</sup> = 7.4% (All)<br><i>p</i> = 0.036, r<sup>2</sup> = 11.7% (Before)</b><br><i>p</i> = 0.289, r<sup>2</sup> = 6.3% (After)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.117",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
uunio

uunifs <- ggplot(mds_uuni_f_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_uuni_f_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Unweighted UniFrac - Fecal")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.997, r<sup>2</sup> = 0.9% (Both)<br><i>p</i> = 0.995, r<sup>2</sup> = 1.6% (Black)<br><i>p</i> = 0.995, r<sup>2</sup> = 1.4% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.228",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
uunifs

uunios <- ggplot(mds_uuni_o_points2, aes(x = MDS1, y = MDS2, 
                                                 color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","After"="#2171b5")) + 
  scale_fill_manual(values = c("Before"="#238b45","After"="#2171b5")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(data = mds_uuni_o_points2, aes(x = MDS1, y = MDS2, 
                                              fill = `Study period`), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) + 
  ggtitle("Unweighted UniFrac - Oral")  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.615, r<sup>2</sup> = 2.2% (Both)<br><i>p</i> = 0.946, r<sup>2</sup> = 2.3% (Black)<br><i>p</i> = 0.408, r<sup>2</sup> = 5.1% (White)<br>", 
           x = Inf, y = Inf, size = 4,
           hjust = 1.1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.117",
           x = Inf, y = -Inf, size = 4,
           hjust = 1.1, vjust = 0)
uunios
```

```{r nmds_save_uuni}
tiff(file = "figures/nmds_plot_taxa_uuni_e_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(uunif + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(uunif + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(uunio + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()

tiff(file = "figures/nmds_plot_taxa_uuni_s_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(uunifs + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(uunifs + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(uunios + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

### PERMANOVAs - Before samples

#### Filtering samples
Filter samples from before the diet intervention and recalculate Bray-Curtis and Jaccard metrics

```{r filter_before}
relab_fecal_filtered_before = relab_fecal_filtered %>% 
  filter(`Study period` == "Before")
relab_oral_before = relab_oral %>% 
  filter(`Study period` == "Before")

brayfb = vegdist(relab_fecal_filtered_before[,2:337], method = "bray")
jaccfb = vegdist(relab_fecal_filtered_before[,2:337], method = "jaccard")
brayob = vegdist(relab_oral_before[,2:272], method = "bray")
jaccob = vegdist(relab_oral_before[,2:272], method = "jaccard")
```


#### Bray-Curtis
Fecal
```{r permanova_brayfb}
adonis2(brayfb ~ Ethnicity, data = relab_fecal_filtered_before, 
        permutations = 4999)
```

Oral
```{r permanova_brayob}
adonis2(brayob ~ Ethnicity, data = relab_oral_before, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with before samples.
```{r nmds_bray_before, echo=FALSE, error=FALSE}
mds_bray_fb<-metaMDS(brayfb, k=2, trymax=499)
mds_bray_fb_points<-mds_bray_fb$points
mds_bray_fb_points2<-merge(x=mds_bray_fb_points, 
                           y = relab_fecal_filtered_before, 
                                  by.x = "row.names", by.y = "row.names")
brayfb <- ggplot(mds_bray_fb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity, 
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Bray-Curtis - Fecal")
brayfb

mds_bray_ob<-metaMDS(brayob, k=2, trymax=499)
mds_bray_ob_points<-mds_bray_ob$points
mds_bray_ob_points2<-merge(x=mds_bray_ob_points, y = relab_oral_before, 
                                  by.x = "row.names", by.y = "row.names")
brayob <- ggplot(mds_bray_ob_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Bray-Curtis - Oral")
brayob
```

```{r nmds_save_bray_before}
tiff(file = "figures/nmds_plot_taxa_bray_before_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayfb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayob + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Jaccard
Fecal
```{r permanova_jaccfb}
adonis2(jaccfb ~ Ethnicity, data = relab_fecal_filtered_before, 
        permutations = 4999)
```

Oral
```{r permanova_jaccob}
adonis2(jaccob ~ Ethnicity, data = relab_oral_before, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with before samples.
```{r nmds_jacc_before, echo=FALSE, error=FALSE}
mds_jacc_fb<-metaMDS(jaccfb, k=2, trymax=499, distance = "jaccard")
mds_jacc_fb_points<-mds_jacc_fb$points
mds_jacc_fb_points2<-merge(x=mds_jacc_fb_points, 
                           y = relab_fecal_filtered_before, 
                                  by.x = "row.names", by.y = "row.names")
jaccfb <- ggplot(mds_jacc_fb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Jaccard - Fecal")
jaccfb

mds_jacc_ob<-metaMDS(jaccob, k=2, trymax=499, distance = "jaccard")
mds_jacc_ob_points<-mds_jacc_ob$points
mds_jacc_ob_points2<-merge(x=mds_jacc_ob_points, y = relab_oral_before, 
                                  by.x = "row.names", by.y = "row.names")
jaccob <- ggplot(mds_jacc_ob_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Jaccard - Oral")
jaccob
```

```{r nmds_save_jacc_before}
tiff(file = "figures/nmds_plot_taxa_jacc_before_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccfb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccob + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Weighted UniFrac
Fecal
```{r permanova_wunif_before}
adonis2(wuni_fecalb ~ Ethnicity, data = relab_fecal_filtered_before, 
        permutations = 4999)
```

Oral
```{r permanova_wunio_before}
adonis2(wuni_oralb ~ Ethnicity, data = relab_oral_before, 
        permutations = 4999)
```

NMDS visualization of weighted Unifrac distances with before samples.
```{r nmds_wuni_before, echo=FALSE, error=FALSE}
mds_wuni_fb<-metaMDS(wuni_fecalb, k=2, trymax=499)
mds_wuni_fb_points<-mds_wuni_fb$points
mds_wuni_fb_points2<-merge(x=mds_wuni_fb_points, 
                           y = relab_fecal_filtered_before, 
                                  by.x = "row.names", by.y = "file")
wunifb <- ggplot(mds_wuni_fb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Weighted Unifrac - Fecal")
wunifb

mds_wuni_ob<-metaMDS(wuni_oralb, k=2, trymax=499)
mds_wuni_ob_points<-mds_wuni_ob$points
mds_wuni_ob_points2<-merge(x=mds_wuni_ob_points, y = relab_oral_before, 
                                  by.x = "row.names", by.y = "file")
wuniob <- ggplot(mds_wuni_ob_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Weighted UniFrac - Oral")
wuniob
```

```{r nmds_save_wuni_before}
tiff(file = "figures/nmds_plot_taxa_wuni_before_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(wunifb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(wunifb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(wuniob + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Unweighted UniFrac
Fecal
```{r permanova_uunif_before}
adonis2(uuni_fecalb ~ Ethnicity, data = relab_fecal_filtered_before, 
        permutations = 4999)
```

Oral
```{r permanova_uunio_before}
adonis2(uuni_oralb ~ Ethnicity, data = relab_oral_before, 
        permutations = 4999)
```

NMDS visualization of unweighted UniFrac distances with before samples.
```{r nmds_uuni_before, echo=FALSE, error=FALSE}
mds_uuni_fb<-metaMDS(uuni_fecalb, k=2, trymax=499)
mds_uuni_fb_points<-mds_uuni_fb$points
mds_uuni_fb_points2<-merge(x=mds_uuni_fb_points, 
                           y = relab_fecal_filtered_before, 
                                  by.x = "row.names", by.y = "file")
uunifb <- ggplot(mds_uuni_fb_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Unweighted Unifrac - Fecal")
uunifb

mds_uuni_ob<-metaMDS(uuni_oralb, k=2, trymax=499)
mds_uuni_ob_points<-mds_uuni_ob$points
mds_uuni_ob_points2<-merge(x=mds_uuni_ob_points, y = relab_oral_before, 
                                  by.x = "row.names", by.y = "file")
uuniob <- ggplot(mds_uuni_ob_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Unweighted UniFrac - Oral")
uuniob
```

```{r nmds_save_uuni_before}
tiff(file = "figures/nmds_plot_taxa_uuni_before_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(uunifb + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(uunifb + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(uuniob + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```
### PERMANOVAs - During samples

#### Filter samples
Filter samples from during the diet intervention and recalculate Bray-Curtis and Jaccard metrics
```{r filter_during}
relab_fecal_filtered_during = relab_fecal_filtered %>% 
  filter(`Study period` == "During")

brayfd = vegdist(relab_fecal_filtered_during[,2:337], method = "bray")
jaccfd = vegdist(relab_fecal_filtered_during[,2:337], method = "jaccard")
```

#### Bray-Curtis
Fecal
```{r permanova_brayfd}
adonis2(brayfd ~ Ethnicity, data = relab_fecal_filtered_during, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with during samples.
```{r nmds_bray_during, echo=FALSE, error=FALSE}
mds_bray_fd<-metaMDS(brayfd, k=2, trymax=499)
mds_bray_fd_points<-mds_bray_fd$points
mds_bray_fd_points2<-merge(x=mds_bray_fd_points, 
                           y = relab_fecal_filtered_during, 
                                  by.x = "row.names", by.y = "row.names")
brayfd <- ggplot(mds_bray_fd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity, 
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Bray-Curtis - Fecal")
brayfd
```

```{r nmds_save_bray_during}
tiff(file = "figures/nmds_plot_taxa_bray_during_y1.tif", res = 150, width = 11, 
     height = 7, units = "in")
brayfd
dev.off()
```

#### Jaccard
Fecal
```{r permanova_jaccfd}
adonis2(jaccfd ~ Ethnicity, data = relab_fecal_filtered_during, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with during samples.
```{r nmds_jacc_during, echo=FALSE, error=FALSE}
mds_jacc_fd<-metaMDS(jaccfd, k=2, trymax=499, distance = "jaccard")
mds_jacc_fd_points<-mds_jacc_fd$points
mds_jacc_fd_points2<-merge(x=mds_jacc_fd_points, 
                           y = relab_fecal_filtered_during, 
                                  by.x = "row.names", by.y = "row.names")
jaccfd <- ggplot(mds_jacc_fd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Jaccard - Fecal")
jaccfd
```

```{r nmds_save_jacc_during}
tiff(file = "figures/nmds_plot_taxa_jacc_during_y1.tif", res = 150, width = 11, 
     height = 7, units = "in")
jaccfd
dev.off()
```

#### Weighted UniFrac
Fecal
```{r permanova_wunif_during}
adonis2(wuni_fecald ~ Ethnicity, data = relab_fecal_filtered_during, 
        permutations = 4999)
```

NMDS visualization of weighted Unifrac distances with during samples.
```{r nmds_wuni_during, echo=FALSE, error=FALSE}
mds_wuni_fd<-metaMDS(wuni_fecald, k=2, trymax=499)
mds_wuni_fd_points<-mds_wuni_fd$points
mds_wuni_fd_points2<-merge(x=mds_wuni_fd_points, 
                           y = relab_fecal_filtered_during, 
                                  by.x = "row.names", by.y = "file")
wunifd <- ggplot(mds_wuni_fd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Weighted Unifrac - Fecal")
wunifd
```

```{r nmds_save_wuni_during}
tiff(file = "figures/nmds_plot_taxa_wuni_during_y1.tif", res = 150, width = 11, 
     height = 7, units = "in")
wunifd
dev.off()
```

#### Unweighted UniFrac
Fecal
```{r permanova_uunif_during}
adonis2(uuni_fecald ~ Ethnicity, data = relab_fecal_filtered_during, 
        permutations = 4999)
```

NMDS visualization of unweighted Unifrac distance with during samples.
```{r nmds_uuni_during, echo=FALSE, error=FALSE}
mds_uuni_fd<-metaMDS(uuni_fecald, k=2, trymax=499)
mds_uuni_fd_points<-mds_uuni_fd$points
mds_uuni_fd_points2<-merge(x=mds_uuni_fd_points, 
                           y = relab_fecal_filtered_during, 
                                  by.x = "row.names", by.y = "file")
uunifd <- ggplot(mds_uuni_fd_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Unweighted Unifrac - Fecal")
uunifd
```

```{r nmds_save_uuni_during}
tiff(file = "figures/nmds_plot_taxa_uuni_during_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
uunifd
dev.off()
```
### PERMANOVAs - After samples

#### Filter samples

Filter after samples and recalculate Bray-Curtis and Jaccard metrics
```{r filter_after}
relab_fecal_filtered_after = relab_fecal_filtered %>% 
  filter(`Study period` == "After")
relab_oral_after = relab_oral %>% 
  filter(`Study period` == "After")

brayfa = vegdist(relab_fecal_filtered_after[,2:337], method = "bray")
jaccfa = vegdist(relab_fecal_filtered_after[,2:337], method = "jaccard")
brayoa = vegdist(relab_oral_after[,2:272], method = "bray")
jaccoa = vegdist(relab_oral_after[,2:272], method = "jaccard")
```
#### Bray-Curtis
Fecal
```{r permanova_brayfa}
adonis2(brayfa ~ Ethnicity, data = relab_fecal_filtered_after, 
        permutations = 4999)
```

Oral
```{r permanova_brayoa}
adonis2(brayoa ~ Ethnicity, data = relab_oral_after, 
        permutations = 4999)
```

NMDS visualization of Bray-Curtis dissimilarity with after samples.
```{r nmds_bray_after, echo=FALSE, error=FALSE}
mds_bray_fa<-metaMDS(brayfa, k=2, trymax=499)
mds_bray_fa_points<-mds_bray_fa$points
mds_bray_fa_points2<-merge(x=mds_bray_fa_points, 
                           y = relab_fecal_filtered_after, 
                                  by.x = "row.names", by.y = "row.names")
brayfa <- ggplot(mds_bray_fa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity, 
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Bray-Curtis - Fecal")
brayfa

mds_bray_oa<-metaMDS(brayoa, k=2, trymax=499)
mds_bray_oa_points<-mds_bray_oa$points
mds_bray_oa_points2<-merge(x=mds_bray_oa_points, y = relab_oral_after, 
                                  by.x = "row.names", by.y = "row.names")
brayoa <- ggplot(mds_bray_oa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Bray-Curtis - Oral")
brayoa
```

```{r nmds_save_bray_after}
tiff(file = "figures/nmds_plot_taxa_bray_after_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(brayfa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(brayfa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(brayoa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Jaccard
Fecal
```{r permanova_jaccfa}
adonis2(jaccfa ~ Ethnicity, data = relab_fecal_filtered_after, 
        permutations = 4999)
```

Oral
```{r permanova_jaccoa}
adonis2(jaccoa ~ Ethnicity, data = relab_oral_after, 
        permutations = 4999)
```

NMDS visualization of Jaccard similarity with after samples.
```{r nmds_jacc_after, echo=FALSE, error=FALSE}
mds_jacc_fa<-metaMDS(jaccfa, k=2, trymax=499, distance = "jaccard")
mds_jacc_fa_points<-mds_jacc_fa$points
mds_jacc_fa_points2<-merge(x=mds_jacc_fa_points, 
                           y = relab_fecal_filtered_after, 
                                  by.x = "row.names", by.y = "row.names")
jaccfa <- ggplot(mds_jacc_fa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Jaccard - Fecal")
jaccfa

mds_jacc_oa<-metaMDS(jaccoa, k=2, trymax=499, distance = "jaccard")
mds_jacc_oa_points<-mds_jacc_oa$points
mds_jacc_oa_points2<-merge(x=mds_jacc_oa_points, y = relab_oral_after, 
                                  by.x = "row.names", by.y = "row.names")
jaccoa <- ggplot(mds_jacc_oa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Jaccard - Oral")
jaccoa
```

```{r nmds_save_jacc_after}
tiff(file = "figures/nmds_plot_taxa_jacc_after_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(jaccfa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(jaccfa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(jaccoa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Weighted UniFrac
Fecal
```{r permanova_wunif_after}
adonis2(wuni_fecala ~ Ethnicity, data = relab_fecal_filtered_after, 
        permutations = 4999)
```

Oral
```{r permanova_wunio_after}
adonis2(wuni_orala ~ Ethnicity, data = relab_oral_after, 
        permutations = 4999)
```

NMDS visualization of weighted Unifrac distances with after samples.
```{r nmds_wuni_after, echo=FALSE, error=FALSE}
mds_wuni_fa<-metaMDS(wuni_fecala, k=2, trymax=499)
mds_wuni_fa_points<-mds_wuni_fa$points
mds_wuni_fa_points2<-merge(x=mds_wuni_fa_points, 
                           y = relab_fecal_filtered_after, 
                                  by.x = "row.names", by.y = "file")
wunifa <- ggplot(mds_wuni_fa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Weighted Unifrac - Fecal")
wunifa

mds_wuni_oa<-metaMDS(wuni_orala, k=2, trymax=499)
mds_wuni_oa_points<-mds_wuni_oa$points
mds_wuni_oa_points2<-merge(x=mds_wuni_oa_points, y = relab_oral_after, 
                                  by.x = "row.names", by.y = "file")
wunioa <- ggplot(mds_wuni_oa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Weighted UniFrac - Oral")
wunioa
```

```{r nmds_save_wuni_after}
tiff(file = "figures/nmds_plot_taxa_wuni_after_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(wunifa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(wunifa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(wunioa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

#### Unweighted UniFrac
Fecal
```{r permanova_uunif_after}
adonis2(uuni_fecala ~ Ethnicity, data = relab_fecal_filtered_after, 
        permutations = 4999)
```

Oral
```{r permanova_uunio_after}
adonis2(uuni_orala ~ Ethnicity, data = relab_oral_after, 
        permutations = 4999)
```

NMDS visualization of unweighted UniFrac distances with after samples.
```{r nmds_uuni_after, echo=FALSE, error=FALSE}
mds_uuni_fa<-metaMDS(uuni_fecala, k=2, trymax=499)
mds_uuni_fa_points<-mds_uuni_fa$points
mds_uuni_fa_points2<-merge(x=mds_uuni_fa_points, 
                           y = relab_fecal_filtered_after, 
                                  by.x = "row.names", by.y = "file")
uunifa <- ggplot(mds_uuni_fa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Unweighted Unifrac - Fecal")
uunifa

mds_uuni_oa<-metaMDS(uuni_orala, k=2, trymax=499)
mds_uuni_oa_points<-mds_uuni_oa$points
mds_uuni_oa_points2<-merge(x=mds_uuni_oa_points, y = relab_oral_after, 
                                  by.x = "row.names", by.y = "file")
uunioa <- ggplot(mds_uuni_oa_points2, aes(x = MDS1, y = MDS2, 
                                                 color = Ethnicity,
                                        shape = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_text(size=rel(2)), 
        axis.title.y=element_text(size=rel(2)),
        plot.title = element_text(size=rel(3)),
        legend.title = element_text(size=rel(2)),
        legend.text = element_text(size = rel(1.8))) +
  stat_ellipse(aes(x = MDS1, y = MDS2, group = Ethnicity, 
                   linetype = Ethnicity), 
               type = "t", level = 0.95) + 
  ggtitle("Taxonomy - Unweighted UniFrac - Oral")
uunioa
```

```{r nmds_save_uuni_after}
tiff(file = "figures/nmds_plot_taxa_uuni_after_y1.tif", res = 150, width = 18, 
     height = 7, units = "in")
legend1 = get_legend(uunifa + theme(legend.box.margin = margin(0, 0, 0, 12)))
col1 = plot_grid(uunifa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('A'), 
                 label_size = 20)
col2 = plot_grid(uunioa + theme(legend.position = "none"),
                 nrow = 1, ncol = 1, labels = c('B'), 
                 label_size = 20) 
plot = plot_grid(col1, col2, legend1, nrow = 1, ncol = 3, 
          rel_widths = c(1.5, 1.5, 0.75), align = "h", axis = "t")
plot
dev.off()
```

### Alpha Diversity

Alpha diversity was calculated using species-level metaphlan results in vegan (2.5-7). Shannon diversity and Shannon-Weaver evenness (Pielou's evenness) were calculated.

```{r alpha_d}
shannond_fecal = diversity(relab_fecal_filtered[,2:337], index = "shannon")
shannond_oral = diversity(relab_oral[,2:272], index = "shannon")

shannone_fecal = shannond_fecal/log(specnumber(relab_fecal_filtered[,2:337]))
shannone_oral = shannond_oral/log(specnumber(relab_oral[,2:272]))

fecal_diversity = cbind(relab_fecal_filtered[,1], relab_fecal_filtered[,338:361], shannond_fecal, shannone_fecal)
fecal_diversity$`Study period` = as.factor(fecal_diversity$`Study period`)
oral_diversity = cbind(relab_oral[,1], relab_oral[,273:296], shannond_oral, shannone_oral)
oral_diversity$`Study period` = as.factor(oral_diversity$`Study period`)
```

A Wilcoxon rank sum test was used to examine differences between self-reported ethnicities and study period (before/during/after) in diversity. 
```{r alpha_models_fecal}
div_fecal = wilcox.test(shannond_fecal ~ Ethnicity, 
                        data = fecal_diversity, 
                        paired = F, exact = F)
div_fecal

div_fecal = lmer(shannond_fecal ~ Ethnicity + `Study period` + (1|subject_id), data = fecal_diversity)
summary(div_fecal)
Anova(div_fecal)
summary(glht(div_fecal,linfct=mcp(`Study period`="Tukey")))

even_fecal = wilcox.test(shannone_fecal ~ Ethnicity, 
                          data = fecal_diversity, 
                        paired = F, exact = F)
even_fecal

div_fecals = kruskal.test(shannond_fecal ~ `Study period`, 
                        data = fecal_diversity)
div_fecals

even_fecals = kruskal.test(shannone_fecal ~ `Study period`, 
                          data = fecal_diversity)
even_fecals
```

```{r alpha_models_oral}
div_oral = wilcox.test(shannond_oral ~ Ethnicity, 
                       data = oral_diversity,
                       paired = F, exact = F)
div_oral

even_oral = wilcox.test(shannone_oral ~ Ethnicity, 
                        data = oral_diversity,
                        paired = F, exact = F)
even_oral

div_orals = wilcox.test(shannond_oral ~ `Study period`, 
                       data = oral_diversity,
                       paired = F, exact = F)
div_orals

even_orals = wilcox.test(shannone_oral ~ `Study period`, 
                        data = oral_diversity,
                        paired = F, exact = F)
even_orals
```
Shannon diversity and evenness did not vary across self-reported ethnicities or study periods in either fecal or oral samples.

```{r alpha_graphs, echo=FALSE}
fecal_diversity$`Study period` = factor(fecal_diversity$`Study period`, 
                                        levels = c("Before", "During", "After"))

fecal_div_all = ggplot(fecal_diversity, aes(Ethnicity, shannond_fecal)) +
  geom_violin(scale = "width", aes(fill = Ethnicity),color=NA, alpha = 0.3,trim=T) +
  geom_jitter(size=1,aes(color = Ethnicity),shape=1) + 
  geom_boxplot(fill='black',width = 0.1, outlier.size = NA, outlier.shape = NA) + 
  stat_summary(fun = "median", size = 2, geom = "point", colour='white', 
               shape=1, stroke = 0.8) + 
  scale_color_manual(values=c("#f5c4ca","#87ceeb")) + 
  scale_fill_manual(values=c("#f5c4ca","#87ceeb")) + 
  theme_classic() + labs(y = "Shannon Diversity") + labs(x = "Ethnicity") +
  ggtitle("Fecal") +
  theme_bw() + theme(legend.position="NULL") + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1)) + 
  theme(axis.text.x=element_text(angle=0,hjust=0.5,vjust=0.5))

oral_div_all = ggplot(oral_diversity, aes(Ethnicity, shannond_oral)) +
  geom_violin(scale = "width", aes(fill = Ethnicity),color=NA, alpha = 0.3,trim=T) +
  geom_jitter(size=1,aes(color = Ethnicity),shape=1) + 
  geom_boxplot(fill='black',width = 0.1, outlier.size = NA, outlier.shape = NA) + 
  stat_summary(fun = "median", size = 2, geom = "point", colour='white', 
               shape=1, stroke = 0.8) + 
  scale_color_manual(values=c("#f5c4ca","#87ceeb")) + 
  scale_fill_manual(values=c("#f5c4ca","#87ceeb")) + 
  theme_classic() + labs(y = "Shannon Diversity") + labs(x = "Ethnicity") +
  ggtitle("Oral") +
  theme_bw() + theme(legend.position="NULL") + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1)) + 
  theme(axis.text.x=element_text(angle=0,hjust=0.5,vjust=0.5))

fecal_even_all = ggplot(fecal_diversity, aes(Ethnicity, shannone_fecal)) +
  geom_violin(scale = "width", aes(fill = Ethnicity),color=NA, alpha = 0.3,trim=T) +
  geom_jitter(size=1,aes(color = Ethnicity),shape=1) + 
  geom_boxplot(fill='black',width = 0.1, outlier.size = NA, outlier.shape = NA) + 
  stat_summary(fun = "median", size = 2, geom = "point", colour='white', 
               shape=1, stroke = 0.8) + 
  scale_color_manual(values=c("#f5c4ca","#87ceeb")) + 
  scale_fill_manual(values=c("#f5c4ca","#87ceeb")) + 
  theme_classic() + labs(y = "Shannon Evenness") + labs(x = "Ethnicity") +
  ggtitle("Fecal") +
  theme_bw() + theme(legend.position="NULL") + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1)) + 
  theme(axis.text.x=element_text(angle=0,hjust=0.5,vjust=0.5))

oral_even_all = ggplot(oral_diversity, aes(Ethnicity, shannone_oral)) +
  geom_violin(scale = "width", aes(fill = Ethnicity),color=NA, alpha = 0.3,trim=T) +
  geom_jitter(size=1,aes(color = factor(Ethnicity)),shape=1) + 
  geom_boxplot(fill='black',width = 0.1, outlier.size = NA, outlier.shape = NA) + 
  stat_summary(fun = "median", size = 2, geom = "point", colour='white', 
               shape=1, stroke = 0.8) + 
  scale_color_manual(values=c("#f5c4ca","#87ceeb")) + 
  scale_fill_manual(values=c("#f5c4ca","#87ceeb")) + 
  theme_classic() + labs(y = "Shannon Evenness") + labs(x = "Ethnicity") + 
  theme_bw() + theme(legend.position="NULL") + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1)) + 
  theme(axis.text.x=element_text(angle=0,hjust=0.5,vjust=0.5))

fecal_div_study = ggplot(fecal_diversity, aes(`Study period`, shannond_fecal)) +
  geom_violin(scale = "width", aes(fill = `Study period`),color=NA, alpha = 0.3,trim=T) +
  geom_jitter(size=1,aes(color = `Study period`),shape=1) + 
  geom_boxplot(fill='black',width = 0.1, outlier.size = NA, outlier.shape = NA) + 
  stat_summary(fun = "median", size = 2, geom = "point", colour='white', 
               shape=1, stroke = 0.8) + 
  scale_color_manual(values=c("Before"="#238b45",
                              "During" = "#d94801", 
                              "After"="#2171b5")) + 
  scale_fill_manual(values=c("Before"="#238b45",
                             "During" = "#d94801", 
                             "After"="#2171b5")) + 
    facet_wrap(. ~ Ethnicity, scales = "free") + 
  theme_classic() + labs(y = "Shannon Diversity") + labs(x = "Study period") +
  ggtitle("Fecal") +
  theme_bw() + theme(legend.position="NULL") + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1)) + 
  theme(axis.text.x=element_text(angle=0,hjust=0.5,vjust=0.5))

oral_div_study = ggplot(oral_diversity, aes(`Study period`, shannond_oral)) +
  geom_violin(scale = "width", aes(fill = `Study period`),color=NA, alpha = 0.3,trim=T) +
  geom_jitter(size=1,aes(color = `Study period`),shape=1) + 
  geom_boxplot(fill='black',width = 0.1, outlier.size = NA, outlier.shape = NA) + 
  stat_summary(fun = "median", size = 2, geom = "point", colour='white', 
               shape=1, stroke = 0.8) + 
  scale_color_manual(values=c("Before"="#238b45", 
                              "After"="#2171b5")) + 
  scale_fill_manual(values=c("Before"="#238b45", 
                             "After"="#2171b5")) +
    facet_wrap(. ~ Ethnicity, scales = "free") + 
  theme_classic() + labs(y = "Shannon Diversity") + labs(x = "Study period") +
  ggtitle("Oral") +
  theme_bw() + theme(legend.position="NULL") + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1)) + 
  theme(axis.text.x=element_text(angle=0,hjust=0.5,vjust=0.5))

fecal_even_study = ggplot(fecal_diversity, aes(`Study period`, shannone_fecal)) +
  geom_violin(scale = "width", aes(fill = `Study period`),color=NA, alpha = 0.3,trim=T) +
  geom_jitter(size=1,aes(color = `Study period`),shape=1) + 
  geom_boxplot(fill='black',width = 0.1, outlier.size = NA, outlier.shape = NA) + 
  stat_summary(fun = "median", size = 2, geom = "point", colour='white', 
               shape=1, stroke = 0.8) + 
  scale_color_manual(values=c("Before"="#238b45",
                              "During" = "#d94801", 
                              "After"="#2171b5")) + 
  scale_fill_manual(values=c("Before"="#238b45",
                             "During" = "#d94801", 
                             "After"="#2171b5")) + 
  facet_wrap(. ~ Ethnicity, scales = "free") + 
  theme_classic() + labs(y = "Shannon Evenness") + labs(x = "Study Period") +
  ggtitle("Fecal") +
  theme_bw() + theme(legend.position="NULL") + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1)) + 
  theme(axis.text.x=element_text(angle=0,hjust=0.5,vjust=0.5))

oral_even_study = ggplot(oral_diversity, aes(`Study period`, shannone_oral)) +
  geom_violin(scale = "width", aes(fill = `Study period`),color=NA, alpha = 0.3,trim=T) +
  geom_jitter(size=1,aes(color = factor(`Study period`)),shape=1) + 
  geom_boxplot(fill='black',width = 0.1, outlier.size = NA, outlier.shape = NA) + 
  stat_summary(fun = "median", size = 2, geom = "point", colour='white', 
               shape=1, stroke = 0.8) + 
  scale_color_manual(values=c("Before"="#238b45", 
                              "After"="#2171b5")) + 
  scale_fill_manual(values=c("Before"="#238b45", 
                             "After"="#2171b5")) + 
  facet_wrap(. ~ Ethnicity, scales = "free") + 
  theme_classic() + labs(y = "Shannon Evenness") + labs(x = "Study period") + 
  theme_bw() + theme(legend.position="NULL") + 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_rect(colour = "black", size=1)) + 
  theme(axis.text.x=element_text(angle=0,hjust=0.5,vjust=0.5))

tiff(file="figures/fecal_alpha_diversity_e_y1.tif", res=150, width=9, height=4, units="in")
ggarrange(fecal_div_all, fecal_even_all, ncol = 2, nrow=1, 
          common.legend = TRUE, legend = "right", align = "h", 
          labels = c(""))
dev.off()

tiff(file="figures/oral_alpha_diversity_e_y1.tif", res=150, width=9, height=4, units="in")
ggarrange(oral_div_all, oral_even_all, ncol = 2, nrow=1, 
          common.legend = TRUE, legend = "right", align = "h", 
          labels = c(""))
dev.off()

tiff(file="figures/fecal_alpha_diversity_s_y1.tif", res=150, width=15, height=4, units="in")
ggarrange(fecal_div_study, fecal_even_study, ncol = 2, nrow=1, 
          common.legend = TRUE, legend = "right", align = "h", 
          labels = c(""))
dev.off()

tiff(file="figures/oral_alpha_diversity_s_y1.tif", res=150, width=15, height=4, units="in")
ggarrange(oral_div_study, oral_even_study, ncol = 2, nrow=1, 
          common.legend = TRUE, legend = "right", align = "h", 
          labels = c(""))
dev.off()
```

### Taxonomic abundance

Generalized linear mixed effects models using a negative binomial distribution were used to assess if self-reported Ethnicity or study period is associated with the abundance of individual species.

```{r taxa_models_fecal}
Ethnicity_estimatematrix = mat.or.vec(336,2)
Ethnicity_pvaluematrix = mat.or.vec(336,2)

relab_fecal_filtered = as.data.frame(relab_fecal_filtered)

for(i in 2:337) {
  variable = relab_fecal_filtered[,i]
  relab_fecal_filtered[,i]
  b = wilcox.test(variable ~ Ethnicity, 
                  data = relab_fecal_filtered,
                  paired = F, exact = F)
  Ethnicity_estimatematrix[i-1,2] = b$statistic
  Ethnicity_pvaluematrix[i-1,2] = b$p.value
  Ethnicity_estimatematrix[i-1,1] = names(relab_fecal_filtered)[i]
  Ethnicity_pvaluematrix[i-1,1] = names(relab_fecal_filtered)[i]
}


taxa_Ethnicity = bind_cols(as.data.frame(Ethnicity_estimatematrix[,1:2]), 
                     as.data.frame(Ethnicity_pvaluematrix[,2]))
write.csv(taxa_Ethnicity, "tables/taxa_fecal_wilcox_Ethnicity_y1.csv")
taxa_Ethnicity_nona = filter(taxa_Ethnicity, taxa_Ethnicity[,3] != "NaN") 
taxa_Ethnicity_nona[,3] = as.numeric(as.character(taxa_Ethnicity_nona[,3]))
taxa_Ethnicity_corrected = bind_cols(taxa_Ethnicity_nona, 
                               as.data.frame(fdrtool(taxa_Ethnicity_nona[,3], 
                                                     statistic = "pvalue", 
                                                     plot = F)))
write.csv(taxa_Ethnicity_corrected, "tables/taxa_fecal_wilcox_Ethnicity_corrected_y1.csv")
```

```{r taxa_models_oral}
Ethnicity_estimatematrix = mat.or.vec(271,2)
Ethnicity_pvaluematrix = mat.or.vec(271,2)

relab_oral = as.data.frame(relab_oral)

for(i in 2:272) {
  variable = relab_oral[,i]
  b = wilcox.test(variable ~ Ethnicity, 
                  data = relab_oral,
                  paired = F, exact = F)
  Ethnicity_estimatematrix[i-1,2] = b$statistic
  Ethnicity_pvaluematrix[i-1,2] = b$p.value
  Ethnicity_estimatematrix[i-1,1] = names(relab_fecal_filtered)[i]
  Ethnicity_pvaluematrix[i-1,1] = names(relab_fecal_filtered)[i]
}

taxa_Ethnicity = bind_cols(as.data.frame(Ethnicity_estimatematrix[,1:2]), 
                     as.data.frame(Ethnicity_pvaluematrix[,2]))
write.csv(taxa_Ethnicity, "tables/taxa_oral_wilcox_Ethnicity_y1.csv")
taxa_Ethnicity_nona = filter(taxa_Ethnicity, taxa_Ethnicity[,3] != "NaN") 
taxa_Ethnicity_nona[,3] = as.numeric(as.character(taxa_Ethnicity_nona[,3]))
taxa_Ethnicity_corrected = bind_cols(taxa_Ethnicity_nona, 
                               as.data.frame(fdrtool(taxa_Ethnicity_nona[,3], 
                                                     statistic = "pvalue", 
                                                     plot = F)))
write.csv(taxa_Ethnicity_corrected, "tables/taxa_oral_wilcox_Ethnicity_corrected_y1.csv")
```

Graph all significant associations
```{r taxa_graphs}
sig_taxa = read.csv("taxa_abundance_fecal_y1.csv", header = T)

taxa = sig_taxa[,1]
plot = list()
for (i in taxa[1:175]) {
  plot[[i]] = ggboxplot(relab_fecal_filtered, x = "Ethnicity", 
                  y = i, color = "Ethnicity", 
                  add = "jitter", 
                  add.params = list(fill = "white"), 
                  ylab = "Relative abundance",
                  title = i) + 
  scale_color_manual(values = c("White" = "#87ceeb", 
                                "Black" = "#f5c4ca"))
}
tiff(file="figures/sig_taxa_plots_fecal_y1.tif", res=150, width=48, height=52, units="in") 
ggarrange(plotlist = plot, nrow = 14, ncol = 13,
          common.legend = T)
dev.off()
```