---
title: "Contig data analysis"
author: "Liz Mallott"
date: "4/2/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Set up environment

```{r libraries}
library(tidyverse)
library(vegan)
library(fdrtool)
library(dendextend)
library(ggtext)
library(cowplot)
```

## Import data

Taxids with taxonomy are imported. Relative abundances of each taxid for individual samples are imported and vertically appended. We then reshape the dataframe. Taxid file and relative abundance file are joined to append taxonomy.

```{r import_taxonomy, echo = F}
taxid = read_tsv("taxonomy_files/tax.txt") %>% 
  select(-contig) %>% 
  distinct()
write_csv(taxid, "tables/unique_taxid.csv")
ra_files = list.files(path = "taxonomy_files", pattern = "run", full.names = T)
rel_abund_long = map_dfr(ra_files, read_tsv)
rel_abund_notax = rel_abund_long %>% pivot_wider(names_from = "id", values_from = "relative")
rel_abund = rel_abund_notax %>% right_join(taxid)
```

Metadata is imported.
```{r import_metadata, echo = F}
metadata = read_tsv("year2_sample_mapping.txt")
metadata2 = metadata %>% 
  mutate(Ethnicity = case_when(Ethnicity == "African_American" ~ "Black", 
                               Ethnicity == "Caucasian" ~ "White"))
```

## Filter data

The relative abundance table is transposed (without taxonomy) and file names are cleaned up. It is then joined with the metadata for PERMANOVA analyses. Individual tables for fecal and oral samples for each study period and for each ethnicity are created.
```{r ra_perm}
rel_abund_t = rel_abund_notax %>% 
  pivot_longer(cols = 2:163, names_to = "file_prefix", values_to = "relative_abundance") %>% 
  pivot_wider(names_from = "taxid", values_from = "relative_abundance") %>% 
  mutate_at("file_prefix", str_replace, "-", "_") %>% 
  left_join(metadata2)
relab_fecal = rel_abund_t %>% 
  filter(fecal_use == 1)
relab_fecal_filtered = relab_fecal %>% 
  mutate(total = rowSums(relab_fecal[,2:9395])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
relab_fecal_filtered_black = relab_fecal_filtered %>% 
  filter(Ethnicity == "Black")
relab_fecal_filtered_white = relab_fecal_filtered %>% 
  filter(Ethnicity == "White")
relab_fecal_filtered_before = relab_fecal_filtered %>% 
  filter(`Study period` == "Before")
relab_fecal_filtered_during = relab_fecal_filtered %>% 
  filter(`Study period` == "During")
relab_fecal_filtered_after = relab_fecal_filtered %>% 
  filter(`Study period` == "After")

relab_oral= rel_abund_t %>% 
  filter(oral_use == 1)
relab_oral_filtered = relab_oral %>% 
  mutate(total = rowSums(relab_oral[,2:9395])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
relab_oral_filtered_black = relab_oral_filtered %>% 
  filter(Ethnicity == "Black")
relab_oral_filtered_white = relab_oral_filtered %>% 
  filter(Ethnicity == "White")
relab_oral_filtered_before = relab_oral_filtered %>% 
  filter(`Study period` == "Before")
relab_oral_filtered_after = relab_oral_filtered %>% 
  filter(`Study period` == "After")
```

The relative abundance table is transposed (with species names but no taxids), column names are cleaned up, low abundance taxa (<1%) are removed, and it is joined with the metadata for individual taxa abundance analyses. Individual tables for fecal and oral samples for each study period and for each ethnicity are created.

```{r ra_taxabund}
rel_abund_tax_t = rel_abund %>% 
  filter(Kingdom == "Bacteria") %>% 
  select(-164) %>% 
  unite("Sci_name", Genus:Species, na.rm = F) %>% 
  pivot_longer(cols = 1:162, names_to = "file_prefix", values_to = "relative_abundance") %>%
  filter(Sci_name != "NA_NA") %>% 
  group_by(Sci_name, file_prefix) %>% 
  summarize(relative_abundance = sum(relative_abundance)) %>% 
  pivot_wider(names_from = "Sci_name", values_from = "relative_abundance") %>% 
  mutate_at("file_prefix", str_replace, "-", "_") %>% 
  select_if(~max(.) >= 0.01) %>% 
  left_join(metadata2)

rel_abund_tax_fecal = rel_abund_tax_t %>% 
  filter(fecal_use == 1) 
rel_abund_tax_fecal_ethnicity = rel_abund_tax_fecal %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_fecal_ethnicity, "tables/relab_fecal.csv")
rel_abund_tax_fecal_before = rel_abund_tax_fecal %>% 
  filter(`Study period` == "Before")
rel_abund_tax_fecal_before_ethnicity = rel_abund_tax_fecal_before %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_fecal_before_ethnicity, "tables/relab_fecal_before.csv")
rel_abund_tax_fecal_during = rel_abund_tax_fecal %>% 
  filter(`Study period` == "During")
rel_abund_tax_fecal_during_ethnicity = rel_abund_tax_fecal_during %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_fecal_during_ethnicity, "tables/relab_fecal_during.csv")
rel_abund_tax_fecal_after = rel_abund_tax_fecal %>% 
  filter(`Study period` == "After")
rel_abund_tax_fecal_after_ethnicity = rel_abund_tax_fecal_after %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_fecal_after_ethnicity, "tables/relab_fecal_after.csv")

rel_abund_tax_oral = rel_abund_tax_t %>% 
  filter(oral_use == 1)
rel_abund_tax_oral_ethnicity = rel_abund_tax_oral %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_oral_ethnicity, "tables/relab_oral.csv")
rel_abund_tax_oral_before = rel_abund_tax_oral %>% 
  filter(`Study period` == "Before")
rel_abund_tax_oral_before_ethnicity = rel_abund_tax_oral_before %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_oral_before_ethnicity, "tables/relab_oral_before.csv")
rel_abund_tax_oral_during = rel_abund_tax_oral %>% 
  filter(`Study period` == "During")
rel_abund_tax_oral_during_ethnicity = rel_abund_tax_oral_during %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_oral_during_ethnicity, "tables/relab_oral_during.csv")
rel_abund_tax_oral_after = rel_abund_tax_oral %>% 
  filter(`Study period` == "After")
rel_abund_tax_oral_after_ethnicity = rel_abund_tax_oral_after %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_oral_after_ethnicity, "tables/relab_oral_after.csv")
```

## PERMANOVAs

Bray-Curtis and Jaccard distance matrices are created using vegan (v2.5-7). Matrices are constructed for all oral samples, all fecal samples, and oral and fecal samples broken down by ethnicity and study stage.

```{r beta_metrics, error=FALSE}
bray_fecal = vegdist(relab_fecal_filtered[,2:9266], method = "bray")
jacc_fecal = vegdist(relab_fecal_filtered[,2:9266], method = "jaccard")
bray_fecal_black = vegdist(relab_fecal_filtered_black[,2:9266], 
                           method = "bray")
bray_fecal_white = vegdist(relab_fecal_filtered_white[,2:9266], 
                           method = "bray")
jacc_fecal_black = vegdist(relab_fecal_filtered_black[,2:9266], 
                           method = "jaccard")
jacc_fecal_white = vegdist(relab_fecal_filtered_white[,2:9266], 
                           method = "jaccard")
bray_fecal_before = vegdist(relab_fecal_filtered_before[,2:9266], 
                           method = "jaccard")
bray_fecal_during = vegdist(relab_fecal_filtered_during[,2:9266], 
                           method = "jaccard")
bray_fecal_after = vegdist(relab_fecal_filtered_after[,2:9266], 
                           method = "jaccard")
jacc_fecal_before = vegdist(relab_fecal_filtered_before[,2:9266], 
                           method = "jaccard")
jacc_fecal_during = vegdist(relab_fecal_filtered_during[,2:9266], 
                           method = "jaccard")
jacc_fecal_after = vegdist(relab_fecal_filtered_after[,2:9266], 
                           method = "jaccard") 

bray_oral = vegdist(relab_oral_filtered[,2:8063], method = "bray")
jacc_oral = vegdist(relab_oral_filtered[,2:272], method = "jaccard")
bray_oral_black = vegdist(relab_oral_filtered_black[,2:8063], method = "bray")
bray_oral_white = vegdist(relab_oral_filtered_white[,2:8063], method = "bray")
jacc_oral_black = vegdist(relab_oral_filtered_black[,2:8063], method = "jaccard")
jacc_oral_white = vegdist(relab_oral_filtered_white[,2:8063], method = "jaccard")
bray_oral_before = vegdist(relab_oral_filtered_before[,2:8063], 
                           method = "jaccard")
bray_oral_after = vegdist(relab_oral_filtered_after[,2:8063], 
                          method = "jaccard")
jacc_oral_before = vegdist(relab_oral_filtered_before[,2:8063], 
                           method = "jaccard")
jacc_oral_after = vegdist(relab_oral_filtered_after[,2:8063], 
                          method = "jaccard")
```

Differences in the taxonomic structure of the community associated with self-reported ethnicity, subject id, and study stage is assessed using permutational multivariate analysis of variance.

#### Bray-Curtis
Fecal
```{r permanova_brayf}
adonis2(bray_fecal ~ subject_id + `Study period` + subject_id*`Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(bray_fecal ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_fecal_filtered, permutations = 4999)

adonis2(bray_fecal ~ Ethnicity, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(bray_fecal_before ~ Ethnicity, 
        data = relab_fecal_filtered_before, 
        permutations = 4999)
adonis2(bray_fecal_during ~ Ethnicity, 
        data = relab_fecal_filtered_during, 
        permutations = 4999)
adonis2(bray_fecal_after ~ Ethnicity, 
        data = relab_fecal_filtered_after, 
        permutations = 4999)

adonis2(bray_fecal ~ `Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(bray_fecal_black ~ `Study period`, 
        data = relab_fecal_filtered_black, 
        permutations = 4999)
adonis2(bray_fecal_white ~ `Study period`, 
        data = relab_fecal_filtered_white, 
        permutations = 4999)
```

Oral
```{r permanova_brayo}
adonis2(bray_oral ~ subject_id + `Study period`, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(bray_oral ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_oral_filtered, permutations = 4999)

adonis2(bray_oral ~ Ethnicity, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(bray_oral_before ~ Ethnicity, 
        data = relab_oral_filtered_before, 
        permutations = 4999)
adonis2(bray_oral_after ~ Ethnicity, 
        data = relab_oral_filtered_after, 
        permutations = 4999)

adonis2(bray_oral ~ `Study period`, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(bray_oral_black ~ `Study period`, 
        data = relab_oral_filtered_black, 
        permutations = 4999)
adonis2(bray_oral_white ~ `Study period`, 
        data = relab_oral_filtered_white, 
        permutations = 4999)

```

#### Jaccard
Fecal
```{r permanova_jaccf}
adonis2(jacc_fecal ~ subject_id + `Study period` + subject_id*`Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_fecal ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_fecal_filtered, permutations = 4999)

adonis2(jacc_fecal ~ Ethnicity, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_fecal_before ~ Ethnicity, 
        data = relab_fecal_filtered_before, 
        permutations = 4999)
adonis2(jacc_fecal_during ~ Ethnicity, 
        data = relab_fecal_filtered_during, 
        permutations = 4999)
adonis2(jacc_fecal_after ~ Ethnicity, 
        data = relab_fecal_filtered_after, 
        permutations = 4999)

adonis2(jacc_fecal ~ `Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_fecal_black ~ `Study period`, 
        data = relab_fecal_filtered_black, 
        permutations = 4999)
adonis2(jacc_fecal_white ~ `Study period`, 
        data = relab_fecal_filtered_white, 
        permutations = 4999)
```

Oral
```{r permanova_jacco}
adonis2(jacc_oral ~ subject_id + `Study period`, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(bray_oral ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_oral_filtered, permutations = 4999)

adonis2(jacc_oral ~ Ethnicity, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(jacc_oral_before ~ Ethnicity, 
        data = relab_oral_filtered_before, 
        permutations = 4999)
adonis2(jacc_oral_after ~ Ethnicity, 
        data = relab_oral_filtered_after, 
        permutations = 4999)

adonis2(jacc_oral ~ `Study period`, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(jacc_oral_black ~ `Study period`, 
        data = relab_oral_filtered_black, 
        permutations = 4999)
adonis2(jacc_oral_white ~ `Study period`, 
        data = relab_oral_filtered_white, 
        permutations = 4999)
```


## Taxonomic abundance

Wilcoxon Rank Sum tests (base R) were used to assess if self-reported Ethnicity is associated with the abundance of individual species identified at the rank of genus or above. P-values are corrected using fdrtool (v.1.2.16).

```{r taxa_models_fecal}
Ethnicity_estimatematrix = mat.or.vec(182,2)
Ethnicity_pvaluematrix = mat.or.vec(182,2)

rel_abund_tax_fecal = as.data.frame(rel_abund_tax_fecal)

for(i in 2:183) {
  variable = rel_abund_tax_fecal[,i]
  b = wilcox.test(variable ~ Ethnicity, 
                  data = rel_abund_tax_fecal,
                  paired = F, exact = F)
  Ethnicity_estimatematrix[i-1,2] = b$statistic
  Ethnicity_pvaluematrix[i-1,2] = b$p.value
  Ethnicity_estimatematrix[i-1,1] = names(rel_abund_tax_fecal)[i]
  Ethnicity_pvaluematrix[i-1,1] = names(rel_abund_tax_fecal)[i]
}


taxa_Ethnicity = bind_cols(as.data.frame(Ethnicity_estimatematrix[,1:2]), 
                     as.data.frame(Ethnicity_pvaluematrix[,2]))
write.csv(taxa_Ethnicity, "tables/taxa_fecal_wilcox_Ethnicity_y2.csv")
taxa_Ethnicity_nona = filter(taxa_Ethnicity, taxa_Ethnicity[,3] != "NaN") 
taxa_Ethnicity_nona[,3] = as.numeric(as.character(taxa_Ethnicity_nona[,3]))
taxa_Ethnicity_corrected = bind_cols(taxa_Ethnicity_nona, 
                               as.data.frame(fdrtool(taxa_Ethnicity_nona[,3], 
                                                     statistic = "pvalue", 
                                                     plot = F)))
write.csv(taxa_Ethnicity_corrected_full, "tables/taxa_fecal_wilcox_Ethnicity_corrected_y2.csv")
```

```{r taxa_models_oral}
Ethnicity_estimatematrix = mat.or.vec(182,2)
Ethnicity_pvaluematrix = mat.or.vec(182,2)

rel_abund_tax_oral = as.data.frame(rel_abund_tax_oral)

for(i in 2:183) {
  variable = rel_abund_tax_oral[,i]
  b = wilcox.test(variable ~ Ethnicity, 
                  data = rel_abund_tax_oral,
                  paired = F, exact = F)
  Ethnicity_estimatematrix[i-1,2] = b$statistic
  Ethnicity_pvaluematrix[i-1,2] = b$p.value
  Ethnicity_estimatematrix[i-1,1] = names(rel_abund_tax_oral)[i]
  Ethnicity_pvaluematrix[i-1,1] = names(rel_abund_tax_oral)[i]
}

taxa_Ethnicity = bind_cols(as.data.frame(Ethnicity_estimatematrix[,1:2]), 
                     as.data.frame(Ethnicity_pvaluematrix[,2]))
write.csv(taxa_Ethnicity, "tables/taxa_oral_wilcox_Ethnicity_y2.csv")
taxa_Ethnicity_nona = filter(taxa_Ethnicity, taxa_Ethnicity[,3] != "NaN") 
taxa_Ethnicity_nona[,3] = as.numeric(as.character(taxa_Ethnicity_nona[,3]))
taxa_Ethnicity_corrected = bind_cols(taxa_Ethnicity_nona, 
                               as.data.frame(fdrtool(taxa_Ethnicity_nona[,3], 
                                                     statistic = "pvalue", 
                                                     plot = F)))
write.csv(taxa_Ethnicity_corrected, "tables/taxa_oral_wilcox_Ethnicity_corrected_y2.csv")
```

## Dendrogram

Import data and filter into fecal and oral metagenomes.

```{r dendro_import}
y1tax = read_tsv("taxon_meta_y1.txt")

y1taxoral = y1tax %>% 
  column_to_rownames("sample.name") %>% 
  filter(source == "oral") %>% 
  dplyr::select(7:9377)

y1taxfecal = y1tax %>% 
  column_to_rownames("sample.name") %>% 
  filter(source == "fecal") %>% 
  dplyr::select(7:9377)

y1vir = read_tsv("virome.txt")
y1virfecal = y1vir %>% 
  column_to_rownames("sample") %>% 
  dplyr::select(5:4429)

y2taxfecal = relab_fecal_filtered %>% 
  column_to_rownames("SampleID")%>% 
  dplyr::select(2:9266)

y2taxoral = relab_oral_filtered %>% 
  column_to_rownames("SampleID")%>% 
  dplyr::select(2:8063)
```
Create labeled dendrograms for fecal and oral samples in each year.

```{r dendro_y1fecal}
y1fecal.dist <- vegdist(y1taxfecal, method = "bray", permutations = 999)
y1fecal.clus <- hclust(y1fecal.dist, "aver")

y1fcd = as.dendrogram(y1fecal.clus) 

y1taxf = y1tax %>% filter(source == "fecal")
Ethnicity = rep("Black", length(rownames(y1taxf)))
is_x = grepl("Caucasian", y1taxf$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y1taxf)))
is_x = grepl("during", y1taxf$stage)
Stage[is_x] = "During"
is_x = grepl("after", y1taxf$stage)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c(After = "#377eb8", Before = "#4daf4a", During = "#e41a1c")
col_Stage = cols_3[Stage]

Subject = rep("1", length(rownames(y1taxfecal)))
is_x = grepl("02", rownames(y1taxfecal))
Subject[is_x] = "2"
is_x = grepl("03", rownames(y1taxfecal))
Subject[is_x] = "3"
is_x = grepl("04", rownames(y1taxfecal))
Subject[is_x] = "4"
is_x = grepl("05", rownames(y1taxfecal))
Subject[is_x] = "5"
is_x = grepl("06", rownames(y1taxfecal))
Subject[is_x] = "6"
is_x = grepl("07", rownames(y1taxfecal))
Subject[is_x] = "7"
is_x = grepl("03", rownames(y1taxfecal))
Subject[is_x] = "3"
is_x = grepl("08", rownames(y1taxfecal))
Subject[is_x] = "8"
is_x = grepl("09", rownames(y1taxfecal))
Subject[is_x] = "9"
is_x = grepl("10", rownames(y1taxfecal))
Subject[is_x] = "10"
is_x = grepl("11", rownames(y1taxfecal))
Subject[is_x] = "11"
is_x = grepl("12", rownames(y1taxfecal))
Subject[is_x] = "12"
is_x = grepl("13", rownames(y1taxfecal))
Subject[is_x] = "13"
is_x = grepl("14", rownames(y1taxfecal))
Subject[is_x] = "14"
is_x = grepl("15", rownames(y1taxfecal))
Subject[is_x] = "15"
is_x = grepl("16", rownames(y1taxfecal))
Subject[is_x] = "16"
is_x = grepl("17", rownames(y1taxfecal))
Subject[is_x] = "17"
is_x = grepl("19", rownames(y1taxfecal))
Subject[is_x] = "19"
is_x = grepl("20", rownames(y1taxfecal))
Subject[is_x] = "20"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 50, l = 70)
col_Subject = cols_19[Subject]
labels_colors(y1fcd) = col_Subject[order.dendrogram(y1fcd)]

svg("figures/y1_fecal_dendrogram.svg", width = 15, height = 10)
y1fcd = assign_values_to_leaves_edgePar(y1fcd, 
                                        value = col_Subject[order.dendrogram(y1fcd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y1fcd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), y1fcd)
dev.off()
```

```{r dendro_y1oral}
y1oral.dist <- vegdist(y1taxoral, method = "bray", permutations = 999)
y1oral.clus <- hclust(y1oral.dist, "aver")

y1ocd = as.dendrogram(y1oral.clus) 

y1taxo = y1tax %>% filter(source == "oral")

Ethnicity = rep("Black", length(rownames(y1taxo)))
is_x = grepl("Caucasian", y1taxo$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y1taxo)))
is_x = grepl("after", y1taxo$stage)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c(After = "#377eb8", Before = "#4daf4a")
col_Stage = cols_3[Stage]

Subject = rep("1", length(rownames(y1taxoral)))
is_x = grepl("02", rownames(y1taxoral))
Subject[is_x] = "2"
is_x = grepl("03", rownames(y1taxoral))
Subject[is_x] = "3"
is_x = grepl("04", rownames(y1taxoral))
Subject[is_x] = "4"
is_x = grepl("05", rownames(y1taxoral))
Subject[is_x] = "5"
is_x = grepl("06", rownames(y1taxoral))
Subject[is_x] = "6"
is_x = grepl("07", rownames(y1taxoral))
Subject[is_x] = "7"
is_x = grepl("08", rownames(y1taxoral))
Subject[is_x] = "8"
is_x = grepl("09", rownames(y1taxoral))
Subject[is_x] = "9"
is_x = grepl("10", rownames(y1taxoral))
Subject[is_x] = "10"
is_x = grepl("11", rownames(y1taxoral))
Subject[is_x] = "11"
is_x = grepl("12", rownames(y1taxoral))
Subject[is_x] = "12"
is_x = grepl("13", rownames(y1taxoral))
Subject[is_x] = "13"
is_x = grepl("14", rownames(y1taxoral))
Subject[is_x] = "14"
is_x = grepl("15", rownames(y1taxoral))
Subject[is_x] = "15"
is_x = grepl("16", rownames(y1taxoral))
Subject[is_x] = "16"
is_x = grepl("17", rownames(y1taxoral))
Subject[is_x] = "17"
is_x = grepl("19", rownames(y1taxoral))
Subject[is_x] = "19"
is_x = grepl("20", rownames(y1taxoral))
Subject[is_x] = "20"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 50, l = 70)
col_Subject = cols_19[Subject]
labels_colors(y1ocd) = col_Subject[order.dendrogram(y1ocd)]

svg("figures/y1_oral_dendrogram.svg", width = 15, height = 10)
y1ocd = assign_values_to_leaves_edgePar(y1ocd, 
                                        value = col_Subject[order.dendrogram(y1ocd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y1ocd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), 
             y1ocd)
dev.off()
```

```{r dendro_y1virus}
y1virus.dist <- vegdist(y1virfecal, method = "bray", permutations = 999)
y1virus.clus <- hclust(y1virus.dist, "aver")

y1vcd = as.dendrogram(y1virus.clus) 

Ethnicity = rep("Black", length(rownames(y1vir)))
is_x = grepl("Caucasian", y1vir$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y1vir)))
is_x = grepl("During", y1vir$Stage)
Stage[is_x] = "During"
is_x = grepl("After", y1vir$Stage)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c("After" = "#377eb8", "Before" = "#4daf4a", "During" = "#e41a1c")
col_Stage = cols_3[Stage]

Subject = rep("1", length(rownames(y1virfecal)))
is_x = grepl("02", rownames(y1virfecal))
Subject[is_x] = "2"
is_x = grepl("03", rownames(y1virfecal))
Subject[is_x] = "3"
is_x = grepl("04", rownames(y1virfecal))
Subject[is_x] = "4"
is_x = grepl("05", rownames(y1virfecal))
Subject[is_x] = "5"
is_x = grepl("06", rownames(y1virfecal))
Subject[is_x] = "6"
is_x = grepl("07", rownames(y1virfecal))
Subject[is_x] = "7"
is_x = grepl("08", rownames(y1virfecal))
Subject[is_x] = "8"
is_x = grepl("09", rownames(y1virfecal))
Subject[is_x] = "9"
is_x = grepl("10", rownames(y1virfecal))
Subject[is_x] = "10"
is_x = grepl("11", rownames(y1virfecal))
Subject[is_x] = "11"
is_x = grepl("12", rownames(y1virfecal))
Subject[is_x] = "12"
is_x = grepl("13", rownames(y1virfecal))
Subject[is_x] = "13"
is_x = grepl("14", rownames(y1virfecal))
Subject[is_x] = "14"
is_x = grepl("15", rownames(y1virfecal))
Subject[is_x] = "15"
is_x = grepl("16", rownames(y1virfecal))
Subject[is_x] = "16"
is_x = grepl("17", rownames(y1virfecal))
Subject[is_x] = "17"
is_x = grepl("19", rownames(y1virfecal))
Subject[is_x] = "19"
is_x = grepl("20", rownames(y1virfecal))
Subject[is_x] = "20"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 50, l = 70)
col_Subject = cols_19[Subject]
labels_colors(y1vcd) = col_Subject[order.dendrogram(y1vcd)]

svg("figures/y1_virus_dendrogram.svg", width = 15, height = 10)
y1vcd = assign_values_to_leaves_edgePar(y1vcd, 
                                        value = col_Subject[order.dendrogram(y1vcd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y1vcd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), 
             y1vcd)
dev.off()
```

```{r dendro_y2fecal}
y2fecal.dist <- vegdist(y2taxfecal, method = "bray", permutations = 999)
y2fecal.clus <- hclust(y2fecal.dist, "aver")

y2fcd = as.dendrogram(y2fecal.clus) 

y2taxf = relab_fecal_filtered %>% 
  dplyr::select(9266:9291)
Ethnicity = rep("Black", length(rownames(y2taxf)))
is_x = grepl("White", y2taxf$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y2taxf)))
is_x = grepl("During", y2taxf$`Study period`)
Stage[is_x] = "During"
is_x = grepl("After", y2taxf$`Study period`)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c("After" = "#377eb8", "Before" = "#4daf4a", "During" = "#e41a1c")
col_Stage = cols_3[Stage]

Subject = rep("21", length(rownames(y2taxfecal)))
is_x = grepl("22", rownames(y2taxfecal))
Subject[is_x] = "22"
is_x = grepl("23", rownames(y2taxfecal))
Subject[is_x] = "23"
is_x = grepl("24", rownames(y2taxfecal))
Subject[is_x] = "24"
is_x = grepl("25", rownames(y2taxfecal))
Subject[is_x] = "25"
is_x = grepl("26", rownames(y2taxfecal))
Subject[is_x] = "26"
is_x = grepl("27", rownames(y2taxfecal))
Subject[is_x] = "27"
is_x = grepl("29", rownames(y2taxfecal))
Subject[is_x] = "29"
is_x = grepl("30", rownames(y2taxfecal))
Subject[is_x] = "30"
is_x = grepl("31", rownames(y2taxfecal))
Subject[is_x] = "31"
is_x = grepl("32", rownames(y2taxfecal))
Subject[is_x] = "32"
is_x = grepl("33", rownames(y2taxfecal))
Subject[is_x] = "33"
is_x = grepl("34", rownames(y2taxfecal))
Subject[is_x] = "34"
is_x = grepl("35", rownames(y2taxfecal))
Subject[is_x] = "35"
is_x = grepl("36", rownames(y2taxfecal))
Subject[is_x] = "36"
is_x = grepl("37", rownames(y2taxfecal))
Subject[is_x] = "37"
is_x = grepl("38", rownames(y2taxfecal))
Subject[is_x] = "38"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 30, l = 50)
col_Subject = cols_19[Subject]
labels_colors(y2fcd) = col_Subject[order.dendrogram(y2fcd)]

svg("figures/y2_fecal_dendrogram.svg", width = 15, height = 10)
y2fcd = assign_values_to_leaves_edgePar(y2fcd, 
                                        value = col_Subject[order.dendrogram(y2fcd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y2fcd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), 
             y2fcd)
dev.off()
```

```{r dendro_y2oral}
y2oral.dist <- vegdist(y2taxoral, method = "bray", permutations = 999)
y2oral.clus <- hclust(y2oral.dist, "aver")

y2ocd = as.dendrogram(y2oral.clus) 

y2taxo = relab_oral_filtered %>% 
  dplyr::select(8064:8088)

Ethnicity = rep("Black", length(rownames(y2taxo)))
is_x = grepl("White", y2taxo$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y2taxo)))
is_x = grepl("After", y2taxo$`Study period`)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c("After" = "#377eb8", "Before" = "#4daf4a")
col_Stage = cols_3[Stage]

Subject = rep("21", length(rownames(y2taxoral)))
is_x = grepl("22", rownames(y2taxoral))
Subject[is_x] = "22"
is_x = grepl("23", rownames(y2taxoral))
Subject[is_x] = "23"
is_x = grepl("24", rownames(y2taxoral))
Subject[is_x] = "24"
is_x = grepl("25", rownames(y2taxoral))
Subject[is_x] = "25"
is_x = grepl("26", rownames(y2taxoral))
Subject[is_x] = "26"
is_x = grepl("27", rownames(y2taxoral))
Subject[is_x] = "27"
is_x = grepl("29", rownames(y2taxoral))
Subject[is_x] = "29"
is_x = grepl("30", rownames(y2taxoral))
Subject[is_x] = "30"
is_x = grepl("31", rownames(y2taxoral))
Subject[is_x] = "31"
is_x = grepl("32", rownames(y2taxoral))
Subject[is_x] = "32"
is_x = grepl("33", rownames(y2taxoral))
Subject[is_x] = "33"
is_x = grepl("34", rownames(y2taxoral))
Subject[is_x] = "34"
is_x = grepl("35", rownames(y2taxoral))
Subject[is_x] = "35"
is_x = grepl("36", rownames(y2taxoral))
Subject[is_x] = "36"
is_x = grepl("37", rownames(y2taxoral))
Subject[is_x] = "37"
is_x = grepl("38", rownames(y2taxoral))
Subject[is_x] = "38"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 30, l = 50)
col_Subject = cols_19[Subject]
labels_colors(y2ocd) = col_Subject[order.dendrogram(y2ocd)]

svg("figures/y2_oral_dendrogram.svg", width = 15, height = 10)
y2ocd = assign_values_to_leaves_edgePar(y2ocd, 
                                        value = col_Subject[order.dendrogram(y2ocd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y2ocd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), 
             y2ocd)
dev.off()
```
## NMDS plots

### Cohort 1 gut taxonomy
```{r nmds_y1ftax}
bray_fecal_y1 = vegdist(y1taxfecal, method = "bray")

y1taxf2 = y1taxf %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White"))

mds_bray_f_y1<-metaMDS(bray_fecal_y1, k=2, trymax=999)
mds_bray_f_y1_points<-mds_bray_f_y1$points
mds_bray_f_y1_points2<-merge(x=mds_bray_f_y1_points, y = y1taxf2, 
                                  by.x = "row.names", 
                             by.y = "sample.name")

brayfy1 <- ggplot(mds_bray_f_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1.2)),
        legend.position = c(0.1, 0.1)) +
  stat_ellipse(data = mds_bray_f_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 6.6% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.134",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfy1
```
### Cohort 2 gut taxonomy
```{r nmds_y2ftax}
bray_fecal_y2 = vegdist(relab_fecal_filtered[,2:9266], method = "bray")

y2taxf = relab_fecal_filtered %>% 
  dplyr::select(9267:9291)

mds_bray_f_y2<-metaMDS(bray_fecal_y2, k=2, trymax=999)
mds_bray_f_y2_points<-mds_bray_f_y2$points
mds_bray_f_y2_points2<-merge(x=mds_bray_f_y2_points, y = y2taxf, 
                                  by.x = "row.names", 
                             by.y = "row.names")

brayfy2 <- ggplot(mds_bray_f_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_f_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 5.5% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.124",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfy2
```

### Cohort 1 oral taxonomy
```{r nmds_y1otax}
bray_oral_y1 = vegdist(y1taxoral, method = "bray")

y1taxo2 = y1taxo %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White"))

mds_bray_o_y1<-metaMDS(bray_oral_y1, k=2, trymax=999)
mds_bray_o_y1_points<-mds_bray_o_y1$points
mds_bray_o_y1_points2<-merge(x=mds_bray_o_y1_points, 
                             y = y1taxo2, 
                                  by.x = "row.names", 
                             by.y = "sample.name")

brayoy1 <- ggplot(mds_bray_o_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 10.4% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.100",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayoy1
```
### Cohort 2 oral taxonomy
```{r nmds_y2otax}
bray_oral_y2 = vegdist(relab_oral_filtered[,2:8063], method = "bray")

y2taxf = relab_oral_filtered %>% 
  dplyr::select(8064:8088)

mds_bray_o_y2<-metaMDS(bray_oral_y2, k=2, trymax=999)
mds_bray_o_y2_points<-mds_bray_o_y2$points
mds_bray_o_y2_points2<-merge(x=mds_bray_o_y2_points, y = y2taxo, 
                                  by.x = "row.names", 
                             by.y = "row.names")

brayoy2 <- ggplot(mds_bray_o_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.022, 
           r<sup>2</sup> = 5.5% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.149",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayoy2
```
### Cohort 1 gut virome taxonomy
```{r nmds_y1fvtax}
vir_tax = read_tsv("Virome_files_y1/virome.txt")

bray_fecalv_y1 = vegdist(vir_tax[,5:4430], method = "bray")

y1taxfv = vir_tax %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White"))

mds_bray_fv_y1<-metaMDS(bray_fecalv_y1, k=2, trymax=999)
mds_bray_fv_y1_points<-mds_bray_fv_y1$points
mds_bray_fv_y1_points2<-merge(x=mds_bray_fv_y1_points, 
                              y = y1taxfv, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfvy1 <- ggplot(mds_bray_fv_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_fv_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 5.2% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.160",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfvy1
```

## COGs

### Import
COGnames with functional categories are imported. Relative abundances of each COG for individual samples are imported and vertically appended. We then reshape the dataframe. COGnames file and relative abundance file are joined to append COG categories

```{r import_cogs, echo = F}
cognames = read_tsv("COG_files/cognames2003-2014.txt") %>% 
  select(-name) %>% 
  distinct()
ra_cog_files = list.files(path = "COG_files", pattern = "run", full.names = T)
rel_abund_cog_long = ra_cog_files %>% 
  set_names() %>% 
  map_dfr(read_tsv, .id = "source") %>% 
  mutate(source2 = stringr::str_replace(source, "COG_files/", "")) %>% 
  mutate(sample = sub('\\.txt$', '', source2))
rel_abund_cog_long_names = rel_abund_cog_long %>% 
  left_join(cognames) %>% 
  dplyr::select(-taxid, -source, -source2)
rel_abund_cog = rel_abund_cog_long_names %>% 
  group_by(func, sample) %>% 
  summarize(relative_abundance = sum(abundance)) %>% 
  pivot_wider(names_from = "sample", values_from = "relative_abundance")
write_tsv(rel_abund_cog, "tables/rel_abund_cog_y2.txt")
```

Metadata is imported.
```{r import_metadata, echo = F}
metadata = read_tsv("year2_sample_mapping.txt")
metadata2 = metadata %>% 
  mutate(Ethnicity = case_when(Ethnicity == "African_American" ~ "Black", 
                               Ethnicity == "Caucasian" ~ "White"))
```

### Filter data

The relative abundance table is transposed and file names are cleaned up. It is then joined with the metadata for PERMANOVA analyses. Individual tables for fecal and oral samples for each study period and for each ethnicity are created.
```{r ra_perm}
rel_abund_cog = read_tsv("tables/rel_abund_cog_y2.txt")
rel_abund_cog_t = rel_abund_cog %>% 
  pivot_longer(cols = 2:163, names_to = "file_prefix", 
               values_to = "abundance") %>% 
  pivot_wider(names_from = "func", values_from = "abundance") %>% 
  mutate_at("file_prefix", str_replace, "-", "_") %>% 
  left_join(metadata2)
relab_cog_fecal = rel_abund_cog_t %>% 
  filter(fecal_use == 1) 
relab_cog_fecal_filtered = relab_cog_fecal %>% 
  mutate_if(is.numeric, ~ (ifelse(is.na(.), 0, .)))
relab_cog_fecal_filtered_black = relab_cog_fecal_filtered %>% 
  filter(Ethnicity == "Black")
relab_cog_fecal_filtered_white = relab_cog_fecal_filtered %>% 
  filter(Ethnicity == "White")
relab_cog_fecal_filtered_before = relab_cog_fecal_filtered %>% 
  filter(`Study period` == "Before")
relab_cog_fecal_filtered_during = relab_cog_fecal_filtered %>% 
  filter(`Study period` == "During")
relab_cog_fecal_filtered_after = relab_cog_fecal_filtered %>% 
  filter(`Study period` == "After")

relab_cog_oral= rel_abund_cog_t %>% 
  filter(oral_use == 1)
relab_cog_oral_filtered = relab_cog_oral %>% 
  mutate_if(is.numeric, ~ (ifelse(is.na(.), 0, .)))
relab_cog_oral_filtered_black = relab_cog_oral_filtered %>% 
  filter(Ethnicity == "Black")
relab_cog_oral_filtered_white = relab_cog_oral_filtered %>% 
  filter(Ethnicity == "White")
relab_cog_oral_filtered_before = relab_cog_oral_filtered %>% 
  filter(`Study period` == "Before")
relab_cog_oral_filtered_after = relab_cog_oral_filtered %>% 
  filter(`Study period` == "After")
```

### PERMANOVAs

Bray-Curtis and Jaccard distance matrices are created using vegan (v2.5-7). Matrices are constructed for all oral samples, all fecal samples, and oral and fecal samples broken down by ethnicity and study stage.

```{r beta_metrics, error=FALSE}
bray_cog_fecal = vegdist(relab_cog_fecal_filtered[,2:121], method = "bray")
jacc_cog_fecal = vegdist(relab_cog_fecal_filtered[,2:121], method = "jaccard")
bray_cog_fecal_black = vegdist(relab_cog_fecal_filtered_black[,2:121], 
                           method = "bray")
bray_cog_fecal_white = vegdist(relab_cog_fecal_filtered_white[,2:121], 
                           method = "bray")
jacc_cog_fecal_black = vegdist(relab_cog_fecal_filtered_black[,2:121], 
                           method = "jaccard")
jacc_cog_fecal_white = vegdist(relab_cog_fecal_filtered_white[,2:121], 
                           method = "jaccard")
bray_cog_fecal_before = vegdist(relab_cog_fecal_filtered_before[,2:121], 
                           method = "jaccard")
bray_cog_fecal_during = vegdist(relab_cog_fecal_filtered_during[,2:121], 
                           method = "jaccard")
bray_cog_fecal_after = vegdist(relab_cog_fecal_filtered_after[,2:121], 
                           method = "jaccard")
jacc_cog_fecal_before = vegdist(relab_cog_fecal_filtered_before[,2:121], 
                           method = "jaccard")
jacc_cog_fecal_during = vegdist(relab_cog_fecal_filtered_during[,2:121], 
                           method = "jaccard")
jacc_cog_fecal_after = vegdist(relab_cog_fecal_filtered_after[,2:121], 
                           method = "jaccard") 

bray_cog_oral = vegdist(relab_cog_oral_filtered[,2:121], method = "bray")
jacc_cog_oral = vegdist(relab_cog_oral_filtered[,2:121], method = "jaccard")
bray_cog_oral_black = vegdist(relab_cog_oral_filtered_black[,2:121], method = "bray")
bray_cog_oral_white = vegdist(relab_cog_oral_filtered_white[,2:121], method = "bray")
jacc_cog_oral_black = vegdist(relab_cog_oral_filtered_black[,2:121], method = "jaccard")
jacc_cog_oral_white = vegdist(relab_cog_oral_filtered_white[,2:121], method = "jaccard")
bray_cog_oral_before = vegdist(relab_cog_oral_filtered_before[,2:121], 
                           method = "jaccard")
bray_cog_oral_after = vegdist(relab_cog_oral_filtered_after[,2:121], 
                          method = "jaccard")
jacc_cog_oral_before = vegdist(relab_cog_oral_filtered_before[,2:121], 
                           method = "jaccard")
jacc_cog_oral_after = vegdist(relab_cog_oral_filtered_after[,2:121], 
                          method = "jaccard")
```

Differences in the COG profile of the community associated with self-reported ethnicity, subject id, and study stage is assessed using permutational multivariate analysis of variance.

#### Bray-Curtis
cog_fecal
```{r permanova_brayf}
adonis2(bray_cog_fecal ~ subject_id + `Study period` + subject_id*`Study period`, 
        data = relab_cog_fecal_filtered, 
        permutations = 4999)
adonis2(bray_cog_fecal ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_cog_fecal_filtered, permutations = 4999)

adonis2(bray_cog_fecal ~ Ethnicity, 
        data = relab_cog_fecal_filtered, 
        permutations = 4999)
adonis2(bray_cog_fecal_before ~ Ethnicity, 
        data = relab_cog_fecal_filtered_before, 
        permutations = 4999)
adonis2(bray_cog_fecal_during ~ Ethnicity, 
        data = relab_cog_fecal_filtered_during, 
        permutations = 4999)
adonis2(bray_cog_fecal_after ~ Ethnicity, 
        data = relab_cog_fecal_filtered_after, 
        permutations = 4999)

adonis2(bray_cog_fecal ~ `Study period`, 
        data = relab_cog_fecal_filtered, 
        permutations = 4999)
adonis2(bray_cog_fecal_black ~ `Study period`, 
        data = relab_cog_fecal_filtered_black, 
        permutations = 4999)
adonis2(bray_cog_fecal_white ~ `Study period`, 
        data = relab_cog_fecal_filtered_white, 
        permutations = 4999)
```

cog_oral
```{r permanova_brayo}
adonis2(bray_cog_oral ~ subject_id + `Study period`, 
        data = relab_cog_oral_filtered, 
        permutations = 4999)
adonis2(bray_cog_oral ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_cog_oral_filtered, permutations = 4999)

adonis2(bray_cog_oral ~ Ethnicity, 
        data = relab_cog_oral_filtered, 
        permutations = 4999)
adonis2(bray_cog_oral_before ~ Ethnicity, 
        data = relab_cog_oral_filtered_before, 
        permutations = 4999)
adonis2(bray_cog_oral_after ~ Ethnicity, 
        data = relab_cog_oral_filtered_after, 
        permutations = 4999)

adonis2(bray_cog_oral ~ `Study period`, 
        data = relab_cog_oral_filtered, 
        permutations = 4999)
adonis2(bray_cog_oral_black ~ `Study period`, 
        data = relab_cog_oral_filtered_black, 
        permutations = 4999)
adonis2(bray_cog_oral_white ~ `Study period`, 
        data = relab_cog_oral_filtered_white, 
        permutations = 4999)

```

#### Jaccard
cog_fecal
```{r permanova_jaccf}
adonis2(jacc_cog_fecal ~ subject_id + `Study period` + subject_id*`Study period`, 
        data = relab_cog_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_cog_fecal ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_cog_fecal_filtered, permutations = 4999)

adonis2(jacc_cog_fecal ~ Ethnicity, 
        data = relab_cog_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_cog_fecal_before ~ Ethnicity, 
        data = relab_cog_fecal_filtered_before, 
        permutations = 4999)
adonis2(jacc_cog_fecal_during ~ Ethnicity, 
        data = relab_cog_fecal_filtered_during, 
        permutations = 4999)
adonis2(jacc_cog_fecal_after ~ Ethnicity, 
        data = relab_cog_fecal_filtered_after, 
        permutations = 4999)

adonis2(jacc_cog_fecal ~ `Study period`, 
        data = relab_cog_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_cog_fecal_black ~ `Study period`, 
        data = relab_cog_fecal_filtered_black, 
        permutations = 4999)
adonis2(jacc_cog_fecal_white ~ `Study period`, 
        data = relab_cog_fecal_filtered_white, 
        permutations = 4999)
```

cog_oral
```{r permanova_jacco}
adonis2(jacc_cog_oral ~ subject_id + `Study period`, 
        data = relab_cog_oral_filtered, 
        permutations = 4999)
adonis2(bray_cog_oral ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_cog_oral_filtered, permutations = 4999)

adonis2(jacc_cog_oral ~ Ethnicity, 
        data = relab_cog_oral_filtered, 
        permutations = 4999)
adonis2(jacc_cog_oral_before ~ Ethnicity, 
        data = relab_cog_oral_filtered_before, 
        permutations = 4999)
adonis2(jacc_cog_oral_after ~ Ethnicity, 
        data = relab_cog_oral_filtered_after, 
        permutations = 4999)

adonis2(jacc_cog_oral ~ `Study period`, 
        data = relab_cog_oral_filtered, 
        permutations = 4999)
adonis2(jacc_cog_oral_black ~ `Study period`, 
        data = relab_cog_oral_filtered_black, 
        permutations = 4999)
adonis2(jacc_cog_oral_white ~ `Study period`, 
        data = relab_cog_oral_filtered_white, 
        permutations = 4999)
```

## ARGs

### Import
Import individual relative abundance files. Collect and clean strings. Summarize based on ARG functional categories.
````{r import_args}
data_dir <- here("/Users/goblinking/Projects/ko_args/args/arg_ra")
class <- read_tsv("/Users/goblinking/Projects/ko_args/args/other/arg.category.txt") %>% 
    rename(.,RF= ResfamID)
data_list <- fs::dir_ls(data_dir, regexp = ".txt$")
data <-  data_list %>% 
  map_dfr(read_tsv, .id = "source") %>% 
  mutate(source2 = stringr::str_replace(source, "/Users/goblinking/Projects/ko_args/args/arg_ra/", "")) %>% 
  mutate(sample = sub('\\.txt$', '', source2)) %>% 
  left_join(.,class, by="RF") %>% 
  select(-source,-source2, -RF, -taxid)
  
data_abund <- data %>% rename(mechanism =`Mechanism Classification`) %>% 
  group_by(mechanism,sample) %>% 
  summarize(total = sum(abundance)) %>% 
  spread(mechanism,total) %>% 
  write_tsv("wide_arg_c2_contig.tsv")

rel_abund_arg = read_tsv("ARG_files/args_c2_contig.txt")
```

Metadata is imported.
```{r import_metadata, echo = F}
metadata = read_tsv("year2_sample_mapping.txt")
metadata2 = metadata %>% 
  mutate(Ethnicity = case_when(Ethnicity == "African_American" ~ "Black", 
                               Ethnicity == "Caucasian" ~ "White"))
```

### Filter data

The relative abundance table is transposed and file names are cleaned up. It is then joined with the metadata for PERMANOVA analyses. Individual tables for fecal and oral samples for each study period and for each ethnicity are created.
```{r ra_perm}
rel_abund_arg = read_tsv("ARG_files/args_c2_contig.txt")

rel_abund_arg_meta = rel_abund_arg %>% 
  left_join(metadata2, by = c("Sample" = "SampleID"))

relab_arg_fecal = rel_abund_arg_meta %>% 
  filter(fecal_use == 1) 
relab_arg_fecal_filtered = relab_arg_fecal %>% 
  mutate_if(is.numeric, ~ (ifelse(is.na(.), 0, .)))
relab_arg_fecal_filtered_black = relab_arg_fecal_filtered %>% 
  filter(Ethnicity == "Black")
relab_arg_fecal_filtered_white = relab_arg_fecal_filtered %>% 
  filter(Ethnicity == "White")
relab_arg_fecal_filtered_before = relab_arg_fecal_filtered %>% 
  filter(`Study period` == "Before")
relab_arg_fecal_filtered_during = relab_arg_fecal_filtered %>% 
  filter(`Study period` == "During")
relab_arg_fecal_filtered_after = relab_arg_fecal_filtered %>% 
  filter(`Study period` == "After")

relab_arg_oral= rel_abund_arg_meta %>% 
  filter(oral_use == 1)
relab_arg_oral_filtered = relab_arg_oral %>% 
  mutate_if(is.numeric, ~ (ifelse(is.na(.), 0, .)))
relab_arg_oral_filtered_black = relab_arg_oral_filtered %>% 
  filter(Ethnicity == "Black")
relab_arg_oral_filtered_white = relab_arg_oral_filtered %>% 
  filter(Ethnicity == "White")
relab_arg_oral_filtered_before = relab_arg_oral_filtered %>% 
  filter(`Study period` == "Before")
relab_arg_oral_filtered_after = relab_arg_oral_filtered %>% 
  filter(`Study period` == "After")
```

### PERMANOVAs

Bray-Curtis and Jaccard distance matrices are created using vegan (v2.5-7). Matrices are constructed for all oral samples, all fecal samples, and oral and fecal samples broken down by ethnicity and study stage.

```{r beta_metrics, error=FALSE}
bray_arg_fecal = vegdist(relab_arg_fecal_filtered[,2:22], method = "bray")
jacc_arg_fecal = vegdist(relab_arg_fecal_filtered[,2:22], method = "jaccard")
bray_arg_fecal_black = vegdist(relab_arg_fecal_filtered_black[,2:22], 
                           method = "bray")
bray_arg_fecal_white = vegdist(relab_arg_fecal_filtered_white[,2:22], 
                           method = "bray")
jacc_arg_fecal_black = vegdist(relab_arg_fecal_filtered_black[,2:22], 
                           method = "jaccard")
jacc_arg_fecal_white = vegdist(relab_arg_fecal_filtered_white[,2:22], 
                           method = "jaccard")
bray_arg_fecal_before = vegdist(relab_arg_fecal_filtered_before[,2:22], 
                           method = "jaccard")
bray_arg_fecal_during = vegdist(relab_arg_fecal_filtered_during[,2:22], 
                           method = "jaccard")
bray_arg_fecal_after = vegdist(relab_arg_fecal_filtered_after[,2:22], 
                           method = "jaccard")
jacc_arg_fecal_before = vegdist(relab_arg_fecal_filtered_before[,2:22], 
                           method = "jaccard")
jacc_arg_fecal_during = vegdist(relab_arg_fecal_filtered_during[,2:22], 
                           method = "jaccard")
jacc_arg_fecal_after = vegdist(relab_arg_fecal_filtered_after[,2:22], 
                           method = "jaccard") 

bray_arg_oral = vegdist(relab_arg_oral_filtered[,2:22], method = "bray")
jacc_arg_oral = vegdist(relab_arg_oral_filtered[,2:22], method = "jaccard")
bray_arg_oral_black = vegdist(relab_arg_oral_filtered_black[,2:22], method = "bray")
bray_arg_oral_white = vegdist(relab_arg_oral_filtered_white[,2:22], method = "bray")
jacc_arg_oral_black = vegdist(relab_arg_oral_filtered_black[,2:22], method = "jaccard")
jacc_arg_oral_white = vegdist(relab_arg_oral_filtered_white[,2:22], method = "jaccard")
bray_arg_oral_before = vegdist(relab_arg_oral_filtered_before[,2:22], 
                           method = "jaccard")
bray_arg_oral_after = vegdist(relab_arg_oral_filtered_after[,2:22], 
                          method = "jaccard")
jacc_arg_oral_before = vegdist(relab_arg_oral_filtered_before[,2:22], 
                           method = "jaccard")
jacc_arg_oral_after = vegdist(relab_arg_oral_filtered_after[,2:22], 
                          method = "jaccard")
```

Differences in the ARG profile of the community associated with self-reported ethnicity, subject id, and study stage is assessed using permutational multivariate analysis of variance.

#### Bray-Curtis
arg_fecal
```{r permanova_brayf}
adonis2(bray_arg_fecal ~ subject_id + `Study period` + subject_id*`Study period`, 
        data = relab_arg_fecal_filtered, 
        permutations = 4999)
adonis2(bray_arg_fecal ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_arg_fecal_filtered, permutations = 4999)

adonis2(bray_arg_fecal ~ Ethnicity, 
        data = relab_arg_fecal_filtered, 
        permutations = 4999)
adonis2(bray_arg_fecal_before ~ Ethnicity, 
        data = relab_arg_fecal_filtered_before, 
        permutations = 4999)
adonis2(bray_arg_fecal_during ~ Ethnicity, 
        data = relab_arg_fecal_filtered_during, 
        permutations = 4999)
adonis2(bray_arg_fecal_after ~ Ethnicity, 
        data = relab_arg_fecal_filtered_after, 
        permutations = 4999)

adonis2(bray_arg_fecal ~ `Study period`, 
        data = relab_arg_fecal_filtered, 
        permutations = 4999)
adonis2(bray_arg_fecal_black ~ `Study period`, 
        data = relab_arg_fecal_filtered_black, 
        permutations = 4999)
adonis2(bray_arg_fecal_white ~ `Study period`, 
        data = relab_arg_fecal_filtered_white, 
        permutations = 4999)
```

arg_oral
```{r permanova_brayo}
adonis2(bray_arg_oral ~ subject_id + `Study period`, 
        data = relab_arg_oral_filtered, 
        permutations = 4999)
adonis2(bray_arg_oral ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_arg_oral_filtered, permutations = 4999)

adonis2(bray_arg_oral ~ Ethnicity, 
        data = relab_arg_oral_filtered, 
        permutations = 4999)
adonis2(bray_arg_oral_before ~ Ethnicity, 
        data = relab_arg_oral_filtered_before, 
        permutations = 4999)
adonis2(bray_arg_oral_after ~ Ethnicity, 
        data = relab_arg_oral_filtered_after, 
        permutations = 4999)

adonis2(bray_arg_oral ~ `Study period`, 
        data = relab_arg_oral_filtered, 
        permutations = 4999)
adonis2(bray_arg_oral_black ~ `Study period`, 
        data = relab_arg_oral_filtered_black, 
        permutations = 4999)
adonis2(bray_arg_oral_white ~ `Study period`, 
        data = relab_arg_oral_filtered_white, 
        permutations = 4999)

```

#### Jaccard
arg_fecal
```{r permanova_jaccf}
adonis2(jacc_arg_fecal ~ subject_id + `Study period` + subject_id*`Study period`, 
        data = relab_arg_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_arg_fecal ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_arg_fecal_filtered, permutations = 4999)

adonis2(jacc_arg_fecal ~ Ethnicity, 
        data = relab_arg_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_arg_fecal_before ~ Ethnicity, 
        data = relab_arg_fecal_filtered_before, 
        permutations = 4999)
adonis2(jacc_arg_fecal_during ~ Ethnicity, 
        data = relab_arg_fecal_filtered_during, 
        permutations = 4999)
adonis2(jacc_arg_fecal_after ~ Ethnicity, 
        data = relab_arg_fecal_filtered_after, 
        permutations = 4999)

adonis2(jacc_arg_fecal ~ `Study period`, 
        data = relab_arg_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_arg_fecal_black ~ `Study period`, 
        data = relab_arg_fecal_filtered_black, 
        permutations = 4999)
adonis2(jacc_arg_fecal_white ~ `Study period`, 
        data = relab_arg_fecal_filtered_white, 
        permutations = 4999)
```

arg_oral
```{r permanova_jacco}
adonis2(jacc_arg_oral ~ subject_id + `Study period`, 
        data = relab_arg_oral_filtered, 
        permutations = 4999)
adonis2(bray_arg_oral ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_arg_oral_filtered, permutations = 4999)

adonis2(jacc_arg_oral ~ Ethnicity, 
        data = relab_arg_oral_filtered, 
        permutations = 4999)
adonis2(jacc_arg_oral_before ~ Ethnicity, 
        data = relab_arg_oral_filtered_before, 
        permutations = 4999)
adonis2(jacc_arg_oral_after ~ Ethnicity, 
        data = relab_arg_oral_filtered_after, 
        permutations = 4999)

adonis2(jacc_arg_oral ~ `Study period`, 
        data = relab_arg_oral_filtered, 
        permutations = 4999)
adonis2(jacc_arg_oral_black ~ `Study period`, 
        data = relab_arg_oral_filtered_black, 
        permutations = 4999)
adonis2(jacc_arg_oral_white ~ `Study period`, 
        data = relab_arg_oral_filtered_white, 
        permutations = 4999)
```

## NMDS plots

### Cohort 1 fecal COGs
```{r nmds_y1fcog}
cogfy1 = read_tsv("year1_files/cog.txt")
metacogy1 = read_tsv("year1_sample_mapping.tsv") %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White")) %>% 
  left_join(cogfy1, by = c("file_prefix" = "id")) %>% 
  filter(fecal_use == 1)

bray_fecal_cog_y1 = vegdist(metacogy1[,26:146], method = "bray")


mds_bray_f_cog_y1<-metaMDS(bray_fecal_cog_y1, k=2, trymax=999)
mds_bray_f_cog_y1_points<-mds_bray_f_cog_y1$points
mds_bray_f_cog_y1_points2<-merge(x=mds_bray_f_cog_y1_points, 
                              y = metacogy1, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfcogy1 <- ggplot(mds_bray_f_cog_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_f_cog_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.002, 
           r<sup>2</sup> = 9.2% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.025",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfcogy1
```

### Cohort 2 fecal COGs
```{r nmds_y2fcog}
bray_fecal_cog_y2 = vegdist(relab_cog_fecal_filtered[,2:121], method = "bray")

mds_bray_f_cog_y2<-metaMDS(bray_fecal_cog_y2, k=2, trymax=999)
mds_bray_f_cog_y2_points<-mds_bray_f_cog_y2$points
mds_bray_f_cog_y2_points2<-merge(x=mds_bray_f_cog_y2_points, 
                              y = relab_cog_fecal_filtered, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfcogy2 <- ggplot(mds_bray_f_cog_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_f_cog_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.146, 
           r<sup>2</sup> = 1.8%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.075",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfcogy2
```

### Cohort 1 oral COGs
```{r nmds_y1ocog}
cogoy1 = read_tsv("year1_files/cog.txt")
metacogoy1 = read_tsv("year1_sample_mapping.tsv") %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White")) %>% 
  left_join(cogoy1, by = c("file_prefix" = "id")) %>% 
  filter(oral_use == 1)

bray_oral_cog_y1 = vegdist(metacogoy1[,26:146], method = "bray")


mds_bray_o_cog_y1<-metaMDS(bray_oral_cog_y1, k=2, trymax=999)
mds_bray_o_cog_y1_points<-mds_bray_o_cog_y1$points
mds_bray_o_cog_y1_points2<-merge(x=mds_bray_o_cog_y1_points, 
                              y = metacogoy1, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayocogy1 <- ggplot(mds_bray_o_cog_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_cog_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.031, 
           r<sup>2</sup> = 8.7% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.015",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayocogy1
```

### Cohort 2 oral COGs
```{r nmds_y2ocog}
bray_oral_cog_y2 = vegdist(relab_cog_oral_filtered[,2:121], method = "bray")

mds_bray_o_cog_y2<-metaMDS(bray_oral_cog_y2, k=2, trymax=999)
mds_bray_o_cog_y2_points<-mds_bray_o_cog_y2$points
mds_bray_o_cog_y2_points2<-merge(x=mds_bray_o_cog_y2_points, 
                              y = relab_cog_oral_filtered, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayocogy2 <- ggplot(mds_bray_o_cog_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_cog_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.269, 
           r<sup>2</sup> = 3.8%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.011",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayocogy2
```

### Cohort 1 virome COGs
```{r nmds_y1fvcog}
vir_cog = read_tsv("Virome_files_y1/virome_cog_y1.txt")

bray_fecalvcog_y1 = vegdist(vir_cog[,5:20], method = "bray")

y1cogfv = vir_cog %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White"))

mds_bray_fvcog_y1<-metaMDS(bray_fecalvcog_y1, k=2, trymax=999)
mds_bray_fvcog_y1_points<-mds_bray_fvcog_y1$points
mds_bray_fvcog_y1_points2<-merge(x=mds_bray_fvcog_y1_points, 
                              y = y1cogfv, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfvcogy1 <- ggplot(mds_bray_fvcog_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_fvcog_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 5.6% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.190",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfvcogy1
```

### Cohort 1 fecal ARGs
```{r nmds_y1farg}
argfy1 = read_tsv("year1_files/args_noout.txt")
metaargy1 = read_tsv("year1_sample_mapping.tsv") %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White")) %>% 
  right_join(argfy1, by = c("file_prefix" = "sample")) %>% 
  filter(fecal_use == 1)

bray_fecal_arg_y1 = vegdist(metaargy1[,34:55], method = "bray")

mds_bray_f_arg_y1<-metaMDS(bray_fecal_arg_y1, k=2, trymax=999)
mds_bray_f_arg_y1_points<-mds_bray_f_arg_y1$points
mds_bray_f_arg_y1_points2<-merge(x=mds_bray_f_arg_y1_points, 
                              y = metaargy1, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfargy1 <- ggplot(mds_bray_f_arg_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity.x, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_f_arg_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity.x), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.004, 
           r<sup>2</sup> = 7.5% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.051",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfargy1
```
### Cohort 2 fecal ARGs
```{r nmds_y2farg}
bray_fecal_arg_y2 = vegdist(relab_arg_fecal_filtered[,2:22], method = "bray")

adonis2(bray_fecal_arg_y2 ~ Ethnicity, data = relab_arg_fecal_filtered, permutations = 4999)

mds_bray_f_arg_y2<-metaMDS(bray_fecal_arg_y2, k=2, trymax=999)
mds_bray_f_arg_y2_points<-mds_bray_f_arg_y2$points
mds_bray_f_arg_y2_points2<-merge(x=mds_bray_f_arg_y2_points, 
                              y = relab_arg_fecal_filtered, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfargy2 <- ggplot(mds_bray_f_arg_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_f_arg_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.536, 
           r<sup>2</sup> = 0.6%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.060",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfargy2
```

### Cohort 1 oral ARGs
```{r nmds_y1oarg}
argoy1 = read_tsv("year1_files/args_noout.txt")
metaargoy1 = read_tsv("year1_sample_mapping.tsv") %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White")) %>% 
  right_join(argoy1, by = c("file_prefix" = "sample")) %>% 
  filter(oral_use == 1)

bray_oral_arg_y1 = vegdist(metaargoy1[,34:55], method = "bray")

mds_bray_o_arg_y1<-metaMDS(bray_oral_arg_y1, k=2, trymax=999)
mds_bray_o_arg_y1_points<-mds_bray_o_arg_y1$points
mds_bray_o_arg_y1_points2<-merge(x=mds_bray_o_arg_y1_points, 
                              y = metaargoy1, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayoargy1 <- ggplot(mds_bray_o_arg_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity.x, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_arg_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity.x), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.001, 
           r<sup>2</sup> = 10.4% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.026",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayoargy1
```
### Cohort 2 oral ARGs
```{r nmds_y2oarg}
bray_oral_arg_y2 = vegdist(relab_arg_oral_filtered[,2:22], method = "bray")

adonis2(bray_oral_arg_y2 ~ Ethnicity, data = relab_arg_oral_filtered, permutations = 4999)

mds_bray_o_arg_y2<-metaMDS(bray_oral_arg_y2, k=2, trymax=999)
mds_bray_o_arg_y2_points<-mds_bray_o_arg_y2$points
mds_bray_o_arg_y2_points2<-merge(x=mds_bray_o_arg_y2_points, 
                              y = relab_arg_oral_filtered, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayoargy2 <- ggplot(mds_bray_o_arg_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_arg_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.391, 
           r<sup>2</sup> = 2.5%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.025",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayoargy2
```
### Cohort 1 virome ARGs
```{r nmds_y1fvarg}
vir_arg = read_tsv("Virome_files_y1/virome_arg_y1.txt")

bray_fecalvarg_y1 = vegdist(vir_arg[,5:20], method = "bray")

y1argfv = vir_arg %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White"))

mds_bray_fvarg_y1<-metaMDS(bray_fecalvarg_y1, k=2, trymax=999)
mds_bray_fvarg_y1_points<-mds_bray_fvarg_y1$points
mds_bray_fvarg_y1_points2<-merge(x=mds_bray_fvarg_y1_points, 
                              y = y1argfv, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfvargy1 <- ggplot(mds_bray_fvarg_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_fvarg_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 5.6% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.190",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfvargy1
```

### Cohort 1 fecal CAZymes
```{r nmds_y1fcaz}
cazfy1 = read_tsv("year1_files/cazyme_noout.txt")
metafcazy1 = read_tsv("year1_sample_mapping.tsv") %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White")) %>% 
  right_join(cazfy1, by = c("file_prefix" = "sample")) %>% 
  filter(fecal_use == 1)

bray_fecal_caz_y1 = vegdist(metafcazy1[,34:58], method = "bray")

mds_bray_f_caz_y1<-metaMDS(bray_fecal_caz_y1, k=2, trymax=999)
mds_bray_f_caz_y1_points<-mds_bray_f_caz_y1$points
mds_bray_f_caz_y1_points2<-merge(x=mds_bray_f_caz_y1_points, 
                              y = metafcazy1, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfcazy1 <- ggplot(mds_bray_f_caz_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity.x, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_f_caz_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity.x), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.130, 
           r<sup>2</sup> = 1.9%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.040",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfcazy1
```

### Cohort 2 fecal CAZymes
```{r nmds_y2fcaz}
relab_caz_fecal_filtered =  read_tsv("CAZyme_files/caz_c2_contig.tsv") %>%
  left_join(metadata2, by = c("sample" = "SampleID")) %>% 
  filter(fecal_use == 1)

bray_fecal_caz_y2 = vegdist(relab_caz_fecal_filtered[,2:26], method = "bray")

adonis2(bray_fecal_caz_y2 ~ Ethnicity, data = relab_caz_fecal_filtered, permutations = 4999)

mds_bray_f_caz_y2<-metaMDS(bray_fecal_caz_y2, k=2, trymax=999)
mds_bray_f_caz_y2_points<-mds_bray_f_caz_y2$points
mds_bray_f_caz_y2_points2<-merge(x=mds_bray_f_caz_y2_points, 
                              y = relab_caz_fecal_filtered, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfcazy2 <- ggplot(mds_bray_f_caz_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_f_caz_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.154, 
           r<sup>2</sup> = 1.7%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.049",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfcazy2
```
### Cohort 1 oral CAZymes
```{r nmds_y1ocaz}
cazoy1 = read_tsv("year1_files/cazyme_noout.txt")
metaocazy1 = read_tsv("year1_sample_mapping.tsv") %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White")) %>% 
  right_join(cazoy1, by = c("file_prefix" = "sample")) %>% 
  filter(oral_use == 1)

bray_oral_caz_y1 = vegdist(metaocazy1[,34:58], method = "bray")

mds_bray_o_caz_y1<-metaMDS(bray_oral_caz_y1, k=2, trymax=999)
mds_bray_o_caz_y1_points<-mds_bray_o_caz_y1$points
mds_bray_o_caz_y1_points2<-merge(x=mds_bray_o_caz_y1_points, 
                              y = metaocazy1, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayocazy1 <- ggplot(mds_bray_o_caz_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity.x, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_caz_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity.x), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 29.4%</b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.026",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayocazy1
```

### Cohort 2 oral CAZymes

```{r nmds_y2ocaz}
relab_caz_oral_filtered =  read_tsv("CAZyme_files/caz_c2_contig.tsv") %>%
  left_join(metadata2, by = c("sample" = "SampleID")) %>% 
  filter(oral_use == 1)

bray_oral_caz_y2 = vegdist(relab_caz_oral_filtered[,2:26], method = "bray")

adonis2(bray_oral_caz_y2 ~ Ethnicity, data = relab_caz_oral_filtered, permutations = 4999)

mds_bray_o_caz_y2<-metaMDS(bray_oral_caz_y2, k=2, trymax=999)
mds_bray_o_caz_y2_points<-mds_bray_o_caz_y2$points
mds_bray_o_caz_y2_points2<-merge(x=mds_bray_o_caz_y2_points, 
                              y = relab_caz_oral_filtered, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayocazy2 <- ggplot(mds_bray_o_caz_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_caz_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.046, 
           r<sup>2</sup> = 9.9%</b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.072",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayocazy2
```

### Cohort 1 virome CAZymes
```{r nmds_y1fvcaz}
vir_caz = read_tsv("Virome_files_y1/virome_cazymes_y1.txt")

bray_fecalvcaz_y1 = vegdist(vir_caz[,5:22], method = "bray")

y1cazfv = vir_caz %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White"))

mds_bray_fvcaz_y1<-metaMDS(bray_fecalvcaz_y1, k=2, trymax=999)
mds_bray_fvcaz_y1_points<-mds_bray_fvcaz_y1$points
mds_bray_fvcaz_y1_points2<-merge(x=mds_bray_fvcaz_y1_points, 
                              y = y1cazfv, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfvcazy1 <- ggplot(mds_bray_fvcaz_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity, inherit.aes = F)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_fvcaz_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0.2, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.005, 
           r<sup>2</sup> = 4.1% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.170",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfvcazy1
```

### Combining NMDS plots
```{r figure2}
svg(file = "figures/figure2.svg", width = 8,
     height = 10)
col1 = plot_grid(brayfy1, brayoy1, brayfvy1,  
                   brayfy2, brayoy2,
                 nrow = 5, ncol = 1, 
                 labels = c("A", "E", "I", "M", "Q"), 
                 label_size = 20)
col2 = plot_grid(brayfcogy1 + brayocogy1 + brayvcogy1 + 
                   brayfcogy2 + brayocogy2,
                 nrow = 5, ncol = 1, 
                 labels = c("B", "F", "J", "N", "R"), 
                 label_size = 20)
##Note - brayocogy1 should be replaced with brayvcogy1 when available
col3 = plot_grid(brayfargy1 + brayoargy1 + brayvargy1 +
                   brayfargy2 + brayoargy2,
                 nrow = 5, ncol = 1, 
                 labels = c("C", "G", "K", "O", "S"), 
                 label_size = 20)
col4 = plot_grid(brayfcazy1 + brayocazy2 + brayfvcazy1 +
                   brayfcazy2 + brayocazy2,
                 nrow = 5, ncol = 1, 
                 labels = c("D", "H", "L", "P", "T"), 
                 label_size = 20)
plot = plot_grid(col1, col2, col3, col4, nrow = 1, ncol = 4, 
          rel_widths = c(1, 1, 1, 1), 
          align = "h", axis = "t")
plot
dev.off()
```
