---
title: "Taxonomy Contigs"
author: "Liz Mallott"
date: "3/29/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Set up environment

```{r libraries}
library(tidyverse)
library(vegan)
library(fdrtool)
library(dendextend)
library(ggtext)
library(cowplot)
```

## Import data

Taxids with taxonomy are imported. Relative abundances of each taxid for individual samples are imported and vertically appended. We then reshape the dataframe. Taxid file and relative abundance file are joined to append taxonomy.

```{r import_taxonomy, echo = F}
taxid = read_tsv("taxonomy_files/tax.txt") %>% 
  dplyr::select(-contig) %>% 
  distinct()
write_csv(taxid, "tables/unique_taxid.csv")
ra_files = list.files(path = "taxonomy_files", pattern = "run", full.names = T)
rel_abund_long = map_dfr(ra_files, read_tsv)
rel_abund_notax = rel_abund_long %>% pivot_wider(names_from = "id", values_from = "relative")
rel_abund = rel_abund_notax %>% right_join(taxid)
```

Metadata is imported.
```{r import_metadata, echo = F}
metadata = read_tsv("tables/year2_sample_mapping.txt")
metadata2 = metadata %>% 
  mutate(Ethnicity = case_when(Ethnicity == "African_American" ~ "Black", 
                               Ethnicity == "Caucasian" ~ "White"))
```

## Filter data

The relative abundance table is transposed (without taxonomy) and file names are cleaned up. It is then joined with the metadata for PERMANOVA analyses. Individual tables for fecal and oral samples for each study period and for each ethnicity are created.
```{r ra_perm}
rel_abund_t = rel_abund_notax %>% 
  pivot_longer(cols = 2:163, names_to = "file_prefix", values_to = "relative_abundance") %>% 
  pivot_wider(names_from = "taxid", values_from = "relative_abundance") %>% 
  mutate_at("file_prefix", str_replace, "-", "_") %>% 
  left_join(metadata2)
relab_fecal = rel_abund_t %>% 
  filter(fecal_use == 1)
relab_fecal_filtered = relab_fecal %>% 
  mutate(total = rowSums(relab_fecal[,2:9395])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
relab_fecal_filtered_black = relab_fecal_filtered %>% 
  filter(Ethnicity == "Black")
relab_fecal_filtered_white = relab_fecal_filtered %>% 
  filter(Ethnicity == "White")
relab_fecal_filtered_before = relab_fecal_filtered %>% 
  filter(`Study period` == "Before")
relab_fecal_filtered_during = relab_fecal_filtered %>% 
  filter(`Study period` == "During")
relab_fecal_filtered_after = relab_fecal_filtered %>% 
  filter(`Study period` == "After")

relab_oral= rel_abund_t %>% 
  filter(oral_use == 1)
relab_oral_filtered = relab_oral %>% 
  mutate(total = rowSums(relab_oral[,2:9395])) %>% 
  filter(total > 0) %>% 
  select_if(~ !is.numeric(.) || sum(.) != 0)
relab_oral_filtered_black = relab_oral_filtered %>% 
  filter(Ethnicity == "Black")
relab_oral_filtered_white = relab_oral_filtered %>% 
  filter(Ethnicity == "White")
relab_oral_filtered_before = relab_oral_filtered %>% 
  filter(`Study period` == "Before")
relab_oral_filtered_after = relab_oral_filtered %>% 
  filter(`Study period` == "After")
```

The relative abundance table is transposed (with species names but no taxids), column names are cleaned up, low abundance taxa (<1%) are removed, and it is joined with the metadata for individual taxa abundance analyses. Individual tables for fecal and oral samples for each study period and for each ethnicity are created.

```{r ra_taxabund}
rel_abund_tax_t = rel_abund %>% 
  filter(Kingdom == "Bacteria") %>% 
  dplyr::select(-164) %>% 
  unite("Sci_name", Genus:Species, na.rm = F) %>% 
  pivot_longer(cols = 1:162, names_to = "file_prefix", values_to = "relative_abundance") %>%
  filter(Sci_name != "NA_NA") %>% 
  group_by(Sci_name, file_prefix) %>% 
  summarize(relative_abundance = sum(relative_abundance)) %>% 
  pivot_wider(names_from = "Sci_name", values_from = "relative_abundance") %>% 
  mutate_at("file_prefix", str_replace, "-", "_") %>% 
  select_if(~max(.) >= 0.01) %>% 
  left_join(metadata2)

rel_abund_tax_fecal = rel_abund_tax_t %>% 
  filter(fecal_use == 1) 
write.csv(rel_abund_tax_fecal, "tables/relative_abundance_values_fecal.csv")
rel_abund_tax_fecal = read_csv("tables/relative_abundance_values_fecal.csv")
rel_abund_tax_fecal_ethnicity = rel_abund_tax_fecal %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_fecal_ethnicity, "tables/relab_fecal.csv")
rel_abund_tax_fecal_before = rel_abund_tax_fecal %>% 
  filter(`Study period` == "Before")
rel_abund_tax_fecal_before_ethnicity = rel_abund_tax_fecal_before %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_fecal_before_ethnicity, "tables/relab_fecal_before.csv")
rel_abund_tax_fecal_during = rel_abund_tax_fecal %>% 
  filter(`Study period` == "During")
rel_abund_tax_fecal_during_ethnicity = rel_abund_tax_fecal_during %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_fecal_during_ethnicity, "tables/relab_fecal_during.csv")
rel_abund_tax_fecal_after = rel_abund_tax_fecal %>% 
  filter(`Study period` == "After")
rel_abund_tax_fecal_after_ethnicity = rel_abund_tax_fecal_after %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_fecal_after_ethnicity, "tables/relab_fecal_after.csv")

rel_abund_tax_oral = rel_abund_tax_t %>% 
  filter(oral_use == 1)
write.csv(rel_abund_tax_oral, "tables/relative_abundance_values_oral.csv")
rel_abund_tax_oral = read_csv("tables/relative_abundance_values_oral.csv")
rel_abund_tax_oral_ethnicity = rel_abund_tax_oral %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_oral_ethnicity, "tables/relab_oral.csv")
rel_abund_tax_oral_before = rel_abund_tax_oral %>% 
  filter(`Study period` == "Before")
rel_abund_tax_oral_before_ethnicity = rel_abund_tax_oral_before %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_oral_before_ethnicity, "tables/relab_oral_before.csv")
rel_abund_tax_oral_during = rel_abund_tax_oral %>% 
  filter(`Study period` == "During")
rel_abund_tax_oral_during_ethnicity = rel_abund_tax_oral_during %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_oral_during_ethnicity, "tables/relab_oral_during.csv")
rel_abund_tax_oral_after = rel_abund_tax_oral %>% 
  filter(`Study period` == "After")
rel_abund_tax_oral_after_ethnicity = rel_abund_tax_oral_after %>% 
  group_by(Ethnicity) %>% 
  summarize_if(is.numeric, mean, na.rm = T)
write_csv(rel_abund_tax_oral_after_ethnicity, "tables/relab_oral_after.csv")
```

## PERMANOVAs

Bray-Curtis and Jaccard distance matrices are created using vegan (v2.5-7). Matrices are constructed for all oral samples, all fecal samples, and oral and fecal samples broken down by ethnicity and study stage.

```{r beta_metrics, error=FALSE}
bray_fecal = vegdist(relab_fecal_filtered[,2:9266], method = "bray")
jacc_fecal = vegdist(relab_fecal_filtered[,2:9266], method = "jaccard", binary = T)
bray_fecal_black = vegdist(relab_fecal_filtered_black[,2:9266], 
                           method = "bray")
bray_fecal_white = vegdist(relab_fecal_filtered_white[,2:9266], 
                           method = "bray")
jacc_fecal_black = vegdist(relab_fecal_filtered_black[,2:9266], 
                           method = "jaccard", binary = T)
jacc_fecal_white = vegdist(relab_fecal_filtered_white[,2:9266], 
                           method = "jaccard", binary = T)
bray_fecal_before = vegdist(relab_fecal_filtered_before[,2:9266], 
                           method = "jaccard", binary = T)
bray_fecal_during = vegdist(relab_fecal_filtered_during[,2:9266], 
                           method = "jaccard", binary = T)
bray_fecal_after = vegdist(relab_fecal_filtered_after[,2:9266], 
                           method = "jaccard", binary = T)
jacc_fecal_before = vegdist(relab_fecal_filtered_before[,2:9266], 
                           method = "jaccard", binary = T)
jacc_fecal_during = vegdist(relab_fecal_filtered_during[,2:9266], 
                           method = "jaccard", binary = T)
jacc_fecal_after = vegdist(relab_fecal_filtered_after[,2:9266], 
                           method = "jaccard", binary = T) 

bray_oral = vegdist(relab_oral_filtered[,2:8063], method = "bray")
jacc_oral = vegdist(relab_oral_filtered[,2:8063], method = "jaccard", binary = T)
bray_oral_black = vegdist(relab_oral_filtered_black[,2:8063], method = "bray")
bray_oral_white = vegdist(relab_oral_filtered_white[,2:8063], method = "bray")
jacc_oral_black = vegdist(relab_oral_filtered_black[,2:8063], method = "jaccard", binary = T)
jacc_oral_white = vegdist(relab_oral_filtered_white[,2:8063], method = "jaccard", binary = T)
bray_oral_before = vegdist(relab_oral_filtered_before[,2:8063], 
                           method = "jaccard", binary = T)
bray_oral_after = vegdist(relab_oral_filtered_after[,2:8063], 
                          method = "jaccard", binary = T)
jacc_oral_before = vegdist(relab_oral_filtered_before[,2:8063], 
                           method = "jaccard", binary = T)
jacc_oral_after = vegdist(relab_oral_filtered_after[,2:8063], 
                          method = "jaccard", binary = T)
```

Differences in the taxonomic structure of the community associated with self-reported ethnicity, subject id, and study stage is assessed using permutational multivariate analysis of variance.

#### Bray-Curtis
Fecal
```{r permanova_brayf}
adonis2(bray_fecal ~ subject_id + `Study period` + subject_id*`Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(bray_fecal ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = relab_fecal_filtered, permutations = 4999)

anova(betadisper(bray_fecal, group = relab_fecal_filtered$Ethnicity))

anova(betadisper(bray_fecal, group = relab_fecal_filtered$`Study period`))

anova(betadisper(bray_fecal, group = relab_fecal_filtered$`Antibiotic use`))

anova(betadisper(bray_fecal, group = relab_fecal_filtered$`Contraceptive use`))

anova(betadisper(bray_fecal, group = relab_fecal_filtered$subject_id))

adonis2(bray_fecal ~ Ethnicity, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(bray_fecal_before ~ Ethnicity, 
        data = relab_fecal_filtered_before, 
        permutations = 4999)
adonis2(bray_fecal_during ~ Ethnicity, 
        data = relab_fecal_filtered_during, 
        permutations = 4999)
adonis2(bray_fecal_after ~ Ethnicity, 
        data = relab_fecal_filtered_after, 
        permutations = 4999)

adonis2(bray_fecal ~ `Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(bray_fecal_black ~ `Study period`, 
        data = relab_fecal_filtered_black, 
        permutations = 4999)
adonis2(bray_fecal_white ~ `Study period`, 
        data = relab_fecal_filtered_white, 
        permutations = 4999)
```

Oral
```{r permanova_brayo}
adonis2(bray_oral ~ subject_id + `Study period`, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(bray_oral ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use`, data = relab_oral_filtered, permutations = 4999)
adonis2(bray_oral ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = relab_oral_filtered, permutations = 4999)

anova(betadisper(bray_oral, group = relab_oral_filtered$Ethnicity))

anova(betadisper(bray_oral, group = relab_oral_filtered$`Study period`))

anova(betadisper(bray_oral, group = relab_oral_filtered$`Antibiotic use`))

anova(betadisper(bray_oral, group = relab_oral_filtered$`Contraceptive use`))

anova(betadisper(bray_oral, group = relab_oral_filtered$subject_id))

adonis2(bray_oral ~ Ethnicity, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(bray_oral_before ~ Ethnicity, 
        data = relab_oral_filtered_before, 
        permutations = 4999)
adonis2(bray_oral_after ~ Ethnicity, 
        data = relab_oral_filtered_after, 
        permutations = 4999)

adonis2(bray_oral ~ `Study period`, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(bray_oral_black ~ `Study period`, 
        data = relab_oral_filtered_black, 
        permutations = 4999)
adonis2(bray_oral_white ~ `Study period`, 
        data = relab_oral_filtered_white, 
        permutations = 4999)

```

#### Jaccard
Fecal
```{r permanova_jaccf}
adonis2(jacc_fecal ~ subject_id + `Study period` + subject_id*`Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_fecal ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = relab_fecal_filtered, permutations = 4999)

anova(betadisper(jacc_fecal, group = relab_fecal_filtered$Ethnicity))

anova(betadisper(jacc_fecal, group = relab_fecal_filtered$`Study period`))

anova(betadisper(jacc_fecal, group = relab_fecal_filtered$`Antibiotic use`))

anova(betadisper(jacc_fecal, group = relab_fecal_filtered$`Contraceptive use`))

anova(betadisper(jacc_fecal, group = relab_fecal_filtered$subject_id))

adonis2(jacc_fecal ~ Ethnicity, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_fecal_before ~ Ethnicity, 
        data = relab_fecal_filtered_before, 
        permutations = 4999)
adonis2(jacc_fecal_during ~ Ethnicity, 
        data = relab_fecal_filtered_during, 
        permutations = 4999)
adonis2(jacc_fecal_after ~ Ethnicity, 
        data = relab_fecal_filtered_after, 
        permutations = 4999)

adonis2(jacc_fecal ~ `Study period`, 
        data = relab_fecal_filtered, 
        permutations = 4999)
adonis2(jacc_fecal_black ~ `Study period`, 
        data = relab_fecal_filtered_black, 
        permutations = 4999)
adonis2(jacc_fecal_white ~ `Study period`, 
        data = relab_fecal_filtered_white, 
        permutations = 4999)
```

Oral
```{r permanova_jacco}
adonis2(jacc_oral ~ subject_id + `Study period`, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(bray_oral ~ Ethnicity + `Study period` + `Antibiotic use` + `Contraceptive use` + subject_id, data = relab_oral_filtered, permutations = 4999)

anova(betadisper(jacc_oral, group = relab_oral_filtered$Ethnicity))

anova(betadisper(jacc_oral, group = relab_oral_filtered$`Study period`))

anova(betadisper(jacc_oral, group = relab_oral_filtered$`Antibiotic use`))

anova(betadisper(jacc_oral, group = relab_oral_filtered$`Contraceptive use`))

anova(betadisper(jacc_oral, group = relab_oral_filtered$subject_id))

adonis2(jacc_oral ~ Ethnicity, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(jacc_oral_before ~ Ethnicity, 
        data = relab_oral_filtered_before, 
        permutations = 4999)
adonis2(jacc_oral_after ~ Ethnicity, 
        data = relab_oral_filtered_after, 
        permutations = 4999)

adonis2(jacc_oral ~ `Study period`, 
        data = relab_oral_filtered, 
        permutations = 4999)
adonis2(jacc_oral_black ~ `Study period`, 
        data = relab_oral_filtered_black, 
        permutations = 4999)
adonis2(jacc_oral_white ~ `Study period`, 
        data = relab_oral_filtered_white, 
        permutations = 4999)
```


## Taxonomic abundance

Wilcoxon Rank Sum tests (base R) were used to assess if self-reported Ethnicity is associated with the abundance of individual species identified at the rank of genus or above. P-values are corrected using fdrtool (v.1.2.16).

```{r taxa_models_fecal}
Ethnicity_estimatematrix = mat.or.vec(182,2)
Ethnicity_pvaluematrix = mat.or.vec(182,2)

rel_abund_tax_fecal = as.data.frame(rel_abund_tax_fecal)

for(i in 2:183) {
  variable = rel_abund_tax_fecal[,i]
  b = wilcox.test(variable ~ Ethnicity, 
                  data = rel_abund_tax_fecal,
                  paired = F, exact = F)
  Ethnicity_estimatematrix[i-1,2] = b$statistic
  Ethnicity_pvaluematrix[i-1,2] = b$p.value
  Ethnicity_estimatematrix[i-1,1] = names(rel_abund_tax_fecal)[i]
  Ethnicity_pvaluematrix[i-1,1] = names(rel_abund_tax_fecal)[i]
}


taxa_Ethnicity = bind_cols(as.data.frame(Ethnicity_estimatematrix[,1:2]), 
                     as.data.frame(Ethnicity_pvaluematrix[,2]))
write.csv(taxa_Ethnicity, "tables/taxa_fecal_wilcox_Ethnicity_y2.csv")
taxa_Ethnicity_nona = filter(taxa_Ethnicity, taxa_Ethnicity[,3] != "NaN") 
taxa_Ethnicity_nona[,3] = as.numeric(as.character(taxa_Ethnicity_nona[,3]))
taxa_Ethnicity_corrected = bind_cols(taxa_Ethnicity_nona, 
                               as.data.frame(fdrtool(taxa_Ethnicity_nona[,3], 
                                                     statistic = "pvalue", 
                                                     plot = F)))
write.csv(taxa_Ethnicity_corrected_full, "tables/taxa_fecal_wilcox_Ethnicity_corrected_y2.csv")
```

```{r taxa_models_oral}
Ethnicity_estimatematrix = mat.or.vec(182,2)
Ethnicity_pvaluematrix = mat.or.vec(182,2)

rel_abund_tax_oral = as.data.frame(rel_abund_tax_oral)

for(i in 2:183) {
  variable = rel_abund_tax_oral[,i]
  b = wilcox.test(variable ~ Ethnicity, 
                  data = rel_abund_tax_oral,
                  paired = F, exact = F)
  Ethnicity_estimatematrix[i-1,2] = b$statistic
  Ethnicity_pvaluematrix[i-1,2] = b$p.value
  Ethnicity_estimatematrix[i-1,1] = names(rel_abund_tax_oral)[i]
  Ethnicity_pvaluematrix[i-1,1] = names(rel_abund_tax_oral)[i]
}

taxa_Ethnicity = bind_cols(as.data.frame(Ethnicity_estimatematrix[,1:2]), 
                     as.data.frame(Ethnicity_pvaluematrix[,2]))
write.csv(taxa_Ethnicity, "tables/taxa_oral_wilcox_Ethnicity_y2.csv")
taxa_Ethnicity_nona = filter(taxa_Ethnicity, taxa_Ethnicity[,3] != "NaN") 
taxa_Ethnicity_nona[,3] = as.numeric(as.character(taxa_Ethnicity_nona[,3]))
taxa_Ethnicity_corrected = bind_cols(taxa_Ethnicity_nona, 
                               as.data.frame(fdrtool(taxa_Ethnicity_nona[,3], 
                                                     statistic = "pvalue", 
                                                     plot = F)))
write.csv(taxa_Ethnicity_corrected, "tables/taxa_oral_wilcox_Ethnicity_corrected_y2.csv")
```

### Heritable taxa

```{r heritable}
fecal_heri = read.csv("relative_abundance_heritable_fecal.csv")
fecal_heri_before = fecal_heri %>% 
  filter(Study.period == "Before")
fecal_heri_during = fecal_heri %>% 
  filter(Study.period == "During")
fecal_heri_after = fecal_heri %>% 
  filter(Study.period == "After")
oral_heri = read.csv("relative_abundance_heritable_oral.csv")
oral_heri_before = oral_heri %>% 
  filter(Study.period == "Before")
oral_heri_after = oral_heri %>% 
  filter(Study.period == "After")

veil_fecal_all = wilcox.test(Veillonella ~ Ethnicity, 
                  data = fecal_heri,
                  paired = F, exact = F)
veil_fecal_all

veil_fecal_before = wilcox.test(Veillonella ~ Ethnicity, 
                  data = fecal_heri_before,
                  paired = F, exact = F)
veil_fecal_before

veil_fecal_during = wilcox.test(Veillonella ~ Ethnicity, 
                  data = fecal_heri_during,
                  paired = F, exact = F)
veil_fecal_during

veil_fecal_after = wilcox.test(Veillonella ~ Ethnicity, 
                  data = fecal_heri_after,
                  paired = F, exact = F)
veil_fecal_after

veil_oral_all = wilcox.test(Veillonella ~ Ethnicity, 
                  data = oral_heri,
                  paired = F, exact = F)
veil_oral_all

veil_oral_before = wilcox.test(Veillonella ~ Ethnicity, 
                  data = oral_heri_before,
                  paired = F, exact = F)
veil_oral_before

veil_oral_after = wilcox.test(Veillonella ~ Ethnicity, 
                  data = oral_heri_after,
                  paired = F, exact = F)
veil_oral_after

chris_fecal_all = wilcox.test(Christensenellaceae ~ Ethnicity, 
                  data = fecal_heri,
                  paired = F, exact = F)
chris_fecal_all

chris_fecal_before = wilcox.test(Christensenellaceae ~ Ethnicity, 
                  data = fecal_heri_before,
                  paired = F, exact = F)
chris_fecal_before

chris_fecal_during = wilcox.test(Christensenellaceae ~ Ethnicity, 
                  data = fecal_heri_during,
                  paired = F, exact = F)
chris_fecal_during

chris_fecal_after = wilcox.test(Christensenellaceae ~ Ethnicity, 
                  data = fecal_heri_after,
                  paired = F, exact = F)
chris_fecal_after

chris_oral_all = wilcox.test(Christensenellaceae ~ Ethnicity, 
                  data = oral_heri,
                  paired = F, exact = F)
chris_oral_all

chris_oral_before = wilcox.test(Christensenellaceae ~ Ethnicity, 
                  data = oral_heri_before,
                  paired = F, exact = F)
chris_oral_before

chris_oral_after = wilcox.test(Christensenellaceae ~ Ethnicity, 
                  data = oral_heri_after,
                  paired = F, exact = F)
chris_oral_after

corio_fecal_all = wilcox.test(Coriobacteriaceae ~ Ethnicity, 
                  data = fecal_heri,
                  paired = F, exact = F)
corio_fecal_all

corio_fecal_before = wilcox.test(Coriobacteriaceae ~ Ethnicity, 
                  data = fecal_heri_before,
                  paired = F, exact = F)
corio_fecal_before

corio_fecal_during = wilcox.test(Coriobacteriaceae ~ Ethnicity, 
                  data = fecal_heri_during,
                  paired = F, exact = F)
corio_fecal_during

corio_fecal_after = wilcox.test(Coriobacteriaceae ~ Ethnicity, 
                  data = fecal_heri_after,
                  paired = F, exact = F)
corio_fecal_after

corio_oral_all = wilcox.test(Coriobacteriaceae ~ Ethnicity, 
                  data = oral_heri,
                  paired = F, exact = F)
corio_oral_all

corio_oral_before = wilcox.test(Coriobacteriaceae ~ Ethnicity, 
                  data = oral_heri_before,
                  paired = F, exact = F)
corio_oral_before

corio_oral_after = wilcox.test(Coriobacteriaceae ~ Ethnicity, 
                  data = oral_heri_after,
                  paired = F, exact = F)
corio_oral_after

odori_fecal_all = wilcox.test(Odoribacteraceae ~ Ethnicity, 
                  data = fecal_heri,
                  paired = F, exact = F)
odori_fecal_all

odori_fecal_before = wilcox.test(Odoribacteraceae ~ Ethnicity, 
                  data = fecal_heri_before,
                  paired = F, exact = F)
odori_fecal_before

odori_fecal_during = wilcox.test(Odoribacteraceae ~ Ethnicity, 
                  data = fecal_heri_during,
                  paired = F, exact = F)
odori_fecal_during

odori_fecal_after = wilcox.test(Odoribacteraceae ~ Ethnicity, 
                  data = fecal_heri_after,
                  paired = F, exact = F)
odori_fecal_after

odori_oral_all = wilcox.test(Odoribacteraceae ~ Ethnicity, 
                  data = oral_heri,
                  paired = F, exact = F)
odori_oral_all

odori_oral_before = wilcox.test(Odoribacteraceae ~ Ethnicity, 
                  data = oral_heri_before,
                  paired = F, exact = F)
odori_oral_before

odori_oral_after = wilcox.test(Odoribacteraceae ~ Ethnicity, 
                  data = oral_heri_after,
                  paired = F, exact = F)
odori_oral_after

pepto_fecal_all = wilcox.test(Peptococcaceae ~ Ethnicity, 
                  data = fecal_heri,
                  paired = F, exact = F)
pepto_fecal_all

pepto_fecal_before = wilcox.test(Peptococcaceae ~ Ethnicity, 
                  data = fecal_heri_before,
                  paired = F, exact = F)
pepto_fecal_before

pepto_fecal_during = wilcox.test(Peptococcaceae ~ Ethnicity, 
                  data = fecal_heri_during,
                  paired = F, exact = F)
pepto_fecal_during

pepto_fecal_after = wilcox.test(Peptococcaceae ~ Ethnicity, 
                  data = fecal_heri_after,
                  paired = F, exact = F)
pepto_fecal_after

pepto_oral_all = wilcox.test(Peptococcaceae ~ Ethnicity, 
                  data = oral_heri,
                  paired = F, exact = F)
pepto_oral_all

pepto_oral_before = wilcox.test(Peptococcaceae ~ Ethnicity, 
                  data = oral_heri_before,
                  paired = F, exact = F)
pepto_oral_before

pepto_oral_after = wilcox.test(Peptococcaceae ~ Ethnicity, 
                  data = oral_heri_after,
                  paired = F, exact = F)
pepto_oral_after

riken_fecal_all = wilcox.test(Rikenellaceae ~ Ethnicity, 
                  data = fecal_heri,
                  paired = F, exact = F)
riken_fecal_all

riken_fecal_before = wilcox.test(Rikenellaceae ~ Ethnicity, 
                  data = fecal_heri_before,
                  paired = F, exact = F)
riken_fecal_before

riken_fecal_during = wilcox.test(Rikenellaceae ~ Ethnicity, 
                  data = fecal_heri_during,
                  paired = F, exact = F)
riken_fecal_during

riken_fecal_after = wilcox.test(Rikenellaceae ~ Ethnicity, 
                  data = fecal_heri_after,
                  paired = F, exact = F)
riken_fecal_after

riken_oral_all = wilcox.test(Rikenellaceae ~ Ethnicity, 
                  data = oral_heri,
                  paired = F, exact = F)
riken_oral_all

riken_oral_before = wilcox.test(Rikenellaceae ~ Ethnicity, 
                  data = oral_heri_before,
                  paired = F, exact = F)
riken_oral_before

riken_oral_after = wilcox.test(Rikenellaceae ~ Ethnicity, 
                  data = oral_heri_after,
                  paired = F, exact = F)
riken_oral_after

verru_fecal_all = wilcox.test(Verrucomicrobiaceae ~ Ethnicity, 
                  data = fecal_heri,
                  paired = F, exact = F)
verru_fecal_all

verru_fecal_before = wilcox.test(Verrucomicrobiaceae ~ Ethnicity, 
                  data = fecal_heri_before,
                  paired = F, exact = F)
verru_fecal_before

verru_fecal_during = wilcox.test(Verrucomicrobiaceae ~ Ethnicity, 
                  data = fecal_heri_during,
                  paired = F, exact = F)
verru_fecal_during

verru_fecal_after = wilcox.test(Verrucomicrobiaceae ~ Ethnicity, 
                  data = fecal_heri_after,
                  paired = F, exact = F)
verru_fecal_after

verru_oral_all = wilcox.test(Verrucomicrobiaceae ~ Ethnicity, 
                  data = oral_heri,
                  paired = F, exact = F)
verru_oral_all

verru_oral_before = wilcox.test(Verrucomicrobiaceae ~ Ethnicity, 
                  data = oral_heri_before,
                  paired = F, exact = F)
verru_oral_before

verru_oral_after = wilcox.test(Verrucomicrobiaceae ~ Ethnicity, 
                  data = oral_heri_after,
                  paired = F, exact = F)
verru_oral_after

victi_fecal_all = wilcox.test(Victivallaceae ~ Ethnicity, 
                  data = fecal_heri,
                  paired = F, exact = F)
victi_fecal_all

victi_fecal_before = wilcox.test(Victivallaceae ~ Ethnicity, 
                  data = fecal_heri_before,
                  paired = F, exact = F)
victi_fecal_before

victi_fecal_during = wilcox.test(Victivallaceae ~ Ethnicity, 
                  data = fecal_heri_during,
                  paired = F, exact = F)
victi_fecal_during

victi_fecal_after = wilcox.test(Victivallaceae ~ Ethnicity, 
                  data = fecal_heri_after,
                  paired = F, exact = F)
victi_fecal_after

victi_oral_all = wilcox.test(Victivallaceae ~ Ethnicity, 
                  data = oral_heri,
                  paired = F, exact = F)
victi_oral_all

victi_oral_before = wilcox.test(Victivallaceae ~ Ethnicity, 
                  data = oral_heri_before,
                  paired = F, exact = F)
victi_oral_before

victi_oral_after = wilcox.test(Victivallaceae ~ Ethnicity, 
                  data = oral_heri_after,
                  paired = F, exact = F)
victi_oral_after

clost_fecal_all = wilcox.test(Clostridiales ~ Ethnicity, 
                  data = fecal_heri,
                  paired = F, exact = F)
clost_fecal_all

clost_fecal_before = wilcox.test(Clostridiales ~ Ethnicity, 
                  data = fecal_heri_before,
                  paired = F, exact = F)
clost_fecal_before

clost_fecal_during = wilcox.test(Clostridiales ~ Ethnicity, 
                  data = fecal_heri_during,
                  paired = F, exact = F)
clost_fecal_during

clost_fecal_after = wilcox.test(Clostridiales ~ Ethnicity, 
                  data = fecal_heri_after,
                  paired = F, exact = F)
clost_fecal_after

clost_oral_all = wilcox.test(Clostridiales ~ Ethnicity, 
                  data = oral_heri,
                  paired = F, exact = F)
clost_oral_all

clost_oral_before = wilcox.test(Clostridiales ~ Ethnicity, 
                  data = oral_heri_before,
                  paired = F, exact = F)
clost_oral_before

clost_oral_after = wilcox.test(Clostridiales ~ Ethnicity, 
                  data = oral_heri_after,
                  paired = F, exact = F)
clost_oral_after

fecal_all_p = c(veil_fecal_all$p.value, chris_fecal_all$p.value,
                corio_fecal_all$p.value, odori_fecal_all$p.value,
                pepto_fecal_all$p.value, riken_fecal_all$p.value,
                verru_fecal_all$p.value, victi_fecal_all$p.value,
                clost_fecal_all$p.value)

fecal_all_padj = p.adjust(fecal_all_p, method = "fdr")
fecal_all_padj

fecal_before_p = c(veil_fecal_before$p.value, chris_fecal_before$p.value,
                corio_fecal_before$p.value, odori_fecal_before$p.value,
                pepto_fecal_before$p.value, riken_fecal_before$p.value,
                verru_fecal_before$p.value, victi_fecal_before$p.value,
                clost_fecal_before$p.value)

fecal_before_padj = p.adjust(fecal_before_p, method = "fdr")
fecal_before_padj

fecal_during_p = c(veil_fecal_during$p.value, chris_fecal_during$p.value,
                corio_fecal_during$p.value, odori_fecal_during$p.value,
                pepto_fecal_during$p.value, riken_fecal_during$p.value,
                verru_fecal_during$p.value, victi_fecal_during$p.value,
                clost_fecal_during$p.value)

fecal_during_padj = p.adjust(fecal_during_p, method = "fdr")
fecal_during_padj

fecal_after_p = c(veil_fecal_after$p.value, chris_fecal_after$p.value,
                corio_fecal_after$p.value, odori_fecal_after$p.value,
                pepto_fecal_after$p.value, riken_fecal_after$p.value,
                verru_fecal_after$p.value, victi_fecal_after$p.value,
                clost_fecal_after$p.value)

fecal_after_padj = p.adjust(fecal_after_p, method = "fdr")
fecal_after_padj

oral_all_p = c(veil_oral_all$p.value, chris_oral_all$p.value,
                corio_oral_all$p.value, odori_oral_all$p.value,
                pepto_oral_all$p.value, riken_oral_all$p.value,
                verru_oral_all$p.value, victi_oral_all$p.value,
                clost_oral_all$p.value)

oral_all_padj = p.adjust(oral_all_p, method = "fdr")
oral_all_padj

oral_before_p = c(veil_oral_before$p.value, chris_oral_before$p.value,
                corio_oral_before$p.value, odori_oral_before$p.value,
                pepto_oral_before$p.value, riken_oral_before$p.value,
                verru_oral_before$p.value, victi_oral_before$p.value,
                clost_oral_before$p.value)

oral_before_padj = p.adjust(oral_before_p, method = "fdr")
oral_before_padj

oral_after_p = c(veil_oral_after$p.value, chris_oral_after$p.value,
                corio_oral_after$p.value, odori_oral_after$p.value,
                pepto_oral_after$p.value, riken_oral_after$p.value,
                verru_oral_after$p.value, victi_oral_after$p.value,
                clost_oral_after$p.value)

oral_after_padj = p.adjust(oral_after_p, method = "fdr")
oral_after_padj

fecal_all_ethnicity = fecal_heri %>%
  group_by(Ethnicity) %>% 
  summarize(across(Veillonella:Clostridiales, ~mean(.)))

fecal_during_ethnicity = fecal_heri_during %>%
  group_by(Ethnicity) %>% 
  summarize(across(Veillonella:Clostridiales, ~mean(.)))
  
fecal_after_ethnicity = fecal_heri_after %>%
  group_by(Ethnicity) %>% 
  summarize(across(Veillonella:Clostridiales, ~mean(.)))
```

## Alpha Diversity

Alpha diversity was calculated relative abundance tables. Shannon diversity and Shannon-Weaver evenness (Pielou's evenness) were calculated.

```{r alpha_d}
shannond_fecal = diversity(relab_fecal_filtered[,2:9266], 
                           index = "shannon")
shannond_oral = diversity(relab_oral[,2:8063], 
                          index = "shannon")

shannone_fecal = shannond_fecal/log(specnumber(relab_fecal_filtered[,2:9266]))
shannone_oral = shannond_oral/log(specnumber(relab_oral[,2:8063]))

fecal_diversity = cbind(relab_fecal_filtered[,1], relab_fecal_filtered[,9267:9291], shannond_fecal, shannone_fecal)
fecal_diversity$`Study period` = as.factor(fecal_diversity$`Study period`)
oral_diversity = cbind(relab_oral[,1], relab_oral_filtered[,8064:8088], shannond_oral, shannone_oral)
oral_diversity$`Study period` = as.factor(oral_diversity$`Study period`)
```

Non-parametric tests were used to examine differences between self-reported ethnicities and study period (before/during/after) in diversity. 
```{r alpha_models_fecal}
div_fecal = wilcox.test(shannond_fecal ~ Ethnicity, 
                        data = fecal_diversity, 
                        paired = F, exact = F)
div_fecal

even_fecal = wilcox.test(shannone_fecal ~ Ethnicity, 
                          data = fecal_diversity, 
                        paired = F, exact = F)
even_fecal

div_fecals = kruskal.test(shannond_fecal ~ `Study period`, 
                        data = fecal_diversity)
div_fecals

even_fecals = kruskal.test(shannone_fecal ~ `Study period`, 
                          data = fecal_diversity)
even_fecals
```

```{r alpha_models_oral}
div_oral = wilcox.test(shannond_oral ~ Ethnicity, 
                       data = oral_diversity,
                       paired = F, exact = F)
div_oral

even_oral = wilcox.test(shannone_oral ~ Ethnicity, 
                        data = oral_diversity,
                        paired = F, exact = F)
even_oral

div_orals = wilcox.test(shannond_oral ~ `Study period`, 
                       data = oral_diversity,
                       paired = F, exact = F)
div_orals

even_orals = wilcox.test(shannone_oral ~ `Study period`, 
                        data = oral_diversity,
                        paired = F, exact = F)
even_orals
```

## Dendrogram

Import data and filter into fecal and oral metagenomes.

```{r dendro_import}
y1tax = read_tsv("tables/taxon_meta_y1.txt")

y1taxoral = y1tax %>% 
  column_to_rownames("sample.name") %>% 
  filter(source == "oral") %>% 
  dplyr::select(7:9377)

y1taxfecal = y1tax %>% 
  column_to_rownames("sample.name") %>% 
  filter(source == "fecal") %>% 
  dplyr::select(7:9377)

y1vir = read_tsv("Virome_files_y1/virome.txt")
y1virfecal = y1vir %>% 
  column_to_rownames("sample") %>% 
  dplyr::select(5:4429)

y2taxfecal = relab_fecal_filtered %>% 
  column_to_rownames("SampleID")%>% 
  dplyr::select(2:9266)

y2taxoral = relab_oral_filtered %>% 
  column_to_rownames("SampleID")%>% 
  dplyr::select(2:8063)
```
Create labeled dendrograms for fecal and oral samples in each year.

```{r dendro_y1fecal}
y1fecal.dist <- vegdist(y1taxfecal, method = "bray", permutations = 999)
y1fecal.clus <- hclust(y1fecal.dist, "aver")

y1fcd = as.dendrogram(y1fecal.clus) 

y1taxf = y1tax %>% filter(source == "fecal")
Ethnicity = rep("Black", length(rownames(y1taxf)))
is_x = grepl("Caucasian", y1taxf$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y1taxf)))
is_x = grepl("during", y1taxf$stage)
Stage[is_x] = "During"
is_x = grepl("after", y1taxf$stage)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c(After = "#377eb8", Before = "#4daf4a", During = "#e41a1c")
col_Stage = cols_3[Stage]

Subject = rep("1", length(rownames(y1taxfecal)))
is_x = grepl("02", rownames(y1taxfecal))
Subject[is_x] = "2"
is_x = grepl("03", rownames(y1taxfecal))
Subject[is_x] = "3"
is_x = grepl("04", rownames(y1taxfecal))
Subject[is_x] = "4"
is_x = grepl("05", rownames(y1taxfecal))
Subject[is_x] = "5"
is_x = grepl("06", rownames(y1taxfecal))
Subject[is_x] = "6"
is_x = grepl("07", rownames(y1taxfecal))
Subject[is_x] = "7"
is_x = grepl("03", rownames(y1taxfecal))
Subject[is_x] = "3"
is_x = grepl("08", rownames(y1taxfecal))
Subject[is_x] = "8"
is_x = grepl("09", rownames(y1taxfecal))
Subject[is_x] = "9"
is_x = grepl("10", rownames(y1taxfecal))
Subject[is_x] = "10"
is_x = grepl("11", rownames(y1taxfecal))
Subject[is_x] = "11"
is_x = grepl("12", rownames(y1taxfecal))
Subject[is_x] = "12"
is_x = grepl("13", rownames(y1taxfecal))
Subject[is_x] = "13"
is_x = grepl("14", rownames(y1taxfecal))
Subject[is_x] = "14"
is_x = grepl("15", rownames(y1taxfecal))
Subject[is_x] = "15"
is_x = grepl("16", rownames(y1taxfecal))
Subject[is_x] = "16"
is_x = grepl("17", rownames(y1taxfecal))
Subject[is_x] = "17"
is_x = grepl("19", rownames(y1taxfecal))
Subject[is_x] = "19"
is_x = grepl("20", rownames(y1taxfecal))
Subject[is_x] = "20"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 50, l = 70)
col_Subject = cols_19[Subject]
labels_colors(y1fcd) = col_Subject[order.dendrogram(y1fcd)]

svg("figures/y1_fecal_dendrogram.svg", width = 15, height = 10)
y1fcd = assign_values_to_leaves_edgePar(y1fcd, 
                                        value = col_Subject[order.dendrogram(y1fcd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y1fcd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), y1fcd)
dev.off()
```

```{r dendro_y1oral}
y1oral.dist <- vegdist(y1taxoral, method = "bray", permutations = 999)
y1oral.clus <- hclust(y1oral.dist, "aver")

y1ocd = as.dendrogram(y1oral.clus) 

y1taxo = y1tax %>% filter(source == "oral")

Ethnicity = rep("Black", length(rownames(y1taxo)))
is_x = grepl("Caucasian", y1taxo$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y1taxo)))
is_x = grepl("after", y1taxo$stage)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c(After = "#377eb8", Before = "#4daf4a")
col_Stage = cols_3[Stage]

Subject = rep("1", length(rownames(y1taxoral)))
is_x = grepl("02", rownames(y1taxoral))
Subject[is_x] = "2"
is_x = grepl("03", rownames(y1taxoral))
Subject[is_x] = "3"
is_x = grepl("04", rownames(y1taxoral))
Subject[is_x] = "4"
is_x = grepl("05", rownames(y1taxoral))
Subject[is_x] = "5"
is_x = grepl("06", rownames(y1taxoral))
Subject[is_x] = "6"
is_x = grepl("07", rownames(y1taxoral))
Subject[is_x] = "7"
is_x = grepl("08", rownames(y1taxoral))
Subject[is_x] = "8"
is_x = grepl("09", rownames(y1taxoral))
Subject[is_x] = "9"
is_x = grepl("10", rownames(y1taxoral))
Subject[is_x] = "10"
is_x = grepl("11", rownames(y1taxoral))
Subject[is_x] = "11"
is_x = grepl("12", rownames(y1taxoral))
Subject[is_x] = "12"
is_x = grepl("13", rownames(y1taxoral))
Subject[is_x] = "13"
is_x = grepl("14", rownames(y1taxoral))
Subject[is_x] = "14"
is_x = grepl("15", rownames(y1taxoral))
Subject[is_x] = "15"
is_x = grepl("16", rownames(y1taxoral))
Subject[is_x] = "16"
is_x = grepl("17", rownames(y1taxoral))
Subject[is_x] = "17"
is_x = grepl("19", rownames(y1taxoral))
Subject[is_x] = "19"
is_x = grepl("20", rownames(y1taxoral))
Subject[is_x] = "20"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 50, l = 70)
col_Subject = cols_19[Subject]
labels_colors(y1ocd) = col_Subject[order.dendrogram(y1ocd)]

svg("figures/y1_oral_dendrogram.svg", width = 15, height = 10)
y1ocd = assign_values_to_leaves_edgePar(y1ocd, 
                                        value = col_Subject[order.dendrogram(y1ocd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y1ocd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), 
             y1ocd)
dev.off()
```

```{r dendro_y1virus}
y1virus.dist <- vegdist(y1virfecal, method = "bray", permutations = 999)
y1virus.clus <- hclust(y1virus.dist, "aver")

y1vcd = as.dendrogram(y1virus.clus) 

Ethnicity = rep("Black", length(rownames(y1vir)))
is_x = grepl("Caucasian", y1vir$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y1vir)))
is_x = grepl("During", y1vir$Stage)
Stage[is_x] = "During"
is_x = grepl("After", y1vir$Stage)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c("After" = "#377eb8", "Before" = "#4daf4a", "During" = "#e41a1c")
col_Stage = cols_3[Stage]

Subject = rep("1", length(rownames(y1virfecal)))
is_x = grepl("02", rownames(y1virfecal))
Subject[is_x] = "2"
is_x = grepl("03", rownames(y1virfecal))
Subject[is_x] = "3"
is_x = grepl("04", rownames(y1virfecal))
Subject[is_x] = "4"
is_x = grepl("05", rownames(y1virfecal))
Subject[is_x] = "5"
is_x = grepl("06", rownames(y1virfecal))
Subject[is_x] = "6"
is_x = grepl("07", rownames(y1virfecal))
Subject[is_x] = "7"
is_x = grepl("08", rownames(y1virfecal))
Subject[is_x] = "8"
is_x = grepl("09", rownames(y1virfecal))
Subject[is_x] = "9"
is_x = grepl("10", rownames(y1virfecal))
Subject[is_x] = "10"
is_x = grepl("11", rownames(y1virfecal))
Subject[is_x] = "11"
is_x = grepl("12", rownames(y1virfecal))
Subject[is_x] = "12"
is_x = grepl("13", rownames(y1virfecal))
Subject[is_x] = "13"
is_x = grepl("14", rownames(y1virfecal))
Subject[is_x] = "14"
is_x = grepl("15", rownames(y1virfecal))
Subject[is_x] = "15"
is_x = grepl("16", rownames(y1virfecal))
Subject[is_x] = "16"
is_x = grepl("17", rownames(y1virfecal))
Subject[is_x] = "17"
is_x = grepl("19", rownames(y1virfecal))
Subject[is_x] = "19"
is_x = grepl("20", rownames(y1virfecal))
Subject[is_x] = "20"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 50, l = 70)
col_Subject = cols_19[Subject]
labels_colors(y1vcd) = col_Subject[order.dendrogram(y1vcd)]

svg("figures/y1_virus_dendrogram.svg", width = 15, height = 10)
y1vcd = assign_values_to_leaves_edgePar(y1vcd, 
                                        value = col_Subject[order.dendrogram(y1vcd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y1vcd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), 
             y1vcd)
dev.off()
```

```{r dendro_y2fecal}
y2fecal.dist <- vegdist(y2taxfecal, method = "bray", permutations = 999)
y2fecal.clus <- hclust(y2fecal.dist, "aver")

y2fcd = as.dendrogram(y2fecal.clus) 

y2taxf = relab_fecal_filtered %>% 
  dplyr::select(9266:9291)
Ethnicity = rep("Black", length(rownames(y2taxf)))
is_x = grepl("White", y2taxf$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y2taxf)))
is_x = grepl("During", y2taxf$`Study period`)
Stage[is_x] = "During"
is_x = grepl("After", y2taxf$`Study period`)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c("After" = "#377eb8", "Before" = "#4daf4a", "During" = "#e41a1c")
col_Stage = cols_3[Stage]

Subject = rep("21", length(rownames(y2taxfecal)))
is_x = grepl("22", rownames(y2taxfecal))
Subject[is_x] = "22"
is_x = grepl("23", rownames(y2taxfecal))
Subject[is_x] = "23"
is_x = grepl("24", rownames(y2taxfecal))
Subject[is_x] = "24"
is_x = grepl("25", rownames(y2taxfecal))
Subject[is_x] = "25"
is_x = grepl("26", rownames(y2taxfecal))
Subject[is_x] = "26"
is_x = grepl("27", rownames(y2taxfecal))
Subject[is_x] = "27"
is_x = grepl("29", rownames(y2taxfecal))
Subject[is_x] = "29"
is_x = grepl("30", rownames(y2taxfecal))
Subject[is_x] = "30"
is_x = grepl("31", rownames(y2taxfecal))
Subject[is_x] = "31"
is_x = grepl("32", rownames(y2taxfecal))
Subject[is_x] = "32"
is_x = grepl("33", rownames(y2taxfecal))
Subject[is_x] = "33"
is_x = grepl("34", rownames(y2taxfecal))
Subject[is_x] = "34"
is_x = grepl("35", rownames(y2taxfecal))
Subject[is_x] = "35"
is_x = grepl("36", rownames(y2taxfecal))
Subject[is_x] = "36"
is_x = grepl("37", rownames(y2taxfecal))
Subject[is_x] = "37"
is_x = grepl("38", rownames(y2taxfecal))
Subject[is_x] = "38"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 30, l = 50)
col_Subject = cols_19[Subject]
labels_colors(y2fcd) = col_Subject[order.dendrogram(y2fcd)]

svg("figures/y2_fecal_dendrogram.svg", width = 15, height = 10)
y2fcd = assign_values_to_leaves_edgePar(y2fcd, 
                                        value = col_Subject[order.dendrogram(y2fcd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y2fcd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), 
             y2fcd)
dev.off()
```

```{r dendro_y2oral}
y2oral.dist <- vegdist(y2taxoral, method = "bray", permutations = 999)
y2oral.clus <- hclust(y2oral.dist, "aver")

y2ocd = as.dendrogram(y2oral.clus) 

y2taxo = relab_oral_filtered %>% 
  dplyr::select(8064:8088)

Ethnicity = rep("Black", length(rownames(y2taxo)))
is_x = grepl("White", y2taxo$Ethnicity)
Ethnicity[is_x] = "White"
Ethnicity = factor(Ethnicity)
n_Ethnicity = length(unique(Ethnicity))
cols_2 = c(Black = "#f5c4ca", White = "#87ceeb")
col_Ethnicity = cols_2[Ethnicity]

Stage = rep("Before", length(rownames(y2taxo)))
is_x = grepl("After", y2taxo$`Study period`)
Stage[is_x] = "After"
Stage = factor(Stage)
n_Stage = length(unique(Stage))
cols_3 = c("After" = "#377eb8", "Before" = "#4daf4a")
col_Stage = cols_3[Stage]

Subject = rep("21", length(rownames(y2taxoral)))
is_x = grepl("22", rownames(y2taxoral))
Subject[is_x] = "22"
is_x = grepl("23", rownames(y2taxoral))
Subject[is_x] = "23"
is_x = grepl("24", rownames(y2taxoral))
Subject[is_x] = "24"
is_x = grepl("25", rownames(y2taxoral))
Subject[is_x] = "25"
is_x = grepl("26", rownames(y2taxoral))
Subject[is_x] = "26"
is_x = grepl("27", rownames(y2taxoral))
Subject[is_x] = "27"
is_x = grepl("29", rownames(y2taxoral))
Subject[is_x] = "29"
is_x = grepl("30", rownames(y2taxoral))
Subject[is_x] = "30"
is_x = grepl("31", rownames(y2taxoral))
Subject[is_x] = "31"
is_x = grepl("32", rownames(y2taxoral))
Subject[is_x] = "32"
is_x = grepl("33", rownames(y2taxoral))
Subject[is_x] = "33"
is_x = grepl("34", rownames(y2taxoral))
Subject[is_x] = "34"
is_x = grepl("35", rownames(y2taxoral))
Subject[is_x] = "35"
is_x = grepl("36", rownames(y2taxoral))
Subject[is_x] = "36"
is_x = grepl("37", rownames(y2taxoral))
Subject[is_x] = "37"
is_x = grepl("38", rownames(y2taxoral))
Subject[is_x] = "38"
Subject = factor(Subject)
n_Subject = length(unique(Subject))
cols_19 = colorspace::rainbow_hcl(n_Subject, c = 30, l = 50)
col_Subject = cols_19[Subject]
labels_colors(y2ocd) = col_Subject[order.dendrogram(y2ocd)]

svg("figures/y2_oral_dendrogram.svg", width = 15, height = 10)
y2ocd = assign_values_to_leaves_edgePar(y2ocd, 
                                        value = col_Subject[order.dendrogram(y2ocd)], edgePar = "col") %>% 
  set("branches_lwd", 4)
layout.matrix = matrix(c(1,2), nrow = 2, ncol = 1)
layout(mat = layout.matrix, heights = c(6, 1))
plot(y2ocd)
colored_bars(cbind(col_Ethnicity, 
                   col_Stage), 
             y2ocd)
dev.off()
```
## NMDS plots

### Cohort 1 gut taxonomy
```{r nmds_y1ftax}
y1tax_ex = read_tsv("tables/taxon_meta_y1_extended.txt")
y1taxoral_ex = y1tax_ex %>% 
  column_to_rownames("sample.name") %>% 
  filter(source == "oral")
y1taxoral_ex_no20 = y1tax_ex %>% 
  column_to_rownames("sample.name") %>% 
  filter(source == "oral") %>% 
  filter(Subject != "mb20")

y1taxfecal_ex = y1tax_ex %>% 
  column_to_rownames("sample.name") %>% 
  filter(source == "fecal")
y1taxfecal_ex_no20 = y1tax_ex %>% 
  column_to_rownames("sample.name") %>% 
  filter(source == "fecal") %>% 
  filter(Subject != "mb20")

bray_fecal_y1_ex = vegdist(y1taxfecal_ex[,9:9379], method = "bray")
bray_oral_y1_ex = vegdist(y1taxoral_ex[,9:9379], method = "bray")
bray_fecal_y1_ex_no20 = vegdist(y1taxfecal_ex_no20[,9:9379], method = "bray")
bray_oral_y1_ex_no20 = vegdist(y1taxoral_ex_no20[,9:9379], method = "bray")

adonis2(bray_fecal_y1_ex ~ Ethnicity + stage + antibiotics + contraception + Subject, data = y1taxfecal_ex, permutations = 4999)

anova(betadisper(bray_fecal_y1_ex, group = y1taxfecal_ex$Ethnicity))

anova(betadisper(bray_fecal_y1_ex, group = y1taxfecal_ex$stage))

anova(betadisper(bray_fecal_y1_ex, group = y1taxfecal_ex$antibiotics))

anova(betadisper(bray_fecal_y1_ex, group = y1taxfecal_ex$contraception))

anova(betadisper(bray_fecal_y1_ex, group = y1taxfecal_ex$Subject))

adonis2(bray_oral_y1_ex ~ Ethnicity + stage + antibiotics + contraception + Subject, data = y1taxoral_ex, permutations = 4999)

anova(betadisper(bray_oral_y1_ex, group = y1taxoral_ex$Ethnicity))

anova(betadisper(bray_oral_y1_ex, group = y1taxoral_ex$stage))

anova(betadisper(bray_oral_y1_ex, group = y1taxoral_ex$antibiotics))

anova(betadisper(bray_oral_y1_ex, group = y1taxoral_ex$contraception))

anova(betadisper(bray_oral_y1_ex, group = y1taxoral_ex$Subject))

adonis2(bray_fecal_y1_ex_no20 ~ Ethnicity + stage + antibiotics + contraception + Subject, data = y1taxfecal_ex_no20, permutations = 4999)

anova(betadisper(bray_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$Ethnicity))

anova(betadisper(bray_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$stage))

anova(betadisper(bray_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$antibiotics))

anova(betadisper(bray_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$contraception))

anova(betadisper(bray_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$Subject))

adonis2(bray_oral_y1_ex_no20 ~ Ethnicity + stage + antibiotics + contraception + Subject, data = y1taxoral_ex_no20, permutations = 4999)

anova(betadisper(bray_oral_y1_ex_no20, group = y1taxoral_ex_no20$Ethnicity))

anova(betadisper(bray_oral_y1_ex_no20, group = y1taxoral_ex_no20$stage))

anova(betadisper(bray_oral_y1_ex_no20, group = y1taxoral_ex_no20$antibiotics))

anova(betadisper(bray_oral_y1_ex_no20, group = y1taxoral_ex_no20$contraception))

anova(betadisper(bray_oral_y1_ex_no20, group = y1taxoral_ex_no20$Subject))

jaccard_fecal_y1_ex = vegdist(y1taxfecal_ex[,9:9379], method = "jaccard", binary = T)
jaccard_oral_y1_ex = vegdist(y1taxoral_ex[,9:9379], method = "jaccard", binary = T)
jaccard_fecal_y1_ex_no20 = vegdist(y1taxfecal_ex_no20[,9:9379], method = "jaccard", binary = T)
jaccard_oral_y1_ex_no20 = vegdist(y1taxoral_ex_no20[,9:9379], method = "jaccard", binary = T)

adonis2(jaccard_fecal_y1_ex ~ Subject*stage, data = y1taxfecal_ex, permutations = 4999)

adonis2(jaccard_fecal_y1_ex ~ Ethnicity + stage + antibiotics + contraception + Subject, data = y1taxfecal_ex, permutations = 4999)

anova(betadisper(jaccard_fecal_y1_ex, group = y1taxfecal_ex$Ethnicity))

anova(betadisper(jaccard_fecal_y1_ex, group = y1taxfecal_ex$stage))

anova(betadisper(jaccard_fecal_y1_ex, group = y1taxfecal_ex$antibiotics))

anova(betadisper(jaccard_fecal_y1_ex, group = y1taxfecal_ex$contraception))

anova(betadisper(jaccard_fecal_y1_ex, group = y1taxfecal_ex$Subject))

adonis2(jaccard_oral_y1_ex ~ Ethnicity + stage + antibiotics + contraception + Subject, data = y1taxoral_ex, permutations = 4999)

anova(betadisper(jaccard_oral_y1_ex, group = y1taxoral_ex$Ethnicity))

anova(betadisper(jaccard_oral_y1_ex, group = y1taxoral_ex$stage))

anova(betadisper(jaccard_oral_y1_ex, group = y1taxoral_ex$antibiotics))

anova(betadisper(jaccard_oral_y1_ex, group = y1taxoral_ex$contraception))

anova(betadisper(jaccard_oral_y1_ex, group = y1taxoral_ex$Subject))

adonis2(jaccard_fecal_y1_ex_no20 ~ Ethnicity + stage + antibiotics + contraception + Subject, data = y1taxfecal_ex_no20, permutations = 4999)

anova(betadisper(jaccard_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$Ethnicity))

anova(betadisper(jaccard_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$stage))

anova(betadisper(jaccard_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$antibiotics))

anova(betadisper(jaccard_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$contraception))

anova(betadisper(jaccard_fecal_y1_ex_no20, group = y1taxfecal_ex_no20$Subject))

adonis2(jaccard_oral_y1_ex_no20 ~ Ethnicity + stage + antibiotics + contraception + Subject, data = y1taxoral_ex_no20, permutations = 4999)

anova(betadisper(jaccard_oral_y1_ex_no20, group = y1taxoral_ex_no20$Ethnicity))

anova(betadisper(jaccard_oral_y1_ex_no20, group = y1taxoral_ex_no20$stage))

anova(betadisper(jaccard_oral_y1_ex_no20, group = y1taxoral_ex_no20$antibiotics))

anova(betadisper(jaccard_oral_y1_ex_no20, group = y1taxoral_ex_no20$contraception))

anova(betadisper(jaccard_oral_y1_ex_no20, group = y1taxoral_ex_no20$Subject))

bray_fecal_y1 = vegdist(y1taxfecal_ex[,9:9379], method = "bray")

y1taxf = y1taxfecal_ex %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White")) %>% 
  mutate(stage = case_when(stage == "before" ~ "Before",
                           stage == "during" ~ "During",
                           stage == "after" ~ "After"))

mds_bray_f_y1<-metaMDS(bray_fecal_y1, k=2, trymax=999)
mds_bray_f_y1_points<-mds_bray_f_y1$points
mds_bray_f_y1_points2<-merge(x=mds_bray_f_y1_points, y = y1taxf, 
                                  by.x = "row.names", 
                             by.y = "row.names")

brayfy1 <- ggplot(mds_bray_f_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1)),
        legend.background = element_rect(fill = NA),
        legend.position = c(0.1, 0.1)) +
  stat_ellipse(data = mds_bray_f_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 6.6% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.134",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfy1


brayfy1s <- ggplot(mds_bray_f_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = stage)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1)),
        legend.background = element_rect(fill = NA),
        legend.position = c(0.1, 0.1)) +
  stat_ellipse(data = mds_bray_f_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = stage), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 0.8%</b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.130",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfy1s

jacc_fecal_y1 = vegdist(y1taxfecal_ex[,9:9379], method = "jaccard", binary = T)

mds_jacc_f_y1<-metaMDS(jacc_fecal_y1, k=2, trymax=999)
mds_jacc_f_y1_points<-mds_jacc_f_y1$points
mds_jacc_f_y1_points2<-merge(x=mds_jacc_f_y1_points, y = y1taxf, 
                                  by.x = "row.names", 
                             by.y = "row.names")

jaccfy1 <- ggplot(mds_jacc_f_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1)),
        legend.background = element_rect(fill = NA),
        legend.position = c(0.1, 0.1)) +
  stat_ellipse(data = mds_jacc_f_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 3.9% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.132",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccfy1

jaccfy1s <- ggplot(mds_jacc_f_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = stage)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.title = element_blank(),
        legend.text = element_text(size = rel(1)),
        legend.background = element_rect(fill = NA),
        legend.position = c(0.1, 0.1)) +
  stat_ellipse(data = mds_jacc_f_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = stage), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.016, 
           r<sup>2</sup> = 1.7%</b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.132",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccfy1s

```
### Cohort 2 gut taxonomy
```{r nmds_y2ftax}
bray_fecal_y2 = vegdist(relab_fecal_filtered[,2:9266], method = "bray")

y2taxf = relab_fecal_filtered %>% 
  dplyr::select(9267:9291)

mds_bray_f_y2<-metaMDS(bray_fecal_y2, k=2, trymax=999)
mds_bray_f_y2_points<-mds_bray_f_y2$points
mds_bray_f_y2_points2<-merge(x=mds_bray_f_y2_points, y = y2taxf, 
                                  by.x = "row.names", 
                             by.y = "row.names")

brayfy2 <- ggplot(mds_bray_f_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_f_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 5.5% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.124",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfy2

brayfy2s <- ggplot(mds_bray_f_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
       legend.position = "none") +
  stat_ellipse(data = mds_bray_f_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = `Study period`), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.002, 
           r<sup>2</sup> = 0.08%</b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.124",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfy2s

jacc_fecal_y2 = vegdist(relab_fecal_filtered[,2:9266], method = "jaccard", binary = T)

mds_jacc_f_y2<-metaMDS(jacc_fecal_y2, k=2, trymax=999)
mds_jacc_f_y2_points<-mds_jacc_f_y2$points
mds_jacc_f_y2_points2<-merge(x=mds_jacc_f_y2_points, y = y2taxf, 
                                  by.x = "row.names", 
                             by.y = "row.names")

jaccfy2 <- ggplot(mds_jacc_f_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_jacc_f_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 3.0% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.126",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccfy2

jaccfy2s <- ggplot(mds_jacc_f_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
       legend.position = "none") +
  stat_ellipse(data = mds_jacc_f_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = `Study period`), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.050, 
           r<sup>2</sup> = 1.3%</b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.126",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccfy2s
```

### Cohort 1 oral taxonomy
```{r nmds_y1otax}
bray_oral_y1 = vegdist(y1taxoral_ex[,9:9379], method = "bray")

y1taxo = y1taxoral_ex %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White")) %>% 
  mutate(stage = case_when(stage == "before" ~ "Before",
                           stage == "after" ~ "After"))

mds_bray_o_y1<-metaMDS(bray_oral_y1, k=2, trymax=999)
mds_bray_o_y1_points<-mds_bray_o_y1$points
mds_bray_o_y1_points2<-merge(x=mds_bray_o_y1_points, y = y1taxo, 
                                  by.x = "row.names", 
                             by.y = "row.names")

brayoy1 <- ggplot(mds_bray_o_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 10.4% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.100",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayoy1

brayoy1s <- ggplot(mds_bray_o_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = stage)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = stage), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.297, 
           r<sup>2</sup> = 0.8%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.100",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayoy1s

jacc_oral_y1 = vegdist(y1taxoral_ex[,9:9379], method = "jaccard", binary = T)

mds_jacc_o_y1<-metaMDS(jacc_oral_y1, k=2, trymax=999)
mds_jacc_o_y1_points<-mds_jacc_o_y1$points
mds_jacc_o_y1_points2<-merge(x=mds_jacc_o_y1_points, y = y1taxo, 
                                  by.x = "row.names", 
                             by.y = "row.names")

jaccoy1 <- ggplot(mds_jacc_o_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_jacc_o_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> = 0.007, 
           r<sup>2</sup> = 3.5% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.151",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccoy1

jaccoy1s <- ggplot(mds_jacc_o_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = stage)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_jacc_o_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = stage), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 783, 
           r<sup>2</sup> = 1.8%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.151",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccoy1s
```
### Cohort 2 oral taxonomy
```{r nmds_y2otax}
bray_oral_y2 = vegdist(relab_oral_filtered[,2:8063], method = "bray")

y2taxo = relab_oral_filtered %>% 
  dplyr::select(8064:8088)

mds_bray_o_y2<-metaMDS(bray_oral_y2, k=2, trymax=999)
mds_bray_o_y2_points<-mds_bray_o_y2$points
mds_bray_o_y2_points2<-merge(x=mds_bray_o_y2_points, y = y2taxo, 
                                  by.x = "row.names", 
                             by.y = "row.names")

brayoy2 <- ggplot(mds_bray_o_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 6.5% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.149",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayoy2

brayoy2s <- ggplot(mds_bray_o_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_o_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = `Study period`), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.139, 
           r<sup>2</sup> = 1.3%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.149",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayoy2s

jacc_oral_y2 = vegdist(relab_oral_filtered[,2:8063], method = "jaccard", binary = T)

mds_jacc_o_y2<-metaMDS(jacc_oral_y2, k=2, trymax=999)
mds_jacc_o_y2_points<-mds_jacc_o_y2$points
mds_jacc_o_y2_points2<-merge(x=mds_jacc_o_y2_points, y = y2taxo, 
                                  by.x = "row.names", 
                             by.y = "row.names")

jaccoy2 <- ggplot(mds_jacc_o_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_jacc_o_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 6.5%</b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.126",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccoy2

jaccoy2s <- ggplot(mds_jacc_o_y2_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = `Study period`)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_jacc_o_y2_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = `Study period`), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.124, 
           r<sup>2</sup> = 1.3%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.126",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccoy2s
```
### Cohort 1 gut virome taxonomy
```{r nmds_y1fvtax}
vir_tax = read_tsv("Virome_files_y1/virome.txt")

bray_fecalv_y1 = vegdist(vir_tax[,7:4432], method = "bray")

adonis2(bray_fecalv_y1 ~ Ethnicity + Stage + antibiotics + contraception + Subject, data = vir_tax, permutations = 4999)
anova(betadisper(bray_fecalv_y1, group = vir_tax$Ethnicity))

y1taxfv = vir_tax %>% 
  mutate(Ethnicity = case_when(
    Ethnicity == "African_American" ~ "Black", 
    Ethnicity == "Caucasian" ~ "White"))

mds_bray_fv_y1<-metaMDS(bray_fecalv_y1, k=2, trymax=999)
mds_bray_fv_y1_points<-mds_bray_fv_y1$points
mds_bray_fv_y1_points2<-merge(x=mds_bray_fv_y1_points, 
                              y = y1taxfv, 
                              by.x = "row.names", 
                              by.y = "row.names")

brayfvy1 <- ggplot(mds_bray_fv_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_bray_fv_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 5.2% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.160",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfvy1

brayfvy1s <- ggplot(mds_bray_fv_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Stage)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
       legend.position = "none") +
  stat_ellipse(data = mds_bray_fv_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Stage), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.181, 
           r<sup>2</sup> = 1.1%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.160",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
brayfvy1s

jacc_fecalv_y1 = vegdist(vir_tax[,7:4432], method = "jaccard", binary = T)

adonis2(jacc_fecalv_y1 ~ Ethnicity + Stage + antibiotics + contraception + Subject, data = vir_tax, permutations = 4999)
anova(betadisper(jacc_fecalv_y1, group = vir_tax$Ethnicity))


mds_jacc_fv_y1<-metaMDS(jacc_fecalv_y1, k=2, trymax=999)
mds_jacc_fv_y1_points<-mds_jacc_fv_y1$points
mds_jacc_fv_y1_points2<-merge(x=mds_jacc_fv_y1_points, 
                              y = y1taxfv, 
                              by.x = "row.names", 
                              by.y = "row.names")

jaccfvy1 <- ggplot(mds_jacc_fv_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Ethnicity)) +
  geom_point(size=3) + scale_color_manual(
    values = c("White"="#87ceeb","Black"="#f5c4ca")) +
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
        legend.position = "none") +
  stat_ellipse(data = mds_jacc_fv_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Ethnicity), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<b><i>p</i> < 0.001, 
           r<sup>2</sup> = 4.8% </b>", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.201",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccfvy1

jaccfvy1s <- ggplot(mds_jacc_fv_y1_points2, 
                  aes(x = MDS1, y = MDS2, 
                      color = Stage)) +
  geom_point(size=3) + scale_color_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + scale_fill_manual(
    values = c("Before"="#238b45","During" = "#d94801", "After"="#2171b5")) + 
  theme(panel.background = element_rect(fill = 'white', colour = 'black'), 
        legend.key=element_blank()) + 
  theme(axis.title.x=element_blank(), 
        axis.title.y=element_blank(),
        plot.title = element_blank(),
       legend.position = "none") +
  stat_ellipse(data = mds_jacc_fv_y1_points2, 
               aes(x = MDS1, y = MDS2, 
                   fill = Stage), 
                   alpha = 0, size = 0.8, geom = "polygon",
               linetype = 2,
               type = "t", level = 0.95) +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "<i>p</i> = 0.253, 
           r<sup>2</sup> = 1.3%", 
           x = Inf, y = Inf, size = 5,
           hjust = 1, vjust = 1.1)  +
  annotate(geom = "richtext", fill = NA, label.color = NA,
           label = "Stress = 0.201",
           x = Inf, y = -Inf, size = 4,
           hjust = 1, vjust = 0)
jaccfvy1s
```
